From 93e075007e0cb654f3ddcb126c0c8613d4c21cf4 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Fri, 16 Feb 2024 15:26:23 -0800
Subject: [PATCH] Add debug log for PSI

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index f3a38c010afd..84d9b0fd8ace 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1532,6 +1532,7 @@ static void damos_set_quota_goal_current_value(struct damos_quota_goal *goal)
 		now_psi_total = damos_get_some_mem_psi_total();
 		goal->current_value = now_psi_total - goal->last_psi_total;
 		goal->last_psi_total = now_psi_total;
+		pr_info("PSI current value %lu\n", goal->current_value);
 		break;
 	default:
 		break;
-- 
2.39.2

