From: SeongJae Park <sj@kernel.org>
Date: Thu, 20 Feb 2025 09:52:53 -0800
Subject: [PATCH] add filters in core,ops filters first

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs-schemes.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/mm/damon/sysfs-schemes.c b/mm/damon/sysfs-schemes.c
index bb25be962f54..5cc1bcb3973e 100644
--- a/mm/damon/sysfs-schemes.c
+++ b/mm/damon/sysfs-schemes.c
@@ -2232,22 +2232,21 @@ static struct damos *damon_sysfs_mk_scheme(
 		return NULL;
 	}
 
-	err = damon_sysfs_add_scheme_filters(scheme, sysfs_scheme->filters);
+	err = damon_sysfs_add_scheme_filters(scheme, sysfs_scheme->core_filters);
 	if (err) {
 		damon_destroy_scheme(scheme);
 		return NULL;
 	}
-	err = damon_sysfs_add_scheme_filters(scheme, sysfs_scheme->core_filters);
+	err = damon_sysfs_add_scheme_filters(scheme, sysfs_scheme->ops_filters);
 	if (err) {
 		damon_destroy_scheme(scheme);
 		return NULL;
 	}
-	err = damon_sysfs_add_scheme_filters(scheme, sysfs_scheme->ops_filters);
+	err = damon_sysfs_add_scheme_filters(scheme, sysfs_scheme->filters);
 	if (err) {
 		damon_destroy_scheme(scheme);
 		return NULL;
 	}
-
 	return scheme;
 }
 
-- 
2.39.5

