From: SeongJae Park <sj@kernel.org>
Date: Fri, 16 Feb 2024 15:26:23 -0800
Subject: [PATCH] Add debug log for PSI

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index b30bf151fe69..b810de3ac213 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2093,6 +2093,7 @@ static void damos_set_quota_goal_current_value(struct damos_quota_goal *goal)
 		now_psi_total = damos_get_some_mem_psi_total();
 		goal->current_value = now_psi_total - goal->last_psi_total;
 		goal->last_psi_total = now_psi_total;
+		pr_info("PSI current value %lu\n", goal->current_value);
 		break;
 	case DAMOS_QUOTA_NODE_MEM_USED_BP:
 	case DAMOS_QUOTA_NODE_MEM_FREE_BP:
-- 
2.39.5

