From 81c6b8307499bf2c65af09c33b94574f8dbee039 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Sat, 11 May 2024 10:48:28 -0700
Subject: [PATCH] address comment

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index ccbfb2722e98..c131beaf7cc5 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -241,11 +241,11 @@ static bool damos_pa_filter_out(struct damos *scheme, struct folio *folio)
 }
 
 enum migration_mode {
-	MIG_PAGEOUT,
+	DAMOS_MIG_PAGEOUT,
 };
 
 static unsigned long damon_pa_migrate(struct damon_region *r, struct damos *s,
-				      enum migration_mode mm)
+				      enum migration_mode mode)
 {
 	unsigned long addr, applied;
 	LIST_HEAD(folio_list);
@@ -288,8 +288,8 @@ static unsigned long damon_pa_migrate(struct damon_region *r, struct damos *s,
 	}
 	if (install_young_filter)
 		damos_destroy_filter(filter);
-	switch (mm) {
-	case MIG_PAGEOUT:
+	switch (mode) {
+	case DAMOS_MIG_PAGEOUT:
 		applied = reclaim_pages(&folio_list);
 		break;
 	default:
@@ -343,7 +343,7 @@ static unsigned long damon_pa_apply_scheme(struct damon_ctx *ctx,
 {
 	switch (scheme->action) {
 	case DAMOS_PAGEOUT:
-		return damon_pa_migrate(r, scheme, MIG_PAGEOUT);
+		return damon_pa_migrate(r, scheme, DAMOS_MIG_PAGEOUT);
 	case DAMOS_LRU_PRIO:
 		return damon_pa_mark_accessed(r, scheme);
 	case DAMOS_LRU_DEPRIO:
-- 
2.39.2

