From: SeongJae Park <sj@kernel.org>
Date: Fri, 17 Jan 2025 13:59:07 -0800
Subject: [PATCH] damon_ca_help: support translating cache group to physical
 address ranges

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 damon_ca_help.py | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/damon_ca_help.py b/damon_ca_help.py
index ed1d8faa2491..e02a1693e699 100755
--- a/damon_ca_help.py
+++ b/damon_ca_help.py
@@ -383,6 +383,9 @@ def main():
     parser.add_argument('max_paddr', help='maximum physical address')
     parser.add_argument('--translate_addr', metavar='<physical address>',
                         help='show cache group/set that will contain the content of the address')
+    parser.add_argument('--translate_group', metavar='<cache group index>',
+                        help='show physical address ranges that will backup the content of the cache group')
+
     args = parser.parse_args()
 
     cache_size = text_to_bytes(args.cache_size)
@@ -436,6 +439,14 @@ def main():
         print('    which is in %s-th cache group' %
               format_nr(cache_group_index, False))
 
+    if args.translate_group is not None:
+        cache_group_idx = text_to_nr(args.translate_group)
+        start_addr = cache_group_idx * sz_page
+        while start_addr < max_paddr:
+            if start_addr > min_paddr:
+                print('[%s, %s)' % (format_nr(start_addr, False),
+                                    format_nr(start_addr + sz_page, False)))
+            start_addr += nr_groups * sz_page
 
 if __name__ == '__main__':
     main()
-- 
2.39.5

