From: SeongJae Park <sj@kernel.org>
Date: Fri, 17 Jan 2025 13:36:54 -0800
Subject: [PATCH] damon_ca_help: support physical address to cache group
 translation

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 damon_ca_help.py | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/damon_ca_help.py b/damon_ca_help.py
index 2dfc9bd12338..ed1d8faa2491 100755
--- a/damon_ca_help.py
+++ b/damon_ca_help.py
@@ -381,6 +381,8 @@ def main():
     parser.add_argument('cache_ways', help='cache associativity')
     parser.add_argument('min_paddr', help='minimum physical address')
     parser.add_argument('max_paddr', help='maximum physical address')
+    parser.add_argument('--translate_addr', metavar='<physical address>',
+                        help='show cache group/set that will contain the content of the address')
     args = parser.parse_args()
 
     cache_size = text_to_bytes(args.cache_size)
@@ -423,5 +425,17 @@ def main():
 
     coverage_per_group = sz_avail_mem / nr_groups
 
+    if args.translate_addr is not None:
+        paddr = text_to_bytes(args.translate_addr)
+        cache_set_index = paddr / cache_line_size % nr_cache_sets
+        cache_group_index = cache_set_index / sets_per_group
+        print('content of physical address %s will be cached at' %
+              format_nr(paddr, False))
+        print('    %s-th cache set,'%
+              format_nr(cache_set_index, False))
+        print('    which is in %s-th cache group' %
+              format_nr(cache_group_index, False))
+
+
 if __name__ == '__main__':
     main()
-- 
2.39.5

