From: SeongJae Park <sj@kernel.org>
Date: Fri, 17 Jan 2025 12:46:15 -0800
Subject: [PATCH] damon_ca_help: calculate and print metrics for DAMON cops

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 damon_ca_help.py | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/damon_ca_help.py b/damon_ca_help.py
index 69820e169b2f..2dfc9bd12338 100755
--- a/damon_ca_help.py
+++ b/damon_ca_help.py
@@ -392,8 +392,36 @@ def main():
     print('cache size: %s' % format_sz(cache_size, False))
     print('cache line size: %s' % format_sz(cache_line_size, False))
     print('cache associativity ways: %s' % format_nr(cache_ways, False))
+
+    sz_cache_set = cache_line_size * cache_ways
+    print('cache set size: %s' % format_sz(sz_cache_set, False))
+
+    nr_cache_sets = cache_size / sz_cache_set
+    print('number of cache sets: %s' % format_nr(nr_cache_sets, False))
+
+    sz_page = 4096
+    print('page size: %s' % format_sz(sz_page, False))
+    # group is a group of cache set that can contain contents of a page
+    sets_per_group = sz_page / cache_line_size
+    print('number of cache sets per group: %s' %
+          format_nr(sets_per_group, False))
+
+    nr_groups = nr_cache_sets / sets_per_group
+    print('number of groups in the cache: %s' % format_nr(nr_groups, False))
+
     print('min physical address: %s' % format_sz(min_paddr, False))
     print('max physical address: %s' % format_sz(max_paddr, False))
 
+    sz_avail_mem = max_paddr - min_paddr
+    print('total configurable memory: %s' % format_sz(sz_avail_mem, False))
+    nr_avail_pages = sz_avail_mem / sz_page
+    print('total configurable pages: %s' % format_nr(nr_avail_pages, False))
+
+    coverage_page = nr_avail_pages / nr_groups
+    print('coverage of each cache group: %s' %
+          format_sz(coverage_page * sz_page, False))
+
+    coverage_per_group = sz_avail_mem / nr_groups
+
 if __name__ == '__main__':
     main()
-- 
2.39.5

