From: SeongJae Park <sj@kernel.org>
Date: Fri, 17 Jan 2025 12:36:14 -0800
Subject: [PATCH] damon_ca_help: show the inptu in a better way

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 damon_ca_help.py | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/damon_ca_help.py b/damon_ca_help.py
index 84b7a9abd079..69820e169b2f 100755
--- a/damon_ca_help.py
+++ b/damon_ca_help.py
@@ -383,7 +383,17 @@ def main():
     parser.add_argument('max_paddr', help='maximum physical address')
     args = parser.parse_args()
 
-    print(args)
+    cache_size = text_to_bytes(args.cache_size)
+    cache_line_size = text_to_bytes(args.cache_line_size)
+    cache_ways = text_to_nr(args.cache_ways)
+    min_paddr = text_to_bytes(args.min_paddr)
+    max_paddr = text_to_bytes(args.max_paddr)
+
+    print('cache size: %s' % format_sz(cache_size, False))
+    print('cache line size: %s' % format_sz(cache_line_size, False))
+    print('cache associativity ways: %s' % format_nr(cache_ways, False))
+    print('min physical address: %s' % format_sz(min_paddr, False))
+    print('max physical address: %s' % format_sz(max_paddr, False))
 
 if __name__ == '__main__':
     main()
-- 
2.39.5

