From: SeongJae Park <sj@kernel.org>
Date: Mon, 24 Feb 2025 17:53:20 -0800
Subject: [PATCH] fixup debug log

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index b5b905861635..2ebeb09b85b6 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1453,6 +1453,9 @@ static unsigned long damon_get_intervals_score(struct damon_ctx *c)
 		}
 	}
 	target_access_events = max_access_events * goal_bp / 10000;
+
+	pr_info("access_events %lu/%lu\n", access_events, target_access_events);
+
 	return access_events * 10000 / target_access_events;
 }
 
@@ -1473,8 +1476,7 @@ static unsigned long damon_get_intervals_adaptation_bp(struct damon_ctx *c)
 	if (adaptation_bp <= 10000)
 		adaptation_bp = 5000 + adaptation_bp / 2;
 
-	pr_info("access_events %lu/%lu, score_bp %lu, adaptation bp %lu\n",
-			heats, target_heats, score_bp, adaptation_bp);
+	pr_info("score_bp %lu, adaptation bp %lu\n", score_bp, adaptation_bp);
 
 	return adaptation_bp;
 }
-- 
2.39.5

