From: SeongJae Park <sj@kernel.org>
Date: Tue, 5 Aug 2025 22:23:03 -0700
Subject: [PATCH] include/linux/kasan-enabled: temporal build fix

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/kasan-enabled.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/include/linux/kasan-enabled.h b/include/linux/kasan-enabled.h
index b5857e15ef14..a53d112b1020 100644
--- a/include/linux/kasan-enabled.h
+++ b/include/linux/kasan-enabled.h
@@ -8,11 +8,22 @@ extern bool kasan_arg_disabled;
 
 DECLARE_STATIC_KEY_FALSE(kasan_flag_enabled);
 
+#ifdef CONFIG_KASAN
+
 static __always_inline bool kasan_enabled(void)
 {
 	return static_branch_likely(&kasan_flag_enabled);
 }
 
+#else /* CONFIG_KASAN */
+
+static inline bool kasan_enabled(void)
+{
+	return false;
+}
+
+#endif
+
 #ifdef CONFIG_KASAN_HW_TAGS
 static inline bool kasan_hw_tags_enabled(void)
 {
-- 
2.39.5

