From 47f4626690d1e83c8450bfe5c9ce7fa86fb8db37 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Thu, 22 Aug 2024 13:56:23 -0700
Subject: [PATCH] include/linux/damon: intorduce regions priorities histogram
 per context

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index fcccaa9b9d54..6342d8f9b0fd 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -661,6 +661,8 @@ struct damon_ctx {
 	unsigned long next_ops_update_sis;
 	/* for waiting until the execution of the kdamond_fn is started */
 	struct completion kdamond_started;
+	/* for scheme quotas prioritization */
+	unsigned long *regions_score_histogram;
 
 /* public: */
 	struct task_struct *kdamond;
-- 
2.39.2

