From: SJ Park <sj@kernel.org>
Date: Mon, 27 Jul 2026 22:54:19 -0700
Subject: [PATCH] mm/damon/tests/core-kunit: fix overflow test (patch 7 fixup)

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/tests/core-kunit.h | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/mm/damon/tests/core-kunit.h b/mm/damon/tests/core-kunit.h
index 94ce36bb9d861..3ef4bd1281f55 100644
--- a/mm/damon/tests/core-kunit.h
+++ b/mm/damon/tests/core-kunit.h
@@ -629,10 +629,15 @@ static void damon_test_nr_samples_per_aggr(struct kunit *test)
 	/* Zero aggregation interval doesn't cause division by zero */
 	KUNIT_EXPECT_EQ(test, damon_nr_samples_per_aggr(&attrs), 1);
 
-	/* Too large aggregation interval doesn't cause overflow */
-	attrs.aggr_interval = (unsigned long)UINT_MAX + 1;
-	KUNIT_EXPECT_GE(test, damon_nr_samples_per_aggr(&attrs), 0);
-	KUNIT_EXPECT_LE(test, damon_nr_samples_per_aggr(&attrs), UINT_MAX);
+	/*
+	 * Too large aggregation interval on 64 bit system doesn't cause
+	 * overflow
+	 */
+	if (ULONG_MAX > UINT_MAX) {
+		attrs.aggr_interval = (unsigned long)UINT_MAX + 1;
+		KUNIT_EXPECT_EQ(test, damon_nr_samples_per_aggr(&attrs),
+				UINT_MAX);
+	}
 }
 
 static void damon_test_update_monitoring_result(struct kunit *test)
-- 
2.47.3

