From: SeongJae Park <sj@kernel.org>
Date: Wed, 19 Nov 2025 21:00:18 -0800
Subject: [PATCH] mm/damon/core: set score histogram without filters-excluding
 regions

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index f3fdcb45b6c93..605fedf10756b 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2627,6 +2627,8 @@ static void damos_adjust_quota(struct damon_ctx *c, struct damos *s)
 		damon_for_each_region(r, t) {
 			if (!__damos_valid_target(r, s))
 				continue;
+			if (damos_core_filter_out(c, t, r, s))
+				continue;
 			score = c->ops.get_scheme_score(c, t, r, s);
 			c->regions_score_histogram[score] +=
 				damon_sz_region(r);
-- 
2.47.3

