From: SeongJae Park <sj@kernel.org>
Date: Wed, 20 May 2026 06:59:27 -0700
Subject: [PATCH] mm/damon/core: (fixup 8) fix wrong start>=end condition

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 24f5c1f7de5c7..6b8af7f956b76 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3577,8 +3577,8 @@ static void damon_verify_ctx(struct damon_ctx *c)
 		unsigned int nr_regions = 0;
 
 		damon_for_each_region(r, t) {
-			WARN_ONCE(r->ar.start <= r->ar.end,
-					"region start (%lu) <= end (%lu)\n",
+			WARN_ONCE(r->ar.start >= r->ar.end,
+					"region start (%lu) >= end (%lu)\n",
 					r->ar.start, r->ar.end);
 			WARN_ONCE(prev_r && prev_r->ar.end > r->ar.start,
 					"region overlap (%lu > %lu)\n",
-- 
2.47.3

