From: SeongJae Park <sj@kernel.org>
Date: Fri, 5 Dec 2025 17:14:26 -0800
Subject: [PATCH] mm/damon/core: set damon_access_report->vaddr from page fault
 report

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 8e0c7c4e96bd5..2b6b47cdbf473 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1881,6 +1881,7 @@ void damon_report_access(struct damon_access_report *report)
 void damon_report_page_fault(struct vm_fault *vmf, bool huge_pmd)
 {
 	struct damon_access_report access_report = {
+		.vaddr = vmf->address,
 		.size = 1,	/* todo: set appripriately */
 		.cpu = smp_processor_id(),
 		.tid = task_pid_vnr(current),
@@ -1891,7 +1892,6 @@ void damon_report_page_fault(struct vm_fault *vmf, bool huge_pmd)
 		access_report.paddr = PFN_PHYS(pmd_pfn(vmf->orig_pmd));
 	else
 		access_report.paddr = PFN_PHYS(pte_pfn(vmf->orig_pte));
-	/* todo: report vmf->address as virtual address */
 
 	damon_report_access(&access_report);
 }
-- 
2.47.3

