From: SeongJae Park <sj@kernel.org>
Date: Tue, 10 Feb 2026 22:09:31 -0800
Subject: [PATCH] mm/damon/core: verify damon_del_region()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index aa2fc46815ffd..050aac315d610 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -166,8 +166,24 @@ void damon_add_region(struct damon_region *r, struct damon_target *t)
 	t->nr_regions++;
 }
 
+#ifdef CONFIG_DAMON_HARDENED
+static void damon_verify_del_region(struct damon_target *t)
+{
+	if (t->nr_regions > 0)
+		return;
+	pr_err("nr_regions <=0 but damon_call_region called\n");
+	BUG();
+}
+#else
+static void damon_verify_del_region(struct damon_target *t)
+{
+}
+#endif
+
 static void damon_del_region(struct damon_region *r, struct damon_target *t)
 {
+	damon_verify_del_region(t);
+
 	list_del(&r->list);
 	t->nr_regions--;
 }
-- 
2.47.3

