From: SeongJae Park <sj@kernel.org>
Date: Tue, 9 Jun 2026 22:27:10 -0700
Subject: [PATCH] mm/damon/core: change __damon_stop() to return nothing

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index c0674481a29ba..671cac9c4ecf8 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2066,10 +2066,8 @@ int damon_start(struct damon_ctx **ctxs, int nr_ctxs, bool exclusive)
 /*
  * __damon_stop() - Stops monitoring of a given context.
  * @ctx:	monitoring context
- *
- * Return: 0 on success, negative error code otherwise.
  */
-static int __damon_stop(struct damon_ctx *ctx)
+static void __damon_stop(struct damon_ctx *ctx)
 {
 	struct task_struct *tsk;
 
@@ -2079,11 +2077,9 @@ static int __damon_stop(struct damon_ctx *ctx)
 		get_task_struct(tsk);
 		mutex_unlock(&ctx->kdamond_lock);
 		kthread_stop_put(tsk);
-		return 0;
+		return;
 	}
 	mutex_unlock(&ctx->kdamond_lock);
-
-	return -EPERM;
 }
 
 /**
-- 
2.47.3

