From: SeongJae Park <sj@kernel.org>
Date: Sat, 16 Aug 2025 14:42:43 -0700
Subject: [PATCH] mm/damon/core: deallocate ops_attrs->tids from destroy_ctx()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 18074b158393..15a4cc957ac2 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -579,6 +579,9 @@ void damon_destroy_ctx(struct damon_ctx *ctx)
 	damon_for_each_scheme_safe(s, next_s, ctx)
 		damon_destroy_scheme(s);
 
+	if (ctx->ops_attrs.nr_tids)
+		kfree(ctx->ops_attrs.tids);
+
 	kfree(ctx);
 }
 
-- 
2.47.3

