From: SeongJae Park <sj@kernel.org>
Date: Sat, 28 Jun 2025 13:59:20 -0700
Subject: [PATCH] mm/damon/lru_sort: use damon_initialized()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/lru_sort.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/mm/damon/lru_sort.c b/mm/damon/lru_sort.c
index 1e9e19f15b33..0489ec8001c2 100644
--- a/mm/damon/lru_sort.c
+++ b/mm/damon/lru_sort.c
@@ -369,7 +369,7 @@ static int damon_lru_sort_enabled_store(const char *val,
 		return 0;
 
 	/* Called before init function.  The function will handle this. */
-	if (!ctx)
+	if (!damon_initialized())
 		goto set_param_out;
 
 	err = damon_lru_sort_turn(enable);
@@ -428,6 +428,8 @@ static int __init damon_lru_sort_init(void)
 
 	if (err)
 		return err;
+	if (!damon_initialized())
+		return -ENOMEM;
 
 	ctx->callback.after_wmarks_check = damon_lru_sort_after_wmarks_check;
 	ctx->callback.after_aggregation = damon_lru_sort_after_aggregation;
-- 
2.39.5

