From: SeongJae Park <sj@kernel.org>
Date: Sat, 3 Jan 2026 10:04:42 -0800
Subject: [PATCH] mm/damon: add CONFIG_DAMON_HARDENED

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/Kconfig | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/mm/damon/Kconfig b/mm/damon/Kconfig
index 8c868f7035fce..40d212fc4fac3 100644
--- a/mm/damon/Kconfig
+++ b/mm/damon/Kconfig
@@ -110,4 +110,15 @@ config DAMON_STAT_ENABLED_DEFAULT
 	  Whether to enable DAMON_STAT by default.  Users can disable it in
 	  boot or runtime using its 'enabled' parameter.
 
+config DAMON_HARDENED
+	bool "Check integrity of DAMON code"
+	depends on DAMON
+	help
+	  This enables additional DAMON debugging-purpose integrity checks in
+	  DAMON code.  This can impose additional overhead but can be useful at
+	  finding bugs.  Hence it is enouraged to enable on DAMON development
+	  setups.
+
+	  If unsure, say N.
+
 endmenu
-- 
2.47.3

