From: SeongJae Park <sj@kernel.org>
Date: Wed, 11 Mar 2026 17:10:08 -0700
Subject: [PATCH] mm/damon: remove biggest_system_ram_default() declaration

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index aafd7471e5330..37aae47088402 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -1148,11 +1148,6 @@ int damon_set_region_system_rams_default(struct damon_target *t,
 				unsigned long addr_unit,
 				unsigned long min_region_sz);
 
-int damon_set_region_biggest_system_ram_default(struct damon_target *t,
-				unsigned long *start, unsigned long *end,
-				unsigned long addr_unit,
-				unsigned long min_region_sz);
-
 #ifdef CONFIG_ACMA
 
 unsigned long damon_alloced_bytes(void);
-- 
2.47.3

