From: SeongJae Park <sj@kernel.org>
Date: Mon, 10 Feb 2025 16:47:50 -0800
Subject: [PATCH] mm/damon/core: add debugging log for intervals auto-tuning

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 2a509731a2a8..72e8f70265f8 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1626,6 +1626,9 @@ static unsigned long damon_get_intervals_score(struct damon_ctx *c)
 	}
 	target_access_events = max_access_events * goal_bp / 10000;
 	target_access_events = target_access_events ? : 1;
+
+	pr_info("access_events %lu/%lu\n", access_events, target_access_events);
+
 	return access_events * 10000 / target_access_events;
 }
 
@@ -1645,6 +1648,9 @@ static unsigned long damon_get_intervals_adaptation_bp(struct damon_ctx *c)
 	 */
 	if (adaptation_bp <= 10000)
 		adaptation_bp = 5000 + adaptation_bp / 2;
+
+	pr_info("score_bp %lu, adaptation bp %lu\n", score_bp, adaptation_bp);
+
 	return adaptation_bp;
 }
 
@@ -1667,6 +1673,9 @@ static void kdamond_tune_intervals(struct damon_ctx *c)
 	new_attrs.aggr_interval = new_attrs.sample_interval *
 		c->attrs.aggr_samples;
 	trace_damon_monitor_intervals_tune(new_attrs.sample_interval);
+
+	pr_info("tune intervals to %lu %lu\n\n",
+			new_attrs.sample_interval, new_attrs.aggr_interval);
 	damon_set_attrs(c, &new_attrs);
 }
 
-- 
2.47.3

