From: SeongJae Park <sj@kernel.org>
Date: Sat, 26 Jul 2025 10:48:54 -0700
Subject: [PATCH] mm/damon/core: remove debug code for nr_accesses_bp
 corruption

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 007ab4512bfd..1cf6fc7ab1a4 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1555,15 +1555,6 @@ static void kdamond_reset_aggregated(struct damon_ctx *c)
 			damon_warn_fix_nr_accesses_corruption(r);
 			r->last_nr_accesses = r->nr_accesses;
 			r->nr_accesses = 0;
-			if (r->nr_accesses_bp != r->last_nr_accesses * 10000) {
-				pr_info("reset time invalid region found!\n");
-				pr_info("nr_accesses_bp %u last_nr_acceses %u\n",
-						r->nr_accesses_bp, r->last_nr_accesses);
-				pr_info("passed_sis %lu next_aggregation_sis %lu\n",
-						c->passed_sample_intervals,
-						c->next_aggregation_sis);
-				BUG();
-			}
 		}
 		ti++;
 	}
-- 
2.39.5

