From: SeongJae Park <sj@kernel.org>
Date: Sun, 1 Mar 2026 15:19:19 -0800
Subject: [PATCH] mm/damon/core: set goal_tuner in damon_new_scheme()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 8f4d58aa469a9..9afad1d7bc1d3 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -475,9 +475,9 @@ struct damos *damon_new_scheme(struct damos_access_pattern *pattern,
 	INIT_LIST_HEAD(&scheme->list);
 
 	scheme->quota = *(damos_quota_init(quota));
-	/* quota.goals and .goal_tuner should be separately set by caller */
+	/* quota.goals should be separately set by caller */
 	INIT_LIST_HEAD(&scheme->quota.goals);
-	scheme->quota.goal_tuner = DAMOS_QUOTA_GOAL_TUNER_CONSIST;
+	scheme->quota.goal_tuner = quota->goal_tuner;
 
 	scheme->wmarks = *wmarks;
 	scheme->wmarks.activated = true;
-- 
2.47.3

