From: SeongJae Park <sj@kernel.org>
Date: Mon, 11 May 2026 20:57:29 -0700
Subject: [PATCH] mm/damon/paddr: put folio for filtering

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index f45b9dc592973..abe481d34dfb9 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -222,16 +222,20 @@ static bool damon_pa_filter_pass(phys_addr_t pa, struct damon_probe *p)
 {
 	struct damon_filter *f;
 	struct folio *folio;
-	bool default_pass = true;
+	bool pass = true;
 
 	folio = damon_get_folio(PHYS_PFN(pa));
 	damon_for_each_filter(f, p) {
-		if (damon_pa_filter_match(f, folio))
-			return f->allow;
-		else
-			default_pass = !f->allow;
+		if (damon_pa_filter_match(f, folio)) {
+			pass = f->allow;
+			break;
+		} else {
+			pass = !f->allow;
+		}
 	}
-	return default_pass;
+	if (folio)
+		folio_put(folio);
+	return pass;
 }
 
 static void damon_pa_apply_probes(struct damon_ctx *ctx)
-- 
2.47.3

