From: SeongJae Park <sj@kernel.org>
Date: Wed, 14 May 2025 07:51:01 -0700
Subject: [PATCH] mm/damon/stat: make enabled write handler name consistent

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/stat.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/mm/damon/stat.c b/mm/damon/stat.c
index 82fad5f9e2f7..101ae3026989 100644
--- a/mm/damon/stat.c
+++ b/mm/damon/stat.c
@@ -16,11 +16,11 @@
 #endif
 #define MODULE_PARAM_PREFIX "damon_stat."
 
-static int damon_stat_enable_store(
+static int damon_stat_enabled_store(
 		const char *val, const struct kernel_param *kp);
 
 static const struct kernel_param_ops enabled_param_ops = {
-	.set = damon_stat_enable_store,
+	.set = damon_stat_enabled_store,
 	.get = param_get_bool,
 };
 
@@ -198,7 +198,7 @@ static void damon_stat_stop(void)
 
 static bool damon_stat_init_called = false;
 
-static int damon_stat_enable_store(
+static int damon_stat_enabled_store(
 		const char *val, const struct kernel_param *kp)
 {
 	bool is_enabled = enabled;
-- 
2.39.5

