From: SJ Park <sj@kernel.org>
Date: Tue, 25 Aug 2026 07:21:45 -0700
Subject: [PATCH] mm/damon/core: commit range_{min,max} for probe hits wsum
 filter

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 6099a54341a89..354dcf2204fa0 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1412,6 +1412,10 @@ static void damos_commit_filter_arg(
 	case DAMOS_FILTER_TYPE_HUGEPAGE_SIZE:
 		dst->sz_range = src->sz_range;
 		break;
+	case DAMOS_FILTER_TYPE_PROBE_HITS_WSUM:
+		dst->range_min = src->range_min;
+		dst->range_max = src->range_max;
+		break;
 	default:
 		break;
 	}
-- 
2.47.3

