From: SJ Park <sj@kernel.org>
Date: Fri, 10 Jul 2026 20:06:32 -0700
Subject: [PATCH] mm/damon/core: fixup infinite merge fix

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 3e4a616d9765f..7bf1ae6c6a94a 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3727,9 +3727,13 @@ static void kdamond_merge_regions(struct damon_ctx *c, unsigned int threshold,
 			nr_regions += damon_nr_regions(t);
 		}
 		count_age = false;
-		threshold = max(1, threshold * 2);
-	} while (nr_regions > c->attrs.max_nr_regions &&
-			threshold / 2 < max_thres);
+		if (threshold >= max_thres)
+			break;
+		if (threshold < max_thres / 2)
+			threshold = max(1, threshold * 2);
+		else
+			threshold = max_thres;
+	} while (nr_regions > c->attrs.max_nr_regions);
 }
 
 #ifdef CONFIG_DAMON_DEBUG_SANITY
-- 
2.47.3

