From: SeongJae Park <sj@kernel.org>
Date: Sat, 9 May 2026 10:18:18 -0700
Subject: [PATCH] mm/damon/sysfs: rename t to filter in damon_sysfs_set_probes

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 01c2010c91d07..2b68e1a7de451 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1920,14 +1920,14 @@ static int damon_sysfs_set_probes(struct damon_ctx *ctx,
 		for (j = 0; j < sys_filters->nr; j++) {
 			struct damon_sysfs_filter *sys_filter =
 				sys_filters->filters_arr[j];
-			struct damon_filter *t;
+			struct damon_filter *filter;
 
-			t = damon_new_filter(sys_filter->type,
+			filter = damon_new_filter(sys_filter->type,
 					sys_filter->matching,
 					sys_filter->allow);
-			if (!t)
+			if (!filter)
 				return -ENOMEM;
-			damon_add_filter(c, t);
+			damon_add_filter(c, filter);
 		}
 	}
 	return 0;
-- 
2.47.3

