From: SeongJae Park <sj@kernel.org>
Date: Wed, 8 Oct 2025 20:26:27 -0700
Subject: [PATCH] mm/damon/core: handle remove_on_commit

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 00d39b7af413..d710702831fe 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1233,7 +1233,7 @@ static int damon_commit_targets(
 
 	damon_for_each_target_safe(dst_target, next, dst) {
 		src_target = damon_nth_target(i++, src);
-		if (src_target) {
+		if (src_target && !src_target->remove_on_commit) {
 			err = damon_commit_target(
 					dst_target, damon_target_has_pid(dst),
 					src_target, damon_target_has_pid(src),
-- 
2.39.5

