From: SeongJae Park <sj@kernel.org>
Date: Mon, 27 Jan 2025 17:49:13 -0800
Subject: [PATCH] mm/damon/core: add debugging log for intervals auto-tuning

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 61a7dabc639d..e29a0ee7e937 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1314,6 +1314,8 @@ static void kdamond_tune_intervals(struct damon_ctx *c)
 	new_attrs.sample_interval =
 		new_attrs.aggr_interval * sample_to_aggr_bp / 10000;
 	/* damon_set_attrs() returns error only for wrong parameter */
+	pr_info("tune intervals to %lu %lu\n",
+			new_attrs.sample_interval, new_attrs.aggr_interval);
 	damon_set_attrs(c, &new_attrs);
 
 	c->attrs.access_samples = 0;
-- 
2.39.5

