From: SeongJae Park <sj@kernel.org>
Date: Sat, 23 May 2026 13:48:46 -0700
Subject: [PATCH] mm/damon: mark targets, schemes and rnd_state of damon_ctx as
 private

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 88a824f7eb4c2..c0375035a3a7b 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -991,8 +991,6 @@ struct damon_sample_control {
  * @addr_unit:	Scale factor for core to ops address conversion.
  * @min_region_sz:	Minimum region size.
  * @pause:	Pause kdamond main loop.
- * @adaptive_targets:	Head of monitoring targets (&damon_target) list.
- * @schemes:		Head of schemes (&damos) list.
  */
 struct damon_ctx {
 	struct damon_attrs attrs;
@@ -1048,7 +1046,10 @@ struct damon_ctx {
 	unsigned long min_region_sz;
 	bool pause;
 
+/* private: */
+	/* Head of monitoring targets (&damon_target) list. */
 	struct list_head adaptive_targets;
+	/* Head of schemes (&damos) list. */
 	struct list_head schemes;
 
 	/* Per-ctx PRNG state for damon_rand(); kdamond is the sole consumer. */
-- 
2.47.3

