From: SeongJae Park <sj@kernel.org>
Date: Sat, 18 Apr 2026 12:30:23 -0700
Subject: [PATCH] mm/damon/paddr: implement check_data_attrs

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h |  2 ++
 mm/damon/paddr.c      | 25 +++++++++++++++++++++++++
 2 files changed, 27 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index f2d772d0f6631..79405c1874086 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -1055,6 +1055,8 @@ static inline unsigned long damon_sz_region(struct damon_region *r)
 	return r->ar.end - r->ar.start;
 }
 
+#define damon_for_each_data_attr(a, ctx) \
+	list_for_each_entry(a, &ctx->attrs.data_attrs, list)
 
 #define damon_for_each_region(r, t) \
 	list_for_each_entry(r, &t->regions_list, list)
diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index ddef9caa852b9..45aabd5fc064f 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -184,6 +184,30 @@ static unsigned int damon_pa_check_accesses(struct damon_ctx *ctx)
 	return max_nr_accesses;
 }
 
+static bool damon_pa_is_attr(struct damon_region *r, struct damon_data_attr *a)
+{
+	return false;
+}
+
+static void damon_pa_check_data_attrs(struct damon_ctx *ctx)
+{
+	struct damon_target *t;
+	struct damon_region *r;
+	struct damon_data_attr *a;
+
+	damon_for_each_target(t, ctx) {
+		damon_for_each_region(r, t) {
+			int i = 0;
+
+			damon_for_each_data_attr(a, ctx) {
+				if (damon_pa_is_attr(r, a))
+					r->attr_counts[i]++;
+				i++;
+			}
+		}
+	}
+}
+
 /*
  * damos_pa_filter_out - Return true if the page should be filtered out.
  */
@@ -528,6 +552,7 @@ static int __init damon_pa_initcall(void)
 		.update = NULL,
 		.prepare_access_checks = damon_pa_prepare_access_checks,
 		.check_accesses = damon_pa_check_accesses,
+		.check_data_attrs = damon_pa_check_data_attrs,
 		.target_valid = NULL,
 		.apply_scheme = damon_pa_apply_scheme,
 		.get_scheme_score = damon_pa_scheme_score,
-- 
2.47.3

