From: SeongJae Park <sj@kernel.org>
Date: Sat, 26 Jul 2025 10:48:03 -0700
Subject: [PATCH] mm/damon/core: add TODO for kdamond_check_reported_accesses()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 7fbb00884198..007ab4512bfd 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2779,6 +2779,10 @@ static int kdamond_fn(void *data)
 		kdamond_usleep(sample_interval);
 		ctx->passed_sample_intervals++;
 
+		/*
+		 * TODO: set max_nr_accesses from
+		 * kdamond_check_reported_accesses()
+		 */
 		kdamond_check_reported_accesses(ctx);
 		if (ctx->ops.check_accesses)
 			max_nr_accesses = ctx->ops.check_accesses(ctx);
-- 
2.39.5

