From: SeongJae Park <sj@kernel.org>
Date: Sat, 23 May 2026 12:57:27 -0700
Subject: [PATCH] mm/damon: mark only pid and obsolete of damon_target as
 public

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 1dd37fccd1d32..4fa4e3bf996b6 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -92,9 +92,6 @@ struct damon_region {
 /**
  * struct damon_target - Represents a monitoring target.
  * @pid:		The PID of the virtual address space to monitor.
- * @nr_regions:		Number of monitoring target regions of this target.
- * @regions_list:	Head of the monitoring target regions of this target.
- * @list:		List head for siblings.
  * @obsolete:		Whether the commit destination target is obsolete.
  *
  * Each monitoring context could have multiple targets.  For example, a context
@@ -108,9 +105,14 @@ struct damon_region {
  */
 struct damon_target {
 	struct pid *pid;
+/* private: */
+	/* Number of monitoring target regions of this target. */
 	unsigned int nr_regions;
+	/* Head of the monitoring target regions of this target. */
 	struct list_head regions_list;
+	/* List head for siblings. */
 	struct list_head list;
+/* public: */
 	bool obsolete;
 };
 
-- 
2.47.3

