From: SeongJae Park <sj@kernel.org>
Date: Sat, 25 Apr 2026 11:34:52 -0700
Subject: [PATCH] mm/damon/sysfs: disallow more than DAMON_MAX_PROBES probes
 installation

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 6914396173160..8007961295d3d 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1159,7 +1159,7 @@ static ssize_t nr_probes_store(struct kobject *kobj,
 
 	if (err)
 		return err;
-	if (nr < 0)
+	if (nr < 0 || nr > DAMON_MAX_PROBES)
 		return -EINVAL;
 
 	probes = container_of(kobj, struct damon_sysfs_probes, kobj);
-- 
2.47.3

