From: SeongJae Park <sj@kernel.org>
Date: Sun, 15 Dec 2024 12:44:37 -0800
Subject: [PATCH] mm/damon: add per-region DAMOS filtered-out size statistics

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 1 +
 mm/damon/core.c       | 1 +
 2 files changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index d9c32669de7e..1089a4a655f2 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -73,6 +73,7 @@ struct damon_region {
 	unsigned int age;
 /* private: Internal value for age calculation. */
 	unsigned int last_nr_accesses;
+	unsigned long damos_filtered_out;
 };
 
 /**
diff --git a/mm/damon/core.c b/mm/damon/core.c
index de923b3a1084..174f25962d35 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -140,6 +140,7 @@ struct damon_region *damon_new_region(unsigned long start, unsigned long end)
 
 	region->age = 0;
 	region->last_nr_accesses = 0;
+	region->damos_filtered_out = 0;
 
 	return region;
 }
-- 
2.39.5

