From: SeongJae Park <sj@kernel.org>
Date: Tue, 23 Jun 2026 07:47:26 -0700
Subject: [PATCH] mm/damon: mark all damon_probe fields as private

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index e8805c4da14bb..7434930eda06d 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -852,14 +852,14 @@ struct damon_filter {
 
 /**
  * struct damon_probe - Data region attribute probe.
- *
- * @preps:	Preparation actions to apply to each probing memory.
- * @filters:	Filters for assessing if a given region is for this probe.
- * @list:	Siblings list.
  */
 struct damon_probe {
+/* private: */
+	/* Preparation actions to apply to each probing memory. */
 	struct list_head preps;
+	/* Filters for assessing if a given region is for this probe. */
 	struct list_head filters;
+	/* Siblings list. */
 	struct list_head list;
 };
 
-- 
2.47.3

