From: SeongJae Park <sj@kernel.org>
Date: Sat, 28 Jun 2025 13:58:31 -0700
Subject: [PATCH] mm/damon/reclaim: use damon_initialized()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/reclaim.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/mm/damon/reclaim.c b/mm/damon/reclaim.c
index a675150965e0..b2131f7bf0f5 100644
--- a/mm/damon/reclaim.c
+++ b/mm/damon/reclaim.c
@@ -275,7 +275,7 @@ static int damon_reclaim_enabled_store(const char *val,
 		return 0;
 
 	/* Called before init function.  The function will handle this. */
-	if (!ctx)
+	if (!damon_initialized())
 		goto set_param_out;
 
 	err = damon_reclaim_turn(enable);
@@ -330,6 +330,8 @@ static int __init damon_reclaim_init(void)
 
 	if (err)
 		return err;
+	if (!damon_initialized())
+		return -ENOMEM;
 
 	ctx->callback.after_wmarks_check = damon_reclaim_after_wmarks_check;
 	ctx->callback.after_aggregation = damon_reclaim_after_aggregation;
-- 
2.39.5

