From: SeongJae Park <sj@kernel.org>
Date: Sat, 9 May 2026 09:48:51 -0700
Subject: [PATCH] mm/damon/paddr: put folio on damon_pa_filter_pass()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index d4c4d2ed11d6d..1e9cb7316d94f 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -196,8 +196,10 @@ static bool damon_pa_filter_pass(phys_addr_t pa, struct damon_probe *p)
 		if (f->type == DAMON_FILTER_TYPE_ANON) {
 			struct folio *folio = damon_get_folio(PHYS_PFN(pa));
 
-			if (folio)
+			if (folio) {
 				matched = folio_test_anon(folio);
+				folio_put(folio);
+			}
 		}
 		if (matched)
 			return f->allow;
-- 
2.47.3

