From: SeongJae Park <sj@kernel.org>
Date: Sun, 30 Nov 2025 13:06:46 -0800
Subject: [PATCH] mm/damon/sysfs: fix wrong sample filters directory name

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 6dbe119b6cdc..8babcae5cfb2 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1237,7 +1237,7 @@ static int damon_sysfs_sample_add_dirs(
 		return -ENOMEM;
 	err = kobject_init_and_add(&filters->kobj,
 			&damon_sysfs_sample_filters_ktype, &sample->kobj,
-			"sample_filters");
+			"filters");
 	if (err)
 		goto put_filters_out;
 	sample->filters = filters;
-- 
2.47.3

