From: SeongJae Park <sj@kernel.org>
Date: Sat, 9 May 2026 10:01:25 -0700
Subject: [PATCH] mm/damon/paddr: support DAMON_FILTER_TYPE_MEMCG

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index 69da18c268e61..c2544e9920d31 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -190,6 +190,7 @@ static bool damon_pa_filter_pass(phys_addr_t pa, struct damon_probe *p)
 	struct damon_filter *f;
 	struct folio *folio;
 	bool pass = true;
+	struct mem_cgroup *memcg;
 
 	folio = damon_get_folio(PHYS_PFN(pa));
 	damon_for_each_filter(f, p) {
@@ -198,6 +199,17 @@ static bool damon_pa_filter_pass(phys_addr_t pa, struct damon_probe *p)
 		if (f->type == DAMON_FILTER_TYPE_ANON) {
 			if (folio)
 				matched = folio_test_anon(folio);
+		} else if (f->type == DAMON_FILTER_TYPE_MEMCG) {
+			if (folio) {
+				rcu_read_lock();
+				memcg = folio_memcg_check(folio);
+				if (!memcg)
+					matched = false;
+				else
+					matched = f->memcg_id ==
+						mem_cgroup_id(memcg);
+				rcu_read_unlock();
+			}
 		}
 		if (matched) {
 			pass = f->allow;
-- 
2.47.3

