From: SeongJae Park <sj@kernel.org>
Date: Sat, 23 May 2026 13:38:46 -0700
Subject: [PATCH] mm/damon: mark damos_filter->list as private

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 0153fb7a2675e..1f5ec1b5157d4 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -466,7 +466,6 @@ enum damos_filter_type {
  *		&damon_ctx->adaptive_targets if @type is
  *		DAMOS_FILTER_TYPE_TARGET.
  * @sz_range:	Size range if @type is DAMOS_FILTER_TYPE_HUGEPAGE_SIZE.
- * @list:	List head for siblings.
  *
  * Before applying the &damos->action to a memory region, DAMOS checks if each
  * byte of the region matches to this given condition and avoid applying the
@@ -484,6 +483,8 @@ struct damos_filter {
 		int target_idx;
 		struct damon_size_range sz_range;
 	};
+/* private: */
+	/* List head for siblings. */
 	struct list_head list;
 };
 
-- 
2.47.3

