From: SJ Park <sj@kernel.org>
Date: Sun, 2 Aug 2026 11:05:26 -0700
Subject: [PATCH] mm/damon/core: fixup node_memcg_used_bp (patch 8 fixup)

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index daab1e7d66ca0..be8e2a0c09fc9 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3175,16 +3175,16 @@ static unsigned long damos_get_node_memcg_used_bp(
 	mem_cgroup_put(memcg);
 
 	si_meminfo_node(&i, goal->nid);
-	if (!i.totalram)
-		return 10000;
-	if (goal->metric == DAMOS_QUOTA_NODE_MEMCG_USED_BP) {
-		numerator = used_pages;
-	} else {
-		/* DAMOS_QUOTA_NODE_MEMCG_FREE_BP */
-		if (i.totalram < used_pages)
+	if (!i.totalram || i.totalram < used_pages) {
+		if (goal->metric == DAMOS_QUOTA_NODE_MEMCG_USED_BP)
+			return 10000;
+		else	/* DAMOS_QUOTA_NODE_MEMCG_FREE_BP */
 			return 0;
-		numerator = i.totalram - used_pages;
 	}
+	if (goal->metric == DAMOS_QUOTA_NODE_MEMCG_USED_BP)
+		numerator = used_pages;
+	else	/* DAMOS_QUOTA_NODE_MEMCG_FREE_BP */
+		numerator = i.totalram - used_pages;
 	return mult_frac(numerator, 10000, i.totalram);
 }
 
-- 
2.47.3

