From: SeongJae Park <sj@kernel.org>
Date: Mon, 12 May 2025 16:14:56 -0700
Subject: [PATCH] mm/damon/stat: set last_refreshed_jiffies correctly

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/stat.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/stat.c b/mm/damon/stat.c
index 75857dec03cb..379edf18a142 100644
--- a/mm/damon/stat.c
+++ b/mm/damon/stat.c
@@ -117,6 +117,7 @@ static int damon_stat_after_aggregation(struct damon_ctx *c)
 	if (time_before_eq(jiffies, last_refresh_jiffies +
 				msecs_to_jiffies(5 * MSEC_PER_SEC)))
 		return 0;
+	last_refresh_jiffies = jiffies;
 
 	damon_for_each_target(t, c) {
 		damon_for_each_region(r, t)
-- 
2.39.5

