From: SJ Park <sj@kernel.org>
Date: Sun, 26 Jul 2026 11:59:41 -0700
Subject: [PATCH] mm/damon/sysfs: constify fields (patch 11 fixup)

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 92b28e4baaba9..d0653e2235de8 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -771,8 +771,8 @@ static struct damon_sysfs_prep *damon_sysfs_prep_alloc(void)
 }
 
 struct damon_sysfs_prep_action_name {
-	enum damon_prep_action action;
-	char *name;
+	const enum damon_prep_action action;
+	const char *name;
 };
 
 static const struct damon_sysfs_prep_action_name
-- 
2.47.3

