From: SJ Park <sj@kernel.org>
Date: Mon, 6 Jul 2026 18:07:47 -0700
Subject: [PATCH] mm/damon/core: avoid vaddr with probe weights

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index c9a688451743f..22a93c5efc657 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1493,6 +1493,9 @@ static bool damon_valid_probe_params(struct damon_ctx *ctx)
 	if (!damon_has_probe_weights(ctx))
 		return true;
 
+	if (ctx->ops.id != DAMON_OPS_PADDR)
+		return false;
+
 	sample_interval = ctx->attrs.sample_interval ? : 1;
 	if (ctx->attrs.aggr_interval / sample_interval > U8_MAX)
 		return false;
-- 
2.47.3

