From: SJ Park <sj@kernel.org>
Date: Sun, 14 Jun 2026 11:22:40 -0700
Subject: [PATCH] mm/damon: mark targets, schemes and rnd_state of damon_ctx as
 private

Signed-off-by: SJ Park <sj@kernel.org>
---
 include/linux/damon.h | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index ecce13d6a30db..a0e7e83a3b55c 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -851,9 +851,6 @@ struct damon_attrs {
  * @addr_unit:	Scale factor for core to ops address conversion.
  * @min_region_sz:	Minimum region size.
  * @pause:	Pause kdamond main loop.
- * @adaptive_targets:	Head of monitoring targets (&damon_target) list.
- * @schemes:		Head of schemes (&damos) list.
- * @rnd_state:	Per-ctx PRNG state for damon_rand().
  */
 struct damon_ctx {
 	struct damon_attrs attrs;
@@ -908,9 +905,13 @@ struct damon_ctx {
 	unsigned long min_region_sz;
 	bool pause;
 
+/* private: */
+	/* Head of monitoring targets (&damon_target) list. */
 	struct list_head adaptive_targets;
+	/* Head of schemes (&damos) list. */
 	struct list_head schemes;
 
+	/* @rnd_state:	Per-ctx PRNG state for damon_rand(). */
 	struct rnd_state rnd_state;
 };
 
-- 
2.47.3

