From: SeongJae Park <sj@kernel.org>
Date: Sat, 16 Aug 2025 15:04:41 -0700
Subject: [PATCH] mm/damon/sysfs: implement tids reading

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 420654d887f5..efb4099acc29 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -907,7 +907,26 @@ static ssize_t cpus_store(struct kobject *kobj, struct kobj_attribute *attr,
 static ssize_t tids_show(struct kobject *kobj, struct kobj_attribute *attr,
 		char *buf)
 {
-	return sysfs_emit(buf, "to be constructed\n");
+	struct damon_sysfs_ops_attrs *ops_attrs = container_of(kobj,
+			struct damon_sysfs_ops_attrs, kobj);
+	char *str;
+	int i, ret;
+
+	if (!ops_attrs->tids)
+		return sysfs_emit(buf, "\n");
+
+	str = kcalloc(2048, sizeof(*str), GFP_KERNEL);
+	if (!str)
+		return -ENOMEM;
+	for (i = 0; i < ops_attrs->tids[0]; i++) {
+		snprintf(&str[strlen(str)], 2048 - strlen(str),
+				"%d", ops_attrs->tids[i + 1]);
+		if (i < ops_attrs->tids[0] - 1)
+			snprintf(&str[strlen(str)], 2048 - strlen(str), ",");
+	}
+	ret = sysfs_emit(buf, "%s\n", str);
+	kfree(str);
+	return ret;
 }
 
 static ssize_t tids_store(struct kobject *kobj, struct kobj_attribute *attr,
-- 
2.39.5

