From: SeongJae Park <sj@kernel.org>
Date: Sun, 19 Apr 2026 12:11:23 -0700
Subject: [PATCH] mm/damon/core: free filters when freeing data_attr

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 3 +++
 mm/damon/core.c       | 4 ++++
 2 files changed, 7 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 54a4d1de220f4..54555f0402b5a 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -1056,6 +1056,9 @@ static inline unsigned long damon_sz_region(struct damon_region *r)
 	return r->ar.end - r->ar.start;
 }
 
+#define damon_for_each_filter_safe(f, next, a) \
+	list_for_each_entry_safe(f, next, &a->filters, list)
+
 #define damon_for_each_data_attr(a, ctx) \
 	list_for_each_entry(a, &ctx->data_attrs, list)
 
diff --git a/mm/damon/core.c b/mm/damon/core.c
index 3d7b3f99a9cfc..35f22d78846ec 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -176,6 +176,10 @@ static void damon_del_data_attr(struct damon_data_attr *a)
 
 static void damon_free_data_attr(struct damon_data_attr *a)
 {
+	struct damon_filter *f, *next;
+
+	damon_for_each_filter(f, next, a)
+		damon_free_filter(f);
 	kfree(a);
 }
 
-- 
2.47.3

