From: SJ Park <sj@kernel.org>
Date: Wed, 24 Jun 2026 20:33:37 -0700
Subject: [PATCH] mm/damon/core: setup damon_ctx->has_probe_weights in
 __damon_commit_ctx()

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index b55a3cc39e001..c943530121e0b 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1672,6 +1672,7 @@ static int __damon_commit_ctx(struct damon_ctx *dst, struct damon_ctx *src)
 	int err;
 	struct damos *scheme;
 	struct damos_quota_goal *goal;
+	struct damon_probe *probe;
 
 	dst->maybe_corrupted = true;
 	if (!is_power_of_2(src->min_region_sz))
@@ -1718,6 +1719,11 @@ static int __damon_commit_ctx(struct damon_ctx *dst, struct damon_ctx *src)
 	dst->addr_unit = src->addr_unit;
 	dst->min_region_sz = src->min_region_sz;
 
+	damon_for_each_probe(probe, dst) {
+		if (probe->weight)
+			dst->has_probe_weights = true;
+	}
+
 	dst->maybe_corrupted = false;
 	return 0;
 }
-- 
2.47.3

