From: SeongJae Park <sj@kernel.org>
Date: Sat, 26 Jul 2025 10:55:34 -0700
Subject: [PATCH] mm/damon/core: fix nr_accesses_bp corruption

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 1cf6fc7ab1a4..72a4d59e34ef 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2689,6 +2689,7 @@ static void kdamond_apply_access_report(struct damon_access_report *report,
 		struct damon_target *t, struct damon_ctx *ctx)
 {
 	struct damon_region *r;
+	int i;
 
 	if (ctx->ops.eligible_report && !ctx->ops.eligible_report(report, t))
 		return;
@@ -2699,7 +2700,9 @@ static void kdamond_apply_access_report(struct damon_access_report *report,
 			continue;
 		if (r->ar.end < report->addr + report->size)
 			continue;
-		r->nr_accesses += report->nr_accesses;
+		/* TODO: optimize out loop */
+		for (i = 0; i < report->nr_accesses; i++)
+			damon_update_region_access_rate(r, true, &ctx->attrs);
 	}
 }
 
-- 
2.39.5

