From: SeongJae Park <sj@kernel.org>
Date: Sun, 3 Aug 2025 11:03:32 -0700
Subject: [PATCH] mm/damon: add damon_access_report->cpuid

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 3d8c93ff3312..ae8995a029c9 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -112,6 +112,7 @@ struct damon_target {
  * @addr:		The start address of the reporting region.
  * @size:		The size of the reporting region.
  * @is_write:		Whether the access is write.
+ * @cpu:		CPU that made the access.
  *
  * @pid could be stale, and hence shouldn't be de-referenced.
  */
@@ -120,6 +121,7 @@ struct damon_access_report {
 	unsigned long addr;
 	unsigned long size;
 	bool is_write;
+	unsigned int cpu;
 /* private: */
 	unsigned long report_jiffies;	/* when this report is made */
 };
-- 
2.39.5

