From: SeongJae Park <sj@kernel.org>
Date: Thu, 9 Apr 2026 17:39:14 -0700
Subject: [PATCH] mm/damon/core: fixup damos-split region mergeback

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 24 ++++++++++++++++--------
 1 file changed, 16 insertions(+), 8 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 98ee776d98cd0..17d6676f125ef 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2161,18 +2161,18 @@ static void damon_do_apply_schemes(struct damon_ctx *c,
 
 static void damos_apply_target(struct damon_ctx *c, struct damon_target *t)
 {
-	struct damon_region *r, *next, *orig_region = NULL;
+	struct damon_region *r, *orig_region = NULL;
 	unsigned long orig_end_addr;
 
-	damon_for_each_region_safe(r, next, t) {
+	damon_for_each_region(r, t) {
 		/*
-		 * damon_do_apply_schemes() split the region if applying the
-		 * action to the whole region can make quota exceeded.  That
-		 * split can result in DAMON snapshot having more than
-		 * max_nr_regions regions.
+		 * damon_do_apply_schemes() splits the region if needed to keep
+		 * the quota.  That could result in DAMON snapshot having more
+		 * than max_nr_regions regions.
 		 *
-		 * Merge back the sliced regions to the original region, as
-		 * soon as the schemes-handling of the slice is completed.
+		 * Merge back the sliced regions to the original region if it
+		 * makes sense, as soon as the schemes-handling of the slice is
+		 * completed.
 		 */
 		if (!orig_region || orig_end_addr <= r->ar.start) {
 			orig_region = r;
@@ -2181,8 +2181,16 @@ static void damos_apply_target(struct damon_ctx *c, struct damon_target *t)
 		damon_do_apply_schemes(c, t, r);
 		if (r == orig_region)
 			continue;
+		/*
+		 * If no scheme was applied to the sliced region, the age of
+		 * the slice ain't be reset.  Don't merge that back, since that
+		 * loses the monitored access pattern information.
+		 */
+		if (r->age)
+			continue;
 		orig_region->ar.end = r->ar.end;
 		damon_destroy_region(r, t);
+		r = orig_region;
 	}
 }
 
-- 
2.47.3

