From: SeongJae Park <sj@kernel.org>
Date: Sat, 29 Nov 2025 12:16:30 -0800
Subject: [PATCH] mm/damon/core: set page table scan primitives enabled by
 default

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index d789782df716..9b9ef7b16677 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -621,6 +621,7 @@ struct damon_ctx *damon_new_ctx(void)
 	ctx->attrs.min_nr_regions = 10;
 	ctx->attrs.max_nr_regions = 1000;
 
+	ctx->access_check_control.primitives_enablement.page_table_scan = true;
 	INIT_LIST_HEAD(&ctx->access_check_control.report_filters);
 
 	ctx->addr_unit = 1;
-- 
2.47.3

