From: SeongJae Park <sj@kernel.org>
Date: Fri, 7 Feb 2025 14:42:16 -0800
Subject: [PATCH] mm/damon/core: wordsmith intervals auto-tuning comment

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 8da795a5d69c..9118f3622946 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -663,7 +663,7 @@ int damon_set_attrs(struct damon_ctx *ctx, struct damon_attrs *attrs)
 	if (attrs->sample_interval > attrs->aggr_interval)
 		return -EINVAL;
 
-	/* Only calls from core-external doesn't set this. */
+	/* calls from core-external doesn't set this. */
 	if (!attrs->aggr_samples)
 		attrs->aggr_samples = attrs->aggr_interval / sample_interval;
 
-- 
2.39.5

