From: SeongJae Park <sj@kernel.org>
Date: Tue, 21 Jan 2025 16:59:06 -0800
Subject: [PATCH] mm/damon/core: respect minimum aggregation interval

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 3cc4a59d5253..e918f6f307a0 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1338,6 +1338,8 @@ static void kdamond_tune_intervals(struct damon_ctx *c)
 	new_attrs.aggr_interval = min(
 			c->attrs.aggr_interval * adaptation_bp / 10000,
 			c->attrs.max_aggr_interval);
+	new_attrs.aggr_interval = max(new_attrs.aggr_interval,
+			c->attrs.min_aggr_interval);
 	sample_to_aggr_bp = c->attrs.sample_interval * 10000 /
 		c->attrs.aggr_interval;
 	new_attrs.sample_interval =
@@ -2310,6 +2312,9 @@ static int kdamond_fn(void *data)
 	complete(&ctx->kdamond_started);
 	kdamond_init_intervals_sis(ctx);
 
+	if (ctx->attrs.tune_interval_aggrs)
+		ctx->attrts.min_aggr_interval = ctx->attrs.aggr_interval;
+
 	if (ctx->ops.init)
 		ctx->ops.init(ctx);
 	ctx->regions_score_histogram = kmalloc_array(DAMOS_MAX_SCORE + 1,
-- 
2.39.5

