From: SJ Park <sj@kernel.org>
Date: Tue, 25 Aug 2026 07:31:57 -0700
Subject: [PATCH] mm/damon/core: support wsum based damos filter

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 4dba4ea25a998..76095c358a74c 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2769,6 +2769,13 @@ static bool damos_filter_match(struct damon_ctx *ctx, struct damon_target *t,
 		damon_split_region_at(t, r, end - r->ar.start);
 		matched = true;
 		break;
+	case DAMOS_FILTER_TYPE_PROBE_HITS_WSUM:
+		unsigned long wsum;
+
+		wsum = damon_probe_hits_wsum(r, false, ctx);
+		matched = filter->range_min <= wsum &&
+			wsum <= filter->range_max;
+		break;
 	default:
 		return false;
 	}
-- 
2.47.3

