From: SeongJae Park <sj@kernel.org>
Date: Mon, 1 Dec 2025 22:19:45 -0800
Subject: [PATCH] mm/damon: wordsmith kernel-doc

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 0d5bb5620f5a..6bd395038f6b 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -769,8 +769,8 @@ struct damon_attrs {
 };
 
 /**
- * struct damon_primitives_enabled - Enablement of DAMON primitives.
- * @page_table:		Page table accessed bit scanning.
+ * struct damon_primitives_enabled - Enablement of access sampling primitives.
+ * @page_table:	Page table accessed bit scanning.
  * @page_fault:	Page faults monitoring.
  */
 struct damon_primitives_enabled {
@@ -779,11 +779,10 @@ struct damon_primitives_enabled {
 };
 
 /**
- * enum damon_sample_filter_type - Type of &struct damon_access_report to
- * filter.
- * @DAMON_FILTER_TYPE_CPUMASK:	Access-generated CPUs.
- * @DAMON_FILTER_TYPE_THREADS:	Access-generated threads.
- * @DAMON_FILTER_TYPE_WRITE:	Whether the access was for writing.
+ * enum damon_sample_filter_type - Type of &struct damon_sample_filter.
+ * @DAMON_FILTER_TYPE_CPUMASK:	Filter by access-generated CPUs.
+ * @DAMON_FILTER_TYPE_THREADS:	Filter by access-generated threads.
+ * @DAMON_FILTER_TYPE_WRITE:	Filter by whether the access was for writing.
  */
 enum damon_sample_filter_type {
 	DAMON_FILTER_TYPE_CPUMASK,
-- 
2.47.3

