From: SeongJae Park <sj@kernel.org>
Date: Sun, 30 Nov 2025 13:32:56 -0800
Subject: [PATCH] mm/damon/vaddr: remove eligible_report()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/vaddr.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/mm/damon/vaddr.c b/mm/damon/vaddr.c
index 51ae2996843b..2750c88e7225 100644
--- a/mm/damon/vaddr.c
+++ b/mm/damon/vaddr.c
@@ -752,13 +752,6 @@ static int damos_va_migrate_pmd_entry(pmd_t *pmd, unsigned long addr,
 	return 0;
 }
 
-static bool damon_va_eligible_report(struct damon_access_report *report,
-		struct damon_target *t,
-		struct damon_operations_attrs *ops_attrs)
-{
-	return report->pid == t->pid;
-}
-
 /*
  * Functions for the target validity check and cleanup
  */
@@ -1019,7 +1012,6 @@ static int __init damon_va_initcall(void)
 		.update = damon_va_update,
 		.prepare_access_checks = damon_va_prepare_access_checks,
 		.check_accesses = damon_va_check_accesses,
-		.eligible_report = damon_va_eligible_report,
 		.target_valid = damon_va_target_valid,
 		.cleanup_target = damon_va_cleanup_target,
 		.cleanup = NULL,
-- 
2.47.3

