From: SeongJae Park <sj@kernel.org>
Date: Tue, 28 Jan 2025 14:03:34 -0800
Subject: [PATCH] mm/damon/core: fixup tuning

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 8e6e4550a16b..99b686026290 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1320,7 +1320,7 @@ static void kdamond_tune_intervals(struct damon_ctx *c)
 	/* damon_set_attrs() returns error only for wrong parameter */
 	pr_info("access_ratio bp %lu, score_bp %lu, adaptation bp %lu\n",
 			access_samples * 10000 / max_access_samples, score_bp,
-			adaptabion_bp);
+			adaptation_bp);
 	pr_info("tune intervals to %lu %lu\n\n",
 			new_attrs.sample_interval, new_attrs.aggr_interval);
 	damon_set_attrs(c, &new_attrs);
-- 
2.39.5

