From: SeongJae Park <sj@kernel.org>
Date: Wed, 4 Feb 2026 22:12:28 -0800
Subject: [PATCH] mm/damon/acma: cover all system rams

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/acma.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/mm/damon/acma.c b/mm/damon/acma.c
index 1090232301ce9..be6f971c89d9f 100644
--- a/mm/damon/acma.c
+++ b/mm/damon/acma.c
@@ -426,9 +426,8 @@ static int damon_acma_apply_parameters(void)
 	}
 	damon_add_scheme(ctx, scale_up_scheme);
 
-	return damon_set_region_biggest_system_ram_default(target,
-					&monitor_region_start,
-					&monitor_region_end);
+	return damon_set_region_system_rams_default(target,
+			&monitor_region_start, &monitor_region_end);
 }
 
 static int damon_acma_turn(bool on)
-- 
2.47.3

