From: SeongJae Park <sj@kernel.org>
Date: Wed, 8 Oct 2025 21:08:04 -0700
Subject: [PATCH] mm/damo/sysfs: add todo items for obsolete file input
 handling

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 4ae11dcf17d3..50681088ecd7 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1602,6 +1602,7 @@ static int damon_sysfs_add_target(struct damon_sysfs_target *sys_target,
 			/* caller will destroy targets */
 			return -EINVAL;
 	}
+	/* todo: t->remove_on_commit = sys_target->obsolete */
 	return damon_sysfs_set_regions(t, sys_target->regions, ctx->min_sz_region);
 }
 
@@ -1732,10 +1733,12 @@ static int damon_sysfs_commit_input(void *data)
 	test_ctx = damon_new_ctx();
 	if (!test_ctx)
 		return -ENOMEM;
+	/* todo: commit kdamond->damon_ctx to test_ctx */
 	err = damon_commit_ctx(test_ctx, param_ctx);
 	if (err)
 		goto out;
 	err = damon_commit_ctx(kdamond->damon_ctx, param_ctx);
+	/* todo: re-populate targets directory without obsolete targets */
 out:
 	damon_destroy_ctx(test_ctx);
 	damon_destroy_ctx(param_ctx);
-- 
2.39.5

