From: SeongJae Park <sj@kernel.org>
Date: Fri, 22 Nov 2024 14:29:54 -0800
Subject: [PATCH] mm/damon/core: cancel damon_call() and damos_walk() after
 cleaning ->kdamond

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 84cad195a9f4..c0d302dde4ca 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2288,8 +2288,6 @@ static int kdamond_fn(void *data)
 
 	if (ctx->callback.before_terminate)
 		ctx->callback.before_terminate(ctx);
-	kdamond_callback(ctx, true);
-	damos_walk_cancel(ctx);
 	if (ctx->ops.cleanup)
 		ctx->ops.cleanup(ctx);
 	kfree(ctx->regions_score_histogram);
@@ -2299,6 +2297,9 @@ static int kdamond_fn(void *data)
 	ctx->kdamond = NULL;
 	mutex_unlock(&ctx->kdamond_lock);
 
+	kdamond_callback(ctx, true);
+	damos_walk_cancel(ctx);
+
 	mutex_lock(&damon_lock);
 	nr_running_ctxs--;
 	if (!nr_running_ctxs && running_exclusive_ctxs)
-- 
2.39.5

