From: SeongJae Park <sj@kernel.org>
Date: Mon, 16 Feb 2026 14:40:04 -0800
Subject: [PATCH] mm/damon/core: add comments for damon_apply_min_nr_regions()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index f2b360142b0c8..af4e70abad392 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1320,6 +1320,19 @@ static unsigned long damon_region_sz_limit(struct damon_ctx *ctx)
 static void damon_split_region_at(struct damon_target *t,
 				  struct damon_region *r, unsigned long sz_r);
 
+/*
+ * damon_apply_min_nr_regions() - Make effect of min_nr_regions parameter.
+ * @ctx:	monitoring context.
+ *
+ * This function implement min_nr_regions (minimum number of damon_region
+ * objects in the given monitoring context) behavior.  It first calculates
+ * maximum size of each region for enforcing the min_nr_regions as total size
+ * of the regions divided by the min_nr_regions.  After that, this function
+ * splits regions to ensure all regions are equal to or smaller than the size
+ * limit.  Finally, this function returns the maximum size limit.
+ *
+ * Returns: maximum size of each region for convincing min_nr_regions.
+ */
 static unsigned long damon_apply_min_nr_regions(struct damon_ctx *ctx)
 {
 	unsigned long max_region_sz = damon_region_sz_limit(ctx);
-- 
2.47.3

