From: SeongJae Park <sj@kernel.org>
Date: Sat, 29 Nov 2025 12:55:16 -0800
Subject: [PATCH] mm/damon/sysfs: set damon_sysfs_access_check_ktype

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 76ae47142984..81dddc99d2c9 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1258,6 +1258,22 @@ static void damon_sysfs_access_check_rm_dirs(
 	}
 }
 
+static void damon_sysfs_access_check_release(struct kobject *kobj)
+{
+	kfree(container_of(kobj, struct damon_sysfs_access_check, kobj));
+}
+
+static struct attribute *damon_sysfs_access_check_attrs[] = {
+	NULL,
+};
+ATTRIBUTE_GROUPS(damon_sysfs_access_check);
+
+static const struct kobj_type damon_sysfs_access_check_ktype = {
+	.release = damon_sysfs_access_check_release,
+	.sysfs_ops = &kobj_sysfs_ops,
+	.default_groups = damon_sysfs_access_check_groups,
+};
+
 /*
  * monitoring_attrs directory
  */
-- 
2.47.3

