From: SJ Park <sj@kernel.org>
Date: Sun, 26 Jul 2026 11:59:41 -0700
Subject: [PATCH] mm/damon/sysfs: constify fields (patch 11 fixup)

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index e348c1feed1ad..10f7939d52014 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -760,8 +760,8 @@ struct damon_sysfs_prep {
 };
 
 struct damon_sysfs_prep_action_name {
-	enum damon_prep_action action;
-	char *name;
+	const enum damon_prep_action action;
+	const char *name;
 };
 
 static const struct damon_sysfs_prep_action_name
-- 
2.47.3

