From: SeongJae Park <sj@kernel.org>
Date: Wed, 8 Oct 2025 20:35:36 -0700
Subject: [PATCH] mm/damon/core: error for remove_on_commit target of no
 matching dst

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index da563781cebd..e3d15ad97b60 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1237,6 +1237,9 @@ static int damon_commit_targets(
 	damon_for_each_target_safe(src_target, next, src) {
 		if (j++ < i)
 			continue;
+		/* target to remove has no matching dst */
+		if (src_target->remove_on_commit)
+			return -EINVAL;
 		new_target = damon_new_target();
 		if (!new_target)
 			return -ENOMEM;
-- 
2.47.3

