From: SeongJae Park <sj@kernel.org>
Date: Sun, 2 Mar 2025 13:08:14 -0800
Subject: [PATCH] mm/damon/core: add debug log for reset_regions()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 4d75f4f82708..870f248d2d5f 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1852,6 +1852,15 @@ static void kdamond_reset_aggregated(struct damon_ctx *c)
 			damon_warn_fix_nr_accesses_corruption(r);
 			r->last_nr_accesses = r->nr_accesses;
 			r->nr_accesses = 0;
+			if (r->nr_accesses_bp != r->last_nr_accesses * 10000) {
+				pr_info("reset time invalid region found!\n");
+				pr_info("nr_accesses_bp %u last_nr_acceses %u\n",
+						r->nr_accesses_bp, r->last_nr_accesses);
+				pr_info("passed_sis %lu next_aggregation_sis %lu\n",
+						c->passed_sample_intervals,
+						c->next_aggregation_sis);
+				BUG();
+			}
 		}
 		ti++;
 	}
-- 
2.47.3

