From: SeongJae Park <sj@kernel.org>
Date: Fri, 6 Mar 2026 16:52:08 -0800
Subject: [PATCH] mm/damon/core: fixup: add kernel-doc for goal_tuner

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 63f1e3fdd3866..9610fdeaa7021 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -227,6 +227,7 @@ enum damos_quota_goal_tuner {
  * @ms:			Maximum milliseconds that the scheme can use.
  * @sz:			Maximum bytes of memory that the action can be applied.
  * @goals:		Head of quota tuning goals (&damos_quota_goal) list.
+ * @goal_tuner:		Goal-based @esz tuning algorithm to use.
  * @esz:		Effective size quota in bytes.
  *
  * @weight_sz:		Weight of the region's size for prioritization.
-- 
2.47.3

