From: SJ Park <sj@kernel.org>
Date: Mon, 13 Jul 2026 20:36:49 -0700
Subject: [PATCH] mm/damon/core: mark damos_quota_goal->list private

Signed-off-by: SJ Park <sj@kernel.org>
---
 include/linux/damon.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index d094796861ac8..7310b3926e04a 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -241,6 +241,7 @@ struct damos_quota_goal {
 		u64 last_psi_total;
 
 	};
+/* private: */
 	/* List head for siblings. */
 	struct list_head list;
 };
-- 
2.47.3

