From: SeongJae Park <sj@kernel.org>
Date: Sun, 15 Dec 2024 12:45:04 -0800
Subject: [PATCH] mm/damon/paddr: support per-region filtered_out size stat

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index 0ca1bfd3d300..f380f25e2df2 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -275,8 +275,10 @@ static unsigned long damon_pa_pageout(struct damon_region *r, struct damos *s)
 		if (!folio)
 			continue;
 
-		if (damos_pa_filter_out(s, folio))
+		if (damos_pa_filter_out(s, folio)) {
+			r->damos_filtered_out += folio_size(folio);
 			goto put_folio;
+		}
 
 		folio_clear_referenced(folio);
 		folio_test_clear_young(folio);
@@ -307,8 +309,10 @@ static inline unsigned long damon_pa_mark_accessed_or_deactivate(
 		if (!folio)
 			continue;
 
-		if (damos_pa_filter_out(s, folio))
+		if (damos_pa_filter_out(s, folio)) {
+			r->damos_filtered_out += folio_size(folio);
 			goto put_folio;
+		}
 
 		if (mark_accessed)
 			folio_mark_accessed(folio);
@@ -465,8 +469,10 @@ static unsigned long damon_pa_migrate(struct damon_region *r, struct damos *s)
 		if (!folio)
 			continue;
 
-		if (damos_pa_filter_out(s, folio))
+		if (damos_pa_filter_out(s, folio)) {
+			r->damos_filtered_out += folio_size(folio);
 			goto put_folio;
+		}
 
 		if (!folio_isolate_lru(folio))
 			goto put_folio;
@@ -490,8 +496,10 @@ static unsigned long damon_pa_stat_full(struct damon_region *r, struct damos *s)
 		if (!folio)
 			continue;
 
-		if (damos_pa_filter_out(s, folio))
+		if (damos_pa_filter_out(s, folio)) {
+			r->damos_filtered_out += folio_size(folio);
 			goto put_folio;
+		}
 		applied += folio_size(folio);
 put_folio:
 		folio_put(folio);
@@ -590,6 +598,7 @@ static unsigned long damon_pa_apply_scheme(struct damon_ctx *ctx,
 		struct damon_target *t, struct damon_region *r,
 		struct damos *scheme)
 {
+	r->damos_filtered_out = 0;
 	switch (scheme->action) {
 	case DAMOS_PAGEOUT:
 		return damon_pa_pageout(r, scheme);
-- 
2.39.5

