From: SeongJae Park <sj@kernel.org>
Date: Sat, 18 Apr 2026 08:30:01 -0700
Subject: [PATCH] mm/damon/reclaim: fixup boot parameter override

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/reclaim.c | 15 ++++-----------
 1 file changed, 4 insertions(+), 11 deletions(-)

diff --git a/mm/damon/reclaim.c b/mm/damon/reclaim.c
index e83c73e5b6e95..8821c28a55832 100644
--- a/mm/damon/reclaim.c
+++ b/mm/damon/reclaim.c
@@ -381,27 +381,20 @@ static bool damon_reclaim_enabled(void)
 static int damon_reclaim_enabled_store(const char *val,
 		const struct kernel_param *kp)
 {
-	bool enable;
 	int err;
 
-	err = kstrtobool(val, &enable);
+	err = kstrtobool(val, &enabled);
 	if (err)
 		return err;
 
-	if (damon_reclaim_enabled() == enable)
+	if (damon_reclaim_enabled() == enabled)
 		return 0;
 
 	/* Called before init function.  The function will handle this. */
 	if (!damon_initialized())
-		goto set_param_out;
-
-	err = damon_reclaim_turn(enable);
-	if (err)
-		return err;
+		return 0;
 
-set_param_out:
-	enabled = enable;
-	return err;
+	return damon_reclaim_turn(enabled);
 }
 
 static int damon_reclaim_enabled_load(char *buffer,
-- 
2.47.3

