From: SeongJae Park <sj@kernel.org>
Date: Sat, 2 Aug 2025 11:19:07 -0700
Subject: [PATCH] mm/damon/sysfs: connect use_reports to core layer

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index b65645dbe3e4..0e5072c6724e 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1525,6 +1525,7 @@ static int damon_sysfs_apply_inputs(struct damon_ctx *ctx,
 	err = damon_select_ops(ctx, sys_ctx->ops_id);
 	if (err)
 		return err;
+	ctx->ops_attrs.use_reports = sys_ctx->ops_attrs->use_reports;
 	ctx->addr_unit = sys_ctx->addr_unit;
 	/* addr_unit is respected by only DAMON_OPS_PADDR */
 	if (sys_ctx->ops_id == DAMON_OPS_PADDR)
-- 
2.47.3

