From: SJ Park <sj@kernel.org>
Date: Sat, 4 Jul 2026 14:00:51 -0700
Subject: [PATCH] mm/damon/core: fix wrong access_check_disabled use in
 kdamond_fn

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 3abe611b82d35..2fbc9183b9e3c 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -4162,7 +4162,7 @@ static int kdamond_fn(void *data)
 		if (kdamond_wait_activation(ctx))
 			break;
 
-		if (access_check_disabled && ctx->ops.prepare_access_checks)
+		if (!access_check_disabled && ctx->ops.prepare_access_checks)
 			ctx->ops.prepare_access_checks(ctx);
 		if (ctx->ops.prep_probes)
 			ctx->ops.prep_probes(ctx);
@@ -4170,7 +4170,7 @@ static int kdamond_fn(void *data)
 		kdamond_usleep(sample_interval);
 		ctx->passed_sample_intervals++;
 
-		if (access_check_disabled) {
+		if (!access_check_disabled) {
 			/* todo: make these non-exclusive */
 			if (ctx->sample_control.primitives_enabled.page_fault)
 				max_nr_accesses =
-- 
2.47.3

