From: SeongJae Park <sj@kernel.org>
Date: Mon, 10 Feb 2025 16:47:50 -0800
Subject: [PATCH] mm/damon/core: add debugging log for intervals adaptation
 calculation

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index ba4cae701ebe..18a533d8735c 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1472,6 +1472,10 @@ static unsigned long damon_get_intervals_adaptation_bp(struct damon_ctx *c)
 	 */
 	if (adaptation_bp <= 10000)
 		adaptation_bp = 5000 + adaptation_bp / 2;
+
+	pr_info("access_events %lu/%lu, score_bp %lu, adaptation bp %lu\n",
+			heats, target_heats, score_bp, adaptation_bp);
+
 	return adaptation_bp;
 }
 
-- 
2.39.5

