From: SJ Park <sj@kernel.org>
Date: Wed, 24 Jun 2026 20:31:46 -0700
Subject: [PATCH] mm/damon: add damon_ctx->has_probe_weights

Signed-off-by: SJ Park <sj@kernel.org>
---
 include/linux/damon.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index a744b4b80e189..af80220d4fe4c 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -898,6 +898,9 @@ struct damon_ctx {
 	struct list_head schemes;
 
 	struct rnd_state rnd_state;
+
+	/* has non-zero probe weights */
+	bool has_probe_weights;
 };
 
 /* Get a random number in [@l, @r) using @ctx's lockless PRNG. */
-- 
2.47.3

