From: SeongJae Park <sj@kernel.org>
Date: Fri, 10 Apr 2026 16:54:25 -0700
Subject: [PATCH] mm/damon/core: fixup quota-split region merge back

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 39 ++++++++++++++++++---------------------
 1 file changed, 18 insertions(+), 21 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index b0f673efc53c8..fea1b0722c6bf 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2161,35 +2161,32 @@ static void damon_do_apply_schemes(struct damon_ctx *c,
 
 static void damos_apply_target(struct damon_ctx *c, struct damon_target *t)
 {
-	struct damon_region *r, *orig_region = NULL;
-	unsigned long orig_end_addr;
+	struct damon_region *r;
 
 	damon_for_each_region(r, t) {
+		struct damon_region *prev_r;
+
+		damon_do_apply_schemes(c, t, r);
 		/*
-		 * damon_do_apply_schemes() could split the region for the
+		 * damon_do_apply_scheems() could split the region for the
 		 * quota.  Keeping the new slices is an overhead.  Merge back
-		 * the slices into the original region if there is no reason to
-		 * keep those.
+		 * the slices into the previous region if it doesn't lose any
+		 * information.
 		 */
-		if (!orig_region || orig_end_addr <= r->ar.start) {
-			orig_region = r;
-			orig_end_addr = r->ar.end;
-		}
-		damon_do_apply_schemes(c, t, r);
-		if (r == orig_region)
+		if (damon_first_region(t) == r)
 			continue;
-		/*
-		 * If no scheme was applied to the sliced region, the age of
-		 * the slice ain't be reset.  Don't merge that back.
-		 * Otherwise, the monitored information of the region is lost.
-		 */
-		if (r->age) {
-			orig_region = NULL;
+		prev_r = damon_prev_region(r);
+		if (prev_r->ar.end != r->ar.start)
 			continue;
-		}
-		orig_region->ar.end = r->ar.end;
+		if (prev_r->age != r->age)
+			continue;
+		if (prev_r->last_nr_accesses != r->last_nr_accesses)
+			continue;
+		if (prev_r->nr_accesses != r->nr_accesses)
+			continue;
+		prev_r->ar.end = r->ar.end;
 		damon_destroy_region(r, t);
-		r = orig_region;
+		r = prev_r;
 	}
 }
 
-- 
2.47.3

