From 85dadfdc1dd4845584d69f5722b0496f2d0e69c3 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Sat, 11 Nov 2023 19:36:03 +0000
Subject: [PATCH] mm/damon/core: add debugging-purpose log of tuned esz

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 3365fc472b69..d107084c5d92 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1592,6 +1592,8 @@ static void damos_set_effective_quota(struct damos_quota *quota)
 		esz = quota->sz;
 
 	quota->esz = esz;
+
+	pr_info("esz %lu\n", esz);
 }
 
 static void damos_adjust_quota(struct damon_ctx *c, struct damos *s)
-- 
2.39.2

