From: SeongJae Park <sj@kernel.org>
Date: Sun, 3 Aug 2025 11:18:22 -0700
Subject: [PATCH] mm/damon: add ops_attrs->cpus

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 7fb57f87177b..3610aa244e32 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -609,10 +609,12 @@ enum damon_ops_id {
  *
  * @use_reports:	Whether to use damon_report_access()-ed information.
  * @write_only:		Ignore none-write accesses.
+ * @cpus:		Ignore accesses from CPUs not in this mask.
  */
 struct damon_operations_attrs {
 	bool use_reports;
 	bool write_only;
+	cpumask_t cpus;
 };
 
 /**
-- 
2.39.5

