From: SeongJae Park <sj@kernel.org>
Date: Sat, 16 Aug 2025 14:08:13 -0700
Subject: [PATCH] mm/damon/paddr: support damon_operations_attrs->tids

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index 6659cdf6e083..723fd6a140eb 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -175,11 +175,19 @@ static bool damon_pa_eligible_report(struct damon_access_report *report,
 		struct damon_target *t,
 		struct damon_operations_attrs *ops_attrs)
 {
+	int i;
+
 	if (ops_attrs->write_only && report->is_write)
 		return false;
 	if (!cpumask_test_cpu(report->cpu, &ops_attrs->cpus))
 		return false;
-	return true;
+	if (!ops_attrs->nr_tids)
+		return true;
+	for (i = 0; i < ops_attrs->nr_tids; i++) {
+		if (report->tid == ops_attrs->tids[i])
+			return true;
+	}
+	return false;
 }
 
 /*
-- 
2.39.5

