From: SeongJae Park <sj@kernel.org>
Date: Sat, 16 Aug 2025 14:07:58 -0700
Subject: [PATCH] mm/damon: extend damon_operations_attrs for per-threads
 monitoring

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 090d79696149..db437c7b4343 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -612,11 +612,15 @@ enum damon_ops_id {
  * @use_reports:	Whether to use damon_report_access()-ed information.
  * @write_only:		Ignore none-write accesses.
  * @cpus:		Ignore accesses from CPUs not in this mask.
+ * @tids:		Array of pids to monitor accesses for.
+ * @nr_tids:		Size of @tids.
  */
 struct damon_operations_attrs {
 	bool use_reports;
 	bool write_only;
 	cpumask_t cpus;
+	pid_t *tids;
+	int nr_tids;
 };
 
 /**
-- 
2.39.5

