From: SJ Park <sj@kernel.org>
Date: Sat, 1 Aug 2026 14:21:43 -0700
Subject: [PATCH] mm/damon/vaddr: use ptent for STAT (patch 5 fixup)

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/vaddr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/vaddr.c b/mm/damon/vaddr.c
index 035cd509f6b58..43ecdfc9a7a7c 100644
--- a/mm/damon/vaddr.c
+++ b/mm/damon/vaddr.c
@@ -846,7 +846,7 @@ static int damos_va_stat_pmd_entry(pmd_t *pmd, unsigned long addr,
 
 		if (!damos_va_filter_out(s, folio, vma, addr, pte, NULL))
 			*sz_filter_passed += folio_size(folio);
-		page_idx = folio_page_idx(folio, pte_page(*pte));
+		page_idx = folio_page_idx(folio, ptent);
 		nr = folio_nr_pages(folio) - page_idx;
 		s->last_applied = folio;
 	}
-- 
2.47.3

