From: SeongJae Park <sj@kernel.org>
Date: Wed, 24 Jun 2026 21:01:55 -0700
Subject: [PATCH] mm/damon/core: skip nr_accesses update if probe weights are
 set

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 8488e57a25533..26a2cdb935f59 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -4147,8 +4147,10 @@ static int kdamond_fn(void *data)
 		if (kdamond_wait_activation(ctx))
 			break;
 
-		if (ctx->ops.prepare_access_checks)
-			ctx->ops.prepare_access_checks(ctx);
+		if (!ctx->has_probe_weights) {
+			if (ctx->ops.prepare_access_checks)
+				ctx->ops.prepare_access_checks(ctx);
+		}
 		if (ctx->ops.prep_probes)
 			ctx->ops.prep_probes(ctx);
 
@@ -4156,10 +4158,13 @@ static int kdamond_fn(void *data)
 		ctx->passed_sample_intervals++;
 
 		/* todo: make these non-exclusive */
-		if (ctx->sample_control.primitives_enabled.page_fault)
-			max_nr_accesses = kdamond_check_reported_accesses(ctx);
-		else if (ctx->ops.check_accesses)
-			max_nr_accesses = ctx->ops.check_accesses(ctx);
+		if (!ctx->has_probe_weights) {
+			if (ctx->sample_control.primitives_enabled.page_fault)
+				max_nr_accesses =
+					kdamond_check_reported_accesses(ctx);
+			else if (ctx->ops.check_accesses)
+				max_nr_accesses = ctx->ops.check_accesses(ctx);
+		}
 		if (ctx->ops.apply_probes)
 			ctx->ops.apply_probes(ctx);
 
-- 
2.47.3

