From: SeongJae Park <sj@kernel.org>
Date: Fri, 14 Feb 2025 17:11:27 -0800
Subject: [PATCH] mm/damon/core: set damos->reject_by_default

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 26 ++++++++++++++++++++++++--
 1 file changed, 24 insertions(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index ad5eadacd762..ac6481e94cb0 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -904,6 +904,24 @@ static int damos_commit_ops_filters(struct damos *dst, struct damos *src)
 	return 0;
 }
 
+static void damos_set_reject_by_default(struct damos *s)
+{
+	struct damos_filter *last_filter;
+
+	if (list_empty(&s->filters) && list_empty(&s->ops_filters)) {
+		s->reject_by_default = false;
+		return;
+	}
+
+	if (!list_empty(&s->ops_filters))
+		last_filter = list_last_entry(
+				&s->ops_filters, struct damos_filter, list);
+	else
+		last_filter = list_last_entry(
+				&s->filters, struct damos_filter, list);
+	s->reject_by_default = last_filter->allow;
+}
+
 static int damos_commit_filters(struct damos *dst, struct damos *src)
 {
 	int err;
@@ -911,7 +929,11 @@ static int damos_commit_filters(struct damos *dst, struct damos *src)
 	err = damos_commit_core_filters(dst, src);
 	if (err)
 		return err;
-	return damos_commit_ops_filters(dst, src);
+	err = damos_commit_ops_filters(dst, src);
+	if (err)
+		return err;
+	damos_set_reject_by_default(dst);
+	return 0;
 }
 
 static struct damos *damon_nth_scheme(int n, struct damon_ctx *ctx)
@@ -1590,7 +1612,7 @@ static bool damos_filter_out(struct damon_ctx *ctx, struct damon_target *t,
 		if (damos_filter_match(ctx, t, r, filter))
 			return !filter->allow;
 	}
-	return false;
+	return s->reject_by_default;
 }
 
 /*
-- 
2.39.5

