From: SJ Park <sj@kernel.org>
Date: Sun, 26 Jul 2026 11:45:33 -0700
Subject: [PATCH] mm/damon/core: fixup prep iterator macro param name (patch 5
 fixup)

Signed-off-by: SJ Park <sj@kernel.org>
---
 include/linux/damon.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 6918203045aa1..289a535cb74da 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -1123,11 +1123,11 @@ static inline unsigned long damon_sz_region(struct damon_region *r)
 	return r->ar.end - r->ar.start;
 }
 
-#define damon_for_each_prep(p, prep) \
-	list_for_each_entry(p, &(prep)->preps, list)
+#define damon_for_each_prep(p, probe) \
+	list_for_each_entry(p, &(probe)->preps, list)
 
-#define damon_for_each_prep_safe(p, next, prep) \
-	list_for_each_entry_safe(p, next, &(prep)->preps, list)
+#define damon_for_each_prep_safe(p, next, probe) \
+	list_for_each_entry_safe(p, next, &(probe)->preps, list)
 
 #define damon_for_each_filter(f, p) \
 	list_for_each_entry(f, &(p)->filters, list)
-- 
2.47.3

