From: SeongJae Park <sj@kernel.org>
Date: Thu, 6 Mar 2025 16:27:52 -0800
Subject: [PATCH] mm/damon/core: fix build error when !CONFIG_NUMA

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 34b4d1b948f5..5fcef3e3c436 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1946,6 +1946,7 @@ static inline u64 damos_get_some_mem_psi_total(void)
 
 #endif	/* CONFIG_PSI */
 
+#ifdef CONFIG_NUMA
 static __kernel_ulong_t damos_get_node_mem_bp(
 		struct damos_quota_goal *goal)
 {
@@ -1959,6 +1960,14 @@ static __kernel_ulong_t damos_get_node_mem_bp(
 		numerator = i.freeram;
 	return numerator * 10000 / i.totalram;
 }
+#else
+static __kernel_ulong_t damos_get_node_mem_bp(
+		struct damos_quota_goal *goal)
+{
+	return 0;
+}
+#endif
+
 
 static void damos_set_quota_goal_current_value(struct damos_quota_goal *goal)
 {
-- 
2.39.5

