From: SeongJae Park <sj@kernel.org>
Date: Sun, 30 Nov 2025 10:52:41 -0800
Subject: [PATCH] mm/damon/sysfs: support threads type sample filter

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 014404e4b679..699de65b3a4f 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -778,6 +778,10 @@ damon_sysfs_sample_filter_type_names[] = {
 		.type = DAMON_FILTER_TYPE_CPUMASK,
 		.name = "cpumask",
 	},
+	{
+		.type = DAMON_FILTER_TYPE_THREADS,
+		.name = "threads",
+	},
 	{
 		.type = DAMON_FILTER_TYPE_WRITE,
 		.name = "write",
-- 
2.47.3

