From: SeongJae Park <sj@kernel.org>
Date: Fri, 28 Nov 2025 14:55:50 -0800
Subject: [PATCH] mm/damon: implement damon_report_filters iterate macros

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 171134ae0fe7..ff020b229d11 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -969,6 +969,12 @@ static inline unsigned long damon_sz_region(struct damon_region *r)
 #define damon_for_each_target_safe(t, next, ctx)	\
 	list_for_each_entry_safe(t, next, &(ctx)->adaptive_targets, list)
 
+#define damon_for_each_report_filter(f, control) \
+	list_for_each_entry(f, &control->report_filters, list)
+
+#define damon_for_each_report_filter_safe(f, next, control) \
+	list_for_each_entry_safe(f, next, &control->report_filteers, list)
+
 #define damon_for_each_scheme(s, ctx) \
 	list_for_each_entry(s, &(ctx)->schemes, list)
 
-- 
2.47.3

