From: SJ Park <sj@kernel.org>
Date: Tue, 25 Aug 2026 07:21:45 -0700
Subject: [PATCH] mm/damon/core: commit range_{min,max} for probe hits wsum
 filter

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index abb30819d48c0..f24bf265c6466 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1407,6 +1407,9 @@ static void damos_commit_filter_arg(
 	case DAMOS_FILTER_TYPE_HUGEPAGE_SIZE:
 		dst->sz_range = src->sz_range;
 		break;
+	case DAMOS_FILTER_TYPE_PROBE_HITS_WSUM:
+		dst->range_min = src->range_min;
+		dst->range_max = src->range_max;
 	default:
 		break;
 	}
-- 
2.47.3

