From: SeongJae Park <sj@kernel.org>
Date: Sat, 2 Aug 2025 13:27:00 -0700
Subject: [PATCH] mm/damon/sysfs: pass write_only to core

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 56ad59cd13dd..70cdaed0dadc 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1551,6 +1551,7 @@ static int damon_sysfs_apply_inputs(struct damon_ctx *ctx,
 	if (err)
 		return err;
 	ctx->ops_attrs.use_reports = sys_ctx->ops_attrs->use_reports;
+	ctx->ops_attrs.write_only = sys_ctx->ops_attrs->write_only;
 	ctx->addr_unit = sys_ctx->addr_unit;
 	ctx->min_sz_region = max(DAMON_MIN_REGION / sys_ctx->addr_unit, 1);
 	err = damon_sysfs_set_attrs(ctx, sys_ctx->attrs);
-- 
2.39.5

