From: SeongJae Park <sj@kernel.org>
Date: Sat, 11 Nov 2023 19:36:03 +0000
Subject: [PATCH] mm/damon/core: add debugging-purpose log of tuned esz

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index f9673aa38853..e6d89350436b 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2255,6 +2255,8 @@ static void damos_set_effective_quota(struct damos_quota *quota)
 		esz = quota->sz;
 
 	quota->esz = esz;
+
+	pr_info("esz %lu\n", esz);
 }
 
 static void damos_trace_esz(struct damon_ctx *c, struct damos *s,
-- 
2.47.3

