From: SeongJae Park <sj@kernel.org>
Date: Sun, 12 Jan 2025 11:49:02 -0800
Subject: [PATCH] mm/damon/core: do not call damos_walk_control->walk() if walk
 is completed

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index ad4e17045b6a..97ba0e5b6eac 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1482,6 +1482,9 @@ static void damos_walk_call_walk(struct damon_ctx *ctx, struct damon_target *t,
 {
 	struct damos_walk_control *control;
 
+	if (s->walk_completed)
+		return;
+
 	mutex_lock(&ctx->walk_control_lock);
 	control = ctx->walk_control;
 	mutex_unlock(&ctx->walk_control_lock);
-- 
2.39.5

