From: SeongJae Park <sj@kernel.org>
Date: Tue, 23 Jun 2026 21:16:46 -0700
Subject: [PATCH] mm/damon/paddr: update for probe->preps

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index 1bed532e1ddf8..85cd64a55fff3 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -185,11 +185,11 @@ static unsigned int damon_pa_check_accesses(struct damon_ctx *ctx)
 }
 
 static void damon_pa_prep_probes_region(struct damon_region *r,
-		struct damon_ctx *ctx)
+		struct damon_probe *probe, struct damon_ctx *ctx)
 {
 	struct damon_prep *p;
 
-	damon_for_each_prep(p, ctx) {
+	damon_for_each_prep(p, probe) {
 		switch (p->action) {
 		case DAMON_PREP_SET_PGIDLE:
 			damon_pa_mkold(damon_pa_phys_addr(r->sampling_addr,
@@ -205,10 +205,13 @@ static void damon_pa_prep_probes(struct damon_ctx *ctx)
 {
 	struct damon_target *t;
 	struct damon_region *r;
+	struct damon_probe *p;
 
 	damon_for_each_target(t, ctx) {
-		damon_for_each_region(r, t)
-			damon_pa_prep_probes_region(r, ctx);
+		damon_for_each_region(r, t) {
+			damon_for_each_probe(p, ctx)
+				damon_pa_prep_probes_region(r, p, ctx);
+		}
 	}
 }
 
-- 
2.47.3

