From: SJ Park <sj@kernel.org>
Date: Wed, 24 Jun 2026 20:31:46 -0700
Subject: [PATCH] mm/damon: add damon_ctx->has_probe_weights

Signed-off-by: SJ Park <sj@kernel.org>
---
 include/linux/damon.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 4bcfc77b79b5a..7b82afdf58df1 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -1070,6 +1070,9 @@ struct damon_ctx {
 
 	/* @rnd_state:	Per-ctx PRNG state for damon_rand(). */
 	struct rnd_state rnd_state;
+
+	/* has non-zero probe weights */
+	bool has_probe_weights;
 };
 
 /* Get a random number in [@l, @r) using @ctx's lockless PRNG. */
-- 
2.47.3

