From: SeongJae Park <sj@kernel.org>
Date: Sat, 18 Apr 2026 12:00:54 -0700
Subject: [PATCH] mm/damon: add damon_data_attr->filters

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 25a2677f22d57..374aa3555049e 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -792,12 +792,35 @@ struct damon_intervals_goal {
 	unsigned long max_sample_us;
 };
 
+/**
+ * enum damon_filter_type - Type of memory for &struct damon_filter
+ *
+ * @DAMON_FILTER_TYPE_ANON:	Anonymous pages.
+ */
+enum damon_filter_type {
+	DAMON_FILTER_TYPE_ANON,
+};
+
+/**
+ * struct damon_filter - Data attribute filter.
+ *
+ * @type:	Type of the attribute.
+ * @matcing:	Whether this filter is for the type-matching ones.
+ * @allow:	Whether to allow monitoring the @type-@matching ones.
+ */
+struct damon_filter {
+	enum damon_filter_type type;
+	bool matching;
+	bool allow;
+};
+
 /**
  * struct damon_data_attr - Data attribute to monitor.
  *
  * @list:	List for siblings.
  */
 struct damon_data_attr {
+	struct list_head filters;
 	struct list_head list;
 };
 
-- 
2.47.3

