From: SeongJae Park <sj@kernel.org>
Date: Sun, 4 Jan 2026 15:17:37 -0800
Subject: [PATCH] mm/damon/Kconfig: make DAMON_HARDENED depends on DAMON

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/Kconfig b/mm/damon/Kconfig
index cac18b5b617c..567eb86a53e3 100644
--- a/mm/damon/Kconfig
+++ b/mm/damon/Kconfig
@@ -122,6 +122,7 @@ config DAMON_ACMA
 
 config DAMON_HARDENED
 	bool "Check integrity of DAMON code"
+	depends on DAMON
 	help
 	  This enables additional DAMON debugging-purpose integrity checks in
 	  DAMON code.  This can impose additional overhead but can be useful at
-- 
2.47.3

