From: SeongJae Park <sj@kernel.org>
Date: Tue, 21 Jan 2025 17:34:59 -0800
Subject: [PATCH] mm/damon/sysfs: setup intervals tuning parameters as user
 specified

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index f47611caf424..a53413ea406b 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1201,6 +1201,10 @@ static int damon_sysfs_set_attrs(struct damon_ctx *ctx,
 		.ops_update_interval = sys_intervals->update_us,
 		.min_nr_regions = sys_nr_regions->min,
 		.max_nr_regions = sys_nr_regions->max,
+		.tune_interval_aggrs = sys_intervals->tune_aggrs,
+		.target_access_samples_bp =
+			sys_intervals->target_access_samples_bp,
+		.max_aggr_interval = sys_intervals->max_aggr_us,
 	};
 	return damon_set_attrs(ctx, &attrs);
 }
-- 
2.39.5

