From: SJ Park <sj@kernel.org>
Date: Sun, 2 Aug 2026 11:33:04 -0700
Subject: [PATCH] mm/damon/lru_sort: avoid unnecessary damon_has_started set

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/lru_sort.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mm/damon/lru_sort.c b/mm/damon/lru_sort.c
index 153dc258bbf9a..bd847829a9907 100644
--- a/mm/damon/lru_sort.c
+++ b/mm/damon/lru_sort.c
@@ -422,7 +422,8 @@ static int damon_lru_sort_turn(bool on)
 	err = damon_start(&ctx, 1, true);
 	if (err)
 		return err;
-	damon_lru_sort_damon_has_started = true;
+	if (!damon_lru_sort_damon_has_started)
+		damon_lru_sort_damon_has_started = true;
 	return damon_call(ctx, &call_control);
 }
 
-- 
2.47.3

