From: SeongJae Park <sj@kernel.org>
Date: Sun, 3 Aug 2025 11:32:44 -0700
Subject: [PATCH] mm/damon/paddr: filter reports based on cpus

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index 61bb4111dd19..e235d33f2e2f 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -177,6 +177,8 @@ static bool damon_pa_eligible_report(struct damon_access_report *report,
 {
 	if (ops_attrs->write_only && report->is_write)
 		return false;
+	if (!cpumask_test_cpu(report->cpu, &ops_attrs->cpus))
+		return false;
 	return true;
 }
 
-- 
2.39.5

