From: SJ Park <sj@kernel.org>
Date: Tue, 25 Aug 2026 17:48:03 -0700
Subject: [PATCH] mm/damon: wordsmith probe hits wsum comment

Signed-off-by: SJ Park <sj@kernel.org>
---
 include/linux/damon.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 5ab7300e0b796..46a91762b457a 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -433,7 +433,7 @@ struct damos_stat {
  * @DAMOS_FILTER_TYPE_UNMAPPED:	Unmapped pages.
  * @DAMOS_FILTER_TYPE_ADDR:	Address range.
  * @DAMOS_FILTER_TYPE_TARGET:	Data Access Monitoring target.
- * @DAMOS_FILTER_TYPE_PROBE_HITS_WSUM:	probe_hits wsum range.
+ * @DAMOS_FILTER_TYPE_PROBE_HITS_WSUM:	probe_hits weighted sum range.
  *
  * All types except &DAMOS_FILTER_TYPE_ADDR and &DAMOS_FILTER_TYPE_TARGET
  * are handled by the underlying &struct damon_operations as a part of scheme
-- 
2.47.3

