From: SeongJae Park <sj@kernel.org>
Date: Mon, 15 Jun 2026 21:26:34 -0700
Subject: [PATCH] mm/damon: implement damon_for_each_prep()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index a1b9e6bb882f8..4dcb43180b9b8 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -1111,6 +1111,9 @@ static inline unsigned long damon_sz_region(struct damon_region *r)
 #define damon_for_each_filter_safe(f, next, p) \
 	list_for_each_entry_safe(f, next, &(p)->filters, list)
 
+#define damon_for_each_prep(p, ctx) \
+	list_for_each_entry(p, &(ctx)->preps, list)
+
 #define damon_for_each_probe(p, ctx) \
 	list_for_each_entry(p, &(ctx)->probes, list)
 
-- 
2.47.3

