From: SeongJae Park <sj@kernel.org>
Date: Sun, 12 Oct 2025 14:55:05 -0700
Subject: [PATCH] mm/damon/tests/core-kunit: add test for quota goals commit

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/tests/core-kunit.h | 42 +++++++++++++++++++++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/mm/damon/tests/core-kunit.h b/mm/damon/tests/core-kunit.h
index 3b1f44a71de2..22cec9ec20c5 100644
--- a/mm/damon/tests/core-kunit.h
+++ b/mm/damon/tests/core-kunit.h
@@ -537,6 +537,47 @@ static void damos_test_commit_quota_goal(struct kunit *test)
 	damos_destroy_quota_goal(dst);
 }
 
+static void damos_test_commit_quota_goals(struct kunit *test)
+{
+	struct damos_quota dst = {
+		.reset_interval = 1,
+		.ms = 2,
+		.sz = 3,
+		.weight_sz = 4,
+		.weight_nr_accesses = 5,
+		.weight_age = 6,
+	};
+	struct damos_quota src = dst;
+	struct damos_quota_goal *goal, *dst_goal, *next;
+	int nr_dst_goals = 0;
+
+	INIT_LIST_HEAD(&dst.goals);
+	INIT_LIST_HEAD(&src.goals);
+
+	goal = damos_new_quota_goal(DAMOS_QUOTA_USER_INPUT, 1000);
+	if (!goal)
+		kunit_skip(test, "gaol alloc fail\n");
+	goal->current_value = 500;
+	damos_add_quota_goal(&src, goal);
+	damos_commit_quota_goals(&dst, &src);
+
+	damos_for_each_quota_goal(dst_goal, (&dst)) {
+		nr_dst_goals++;
+		KUNIT_EXPECT_EQ(test, dst_goal->target_value,
+				goal->target_value);
+		KUNIT_EXPECT_EQ(test, dst_goal->metric, goal->metric);
+		KUNIT_EXPECT_EQ(test, dst_goal->current_value,
+				goal->current_value);
+	}
+	KUNIT_EXPECT_EQ(test, nr_dst_goals, 1);
+
+	/* TODO: test in place commit and removal commit */
+
+	damos_destroy_quota_goal(goal);
+	damos_for_each_quota_goal_safe(dst_goal, next, (&dst))
+		damos_destroy_quota_goal(dst_goal);
+}
+
 static void damos_test_commit_quota(struct kunit *test)
 {
 	struct damos_quota dst = {
@@ -1000,6 +1041,7 @@ static struct kunit_case damon_test_cases[] = {
 	KUNIT_CASE(damon_test_moving_sum),
 	KUNIT_CASE(damos_test_new_filter),
 	KUNIT_CASE(damos_test_commit_quota_goal),
+	KUNIT_CASE(damos_test_commit_quota_goals),
 	KUNIT_CASE(damos_test_commit_quota),
 	KUNIT_CASE(damos_test_commit_filter),
 	KUNIT_CASE(damos_test_filter_out),
-- 
2.47.3

