From: SeongJae Park <sj@kernel.org>
Date: Sun, 22 Jun 2025 12:25:44 -0700
Subject: [PATCH] mm/damon/core: support changed active_mem_bp definition

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 3063aa9b7110..e108bf2cba98 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2069,7 +2069,7 @@ static unsigned long damos_get_node_memcg_used_bp(
 /*
  * Returns LRU-active memory to LRU-inactive memory size ratio.
  */
-static unsigned int damos_get_active_inactive_mem_bp(void)
+static unsigned int damos_get_active_mem_bp(void)
 {
 	unsigned long active, inactive;
 
@@ -2078,7 +2078,7 @@ static unsigned int damos_get_active_inactive_mem_bp(void)
 		global_node_page_state(NR_LRU_BASE + LRU_ACTIVE_FILE);
 	inactive = global_node_page_state(NR_LRU_BASE + LRU_INACTIVE_ANON) +
 		global_node_page_state(NR_LRU_BASE + LRU_INACTIVE_FILE);
-	return active * 10000 / inactive;
+	return active * 10000 / (active + inactive);
 }
 
 static void damos_set_quota_goal_current_value(struct damos_quota_goal *goal)
@@ -2104,7 +2104,7 @@ static void damos_set_quota_goal_current_value(struct damos_quota_goal *goal)
 		goal->current_value = damos_get_node_memcg_used_bp(goal);
 		break;
 	case DAMOS_QUOTA_ACTIVE_MEM_BP:
-		goal->current_value = damos_get_active_inactive_mem_bp();
+		goal->current_value = damos_get_active_mem_bp();
 		break;
 	default:
 		break;
-- 
2.39.5

