From: SeongJae Park <sj@kernel.org>
Date: Sun, 19 Apr 2026 10:42:08 -0700
Subject: [PATCH] mm/damon: add damon_filter->list

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 79405c1874086..8711623ac660f 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -814,6 +814,7 @@ struct damon_filter {
 	enum damon_filter_type type;
 	bool matching;
 	bool allow;
+	struct list_head list;
 };
 
 /**
-- 
2.47.3

