From: SJ Park <sj@kernel.org>
Date: Fri, 10 Jul 2026 20:21:23 -0700
Subject: [PATCH] mm/damon/core: refactor inifnite merge fix

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 7bf1ae6c6a94a..36f39b6dba83c 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3719,7 +3719,7 @@ static void kdamond_merge_regions(struct damon_ctx *c, unsigned int threshold,
 	bool count_age = true;
 
 	max_thres = damon_nr_samples_per_aggr(&c->attrs);
-	do {
+	while (true) {
 		nr_regions = 0;
 		damon_for_each_target(t, c) {
 			damon_merge_regions_of(t, threshold, sz_limit, c,
@@ -3727,13 +3727,14 @@ static void kdamond_merge_regions(struct damon_ctx *c, unsigned int threshold,
 			nr_regions += damon_nr_regions(t);
 		}
 		count_age = false;
-		if (threshold >= max_thres)
+		if (nr_regions <= c->attrs.max_nr_regions ||
+				max_thres <= threshold)
 			break;
 		if (threshold < max_thres / 2)
 			threshold = max(1, threshold * 2);
 		else
 			threshold = max_thres;
-	} while (nr_regions > c->attrs.max_nr_regions);
+	}
 }
 
 #ifdef CONFIG_DAMON_DEBUG_SANITY
-- 
2.47.3

