From: SeongJae Park <sj@kernel.org>
Date: Mon, 15 Jun 2026 21:37:46 -0700
Subject: [PATCH] mm/damon: add damon_ctx->preps

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 70e1572b3d161..0082fd78ab0c3 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -1064,6 +1064,8 @@ struct damon_ctx {
 /* private: */
 	/* Head of monitoring targets (&damon_target) list. */
 	struct list_head adaptive_targets;
+	/* Head of damon_prep list. */
+	struct list_head preps;
 	/* Head of schemes (&damos) list. */
 	struct list_head schemes;
 
-- 
2.47.3

