From: SeongJae Park <sj@kernel.org>
Date: Mon, 1 Dec 2025 22:10:51 -0800
Subject: [PATCH] mm/damon: reorder damon_acces_report fields

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 2c68725e9177..e9dafa38fbb3 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -113,20 +113,18 @@ struct damon_target {
 
 /**
  * struct damon_access_report - Represent single acces report information.
- * @tid:		The task id of the accessing task.
  * @addr:		The start address of the reporting region.
  * @size:		The size of the reporting region.
- * @is_write:		Whether the access is write.
  * @cpu:		CPU that made the access.
- *
- * @pid could be stale, and hence shouldn't be de-referenced.
+ * @tid:		The task id of the accessing task.
+ * @is_write:		Whether the access is write.
  */
 struct damon_access_report {
-	pid_t tid;
 	unsigned long addr;
 	unsigned long size;
-	bool is_write;
 	unsigned int cpu;
+	pid_t tid;
+	bool is_write;
 /* private: */
 	unsigned long report_jiffies;	/* when this report is made */
 };
-- 
2.47.3

