From: SeongJae Park <sj@kernel.org>
Date: Tue, 28 Jan 2025 20:04:05 -0800
Subject: [PATCH] mm/damon/core: fix build warnings

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index fd9073480ff4..8da795a5d69c 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1321,13 +1321,14 @@ static unsigned long damon_get_intervals_adaptation_bp(struct damon_ctx *c)
 		10000;
 
 	pr_info("access_samples %lu/%lu, score_bp %lu, adaptation bp %lu\n",
-			access_samples, score_bp, adaptation_bp);
+			access_samples, c->attrs.intervals_goal.samples,
+			score_bp, adaptation_bp);
 	return adaptation_bp;
 }
 
 static void kdamond_tune_intervals(struct damon_ctx *c)
 {
-	unsigned long adaptation_bp, sample_to_aggr_bp;
+	unsigned long adaptation_bp;
 	struct damon_attrs new_attrs;
 	struct damon_intervals_goal *goal;
 
-- 
2.39.5

