From: SeongJae Park <sj@kernel.org>
Date: Mon, 11 May 2026 20:48:13 -0700
Subject: [PATCH] mm/damon/sysfs.c: rename

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 2e0b7eca6bcbf..ed7bb734a236b 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -771,7 +771,7 @@ struct damon_sysfs_filter_type_name {
 static const struct damon_sysfs_filter_type_name
 damon_sysfs_filter_type_names[] = {
 	{
-		.type = DAMON_TEST_TYPE_ANON,
+		.type = DAMON_FILTER_TYPE_ANON,
 		.name = "anon",
 	},
 };
-- 
2.47.3

