From: SeongJae Park <sj@kernel.org>
Date: Wed, 11 Mar 2026 17:10:08 -0700
Subject: [PATCH] mm/damon: remove biggest_system_ram_default() declaration

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index bf500815d6f23..32bbcf7733d64 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -991,11 +991,6 @@ int damon_set_region_system_rams_default(struct damon_target *t,
 				unsigned long addr_unit,
 				unsigned long min_region_sz);
 
-int damon_set_region_biggest_system_ram_default(struct damon_target *t,
-				unsigned long *start, unsigned long *end,
-				unsigned long addr_unit,
-				unsigned long min_region_sz);
-
 #endif	/* CONFIG_DAMON */
 
 #endif	/* _DAMON_H */
-- 
2.47.3

