From: SeongJae Park <sj@kernel.org>
Date: Mon, 9 Mar 2026 18:19:00 -0700
Subject: [PATCH] mm/damon: (fixup) fix typos

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 61885e36c7800..ebbe4ad582f5d 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -117,13 +117,13 @@ struct damon_target {
 };
 
 /**
- * struct damon_access_report - Represent single acces report information.
+ * struct damon_access_report - Represent single access report information.
  * @addr:		The start address of the accessed address range.
  * @size:		The size of the accessed address range.
  *
  * Any DAMON API callers that notified access events can report the information
  * to DAMON using damon_report_access().  This struct contains the reporting
- * infomration.  Refer to damon_report_access() for more details.
+ * information.  Refer to damon_report_access() for more details.
  */
 struct damon_access_report {
 	unsigned long addr;
-- 
2.47.3

