From: SJ Park <sj@kernel.org>
Date: Wed, 1 Jul 2026 07:16:39 -0700
Subject: [PATCH] mm/damon/core: skip nr_accesses update if probe weights are
 set

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index ca37373297de4..a3c46c195482a 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3708,14 +3708,15 @@ static int kdamond_fn(void *data)
 		if (kdamond_wait_activation(ctx))
 			break;
 
-		if (ctx->ops.prepare_access_checks)
+		if (!ctx->has_probe_weights && ctx->ops.prepare_access_checks)
 			ctx->ops.prepare_access_checks(ctx);
 
 		kdamond_usleep(sample_interval);
 		ctx->passed_sample_intervals++;
 
-		if (ctx->ops.check_accesses)
+		if (!ctx->has_probe_weights && ctx->ops.check_accesses)
 			max_nr_accesses = ctx->ops.check_accesses(ctx);
+
 		if (ctx->ops.apply_probes)
 			ctx->ops.apply_probes(ctx);
 
-- 
2.47.3

