From: SJ Park <sj@kernel.org>
Date: Wed, 26 Aug 2026 21:51:26 -0700
Subject: [PATCH] mm/damon/core: fixup last_psi_total initialization

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 3c97ab94278d8..e6d69e82e0319 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1271,6 +1271,8 @@ static struct damos_quota_goal *damos_nth_quota_goal(
 	return NULL;
 }
 
+static u64 damos_get_some_mem_psi_total(void);
+
 static void damos_commit_quota_goal_union(
 		struct damos_quota_goal *dst, struct damos_quota_goal *src)
 {
-- 
2.47.3

