From: SeongJae Park <sj@kernel.org>
Date: Sat, 21 Jun 2025 09:41:15 -0700
Subject: [PATCH] mm/damon: rename id and weight to have _arr suffixes

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index c8563008f722..18a56ddd259d 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -492,8 +492,8 @@ struct damos_access_pattern {
 };
 
 struct damos_migrate_dest {
-	unsigned *node_id;
-	unsigned *weight;
+	unsigned *node_id_arr;
+	unsigned *weight_arr;
 	size_t nr_dests;
 };
 
-- 
2.39.5

