From: SJ Park <sj@kernel.org>
Date: Tue, 7 Jul 2026 23:14:21 -0700
Subject: [PATCH] mm/damon/core: show number of probes in valid_probe_params()

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 28f83ccc63068..3ea55596a9234 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1345,6 +1345,13 @@ static bool damon_valid_probe_params(struct damon_ctx *ctx)
 	unsigned char max_probe_hits;
 	struct damon_probe *probe;
 	unsigned int wsum, wsum_to_add;
+	int nr_probes;
+
+	nr_probes = 0;
+	damon_for_each_probe(probe, ctx)
+		nr_probes++;
+	if (nr_probes > DAMON_MAX_PROBES)
+		return false;
 
 	if (!damon_has_probe_weights(ctx))
 		return true;
-- 
2.47.3

