From: SeongJae Park <sj@kernel.org>
Date: Fri, 15 May 2026 17:08:21 -0700
Subject: [PATCH] mm/damon: (fixup 2) enclose ctx in damon_for_each_probe

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index dc53bc3da74b3..43d71eb24ccb6 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -911,10 +911,10 @@ static inline unsigned long damon_sz_region(struct damon_region *r)
 }
 
 #define damon_for_each_probe(p, ctx) \
-	list_for_each_entry(p, &ctx->probes, list)
+	list_for_each_entry(p, &(ctx)->probes, list)
 
 #define damon_for_each_probe_safe(p, next, ctx) \
-	list_for_each_entry_safe(p, next, &ctx->probes, list)
+	list_for_each_entry_safe(p, next, &(ctx)->probes, list)
 
 #define damon_for_each_region(r, t) \
 	list_for_each_entry(r, &t->regions_list, list)
-- 
2.47.3

