From: SeongJae Park <sj@kernel.org>
Date: Sat, 21 Jun 2025 09:42:00 -0700
Subject: [PATCH] mm/damon/core: initialize and cleanup damos->migrate_dest

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index ee3d6d4b3c9b..41cd947c20d8 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -425,6 +425,7 @@ struct damos *damon_new_scheme(struct damos_access_pattern *pattern,
 	scheme->wmarks = *wmarks;
 	scheme->wmarks.activated = true;
 
+	scheme->migrate_dest = (struct damos_migrate_dest){};
 	scheme->target_nid = target_nid;
 
 	return scheme;
@@ -467,6 +468,9 @@ void damon_destroy_scheme(struct damos *s)
 
 	damos_for_each_filter_safe(f, next, s)
 		damos_destroy_filter(f);
+
+	kfree(s->migrate_dest.node_id_arr);
+	kfree(s->migrate_dest.weight_arr);
 	damon_del_scheme(s);
 	damon_free_scheme(s);
 }
-- 
2.39.5

