From: SeongJae Park <sj@kernel.org>
Date: Thu, 1 Jan 2026 14:00:16 -0800
Subject: [PATCH] mm/damon/core: cancel damos_walk() before ctx->kdamond reset

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index ae7e5645f2b8..c87cb28a5f6f 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3274,14 +3274,13 @@ static int kdamond_fn(void *data)
 		ctx->ops.cleanup(ctx);
 	kfree(ctx->regions_score_histogram);
 	kdamond_call(ctx, true);
+	damos_walk_cancel(ctx);
 
 	pr_debug("kdamond (%d) finishes\n", current->pid);
 	mutex_lock(&ctx->kdamond_lock);
 	ctx->kdamond = NULL;
 	mutex_unlock(&ctx->kdamond_lock);
 
-	damos_walk_cancel(ctx);
-
 	mutex_lock(&damon_lock);
 	nr_running_ctxs--;
 	if (!nr_running_ctxs && running_exclusive_ctxs)
-- 
2.47.3

