From: SJ Park <sj@kernel.org>
Date: Mon, 6 Jul 2026 18:12:25 -0700
Subject: [PATCH] mm/damon/core: use abs_diff() instead of abs()

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 4c9fc8de3d71c..9fdb5599b572b 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3665,7 +3665,7 @@ static void damon_merge_regions_of(struct damon_target *t, unsigned int thres,
 		score = damon_merge_score(r, false, ctx, use_probe_hits);
 		last_score = damon_merge_score(r, true, ctx, use_probe_hits);
 
-		if (abs(score - last_score) > thres)
+		if (abs_diff(score, last_score) > thres)
 			r->age = 0;
 		else if ((score == 0) != (last_score == 0))
 			r->age = 0;
@@ -3676,8 +3676,9 @@ static void damon_merge_regions_of(struct damon_target *t, unsigned int thres,
 			goto set_prev_continue;
 		if (prev->ar.end != r->ar.start)
 			goto set_prev_continue;
-		if (abs(damon_merge_score(prev, false, ctx, use_probe_hits) -
-					score) > thres)
+		if (abs_diff(damon_merge_score(prev, false, ctx,
+						use_probe_hits), score)
+				> thres)
 			goto set_prev_continue;
 		if (damon_sz_region(prev) + damon_sz_region(r) > sz_limit)
 			goto set_prev_continue;
-- 
2.47.3

