From: SeongJae Park <sj@kernel.org>
Date: Sun, 19 Apr 2026 12:24:21 -0700
Subject: [PATCH] mm/damon/sysfs: setup data_attrs

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 37 +++++++++++++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index aa105ba0a8236..a2395e12dc08b 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1878,6 +1878,40 @@ static int damon_sysfs_set_attrs(struct damon_ctx *ctx,
 	return damon_set_attrs(ctx, &attrs);
 }
 
+static int damon_sysfs_set_data_attrs(struct damon_ctx *ctx,
+		struct damon_sysfs_data_attrs *sys_data_attrs)
+{
+	int i;
+
+	for (i = 0; i < sys_data_attrs->nr; i++) {
+		struct damon_sysfs_filters *sys_filters =
+			sys_data_attrs->attrs_arr[i]->filters;
+		struct damon_data_attr *a;
+		int j;
+
+		if (!sys_filters)
+			continue;
+		a = damon_new_data_attr();
+		if (!a)
+			return -ENOMEM;
+		damon_add_data_attr(ctx, a);
+
+		for (j = 0; j < sys_filters->nr; j++) {
+			struct damon_sysfs_filter *sys_filter =
+				sys_filters->filters_arr[j];
+			struct damon_filter *f;
+
+			f = damon_new_filter(sys_filter->type,
+					sys_filter->matching,
+					sys_filter->allow);
+			if (!f)
+				return -ENOMEM;
+			damon_add_filter(a, f);
+		}
+	}
+	return 0;
+}
+
 static int damon_sysfs_set_regions(struct damon_target *t,
 		struct damon_sysfs_regions *sysfs_regions,
 		unsigned long min_region_sz)
@@ -1990,6 +2024,9 @@ static int damon_sysfs_apply_inputs(struct damon_ctx *ctx,
 				DAMON_MIN_REGION_SZ / sys_ctx->addr_unit, 1);
 	ctx->pause = sys_ctx->pause;
 	err = damon_sysfs_set_attrs(ctx, sys_ctx->attrs);
+	if (err)
+		return err;
+	err = damon_sysfs_set_data_attrs(ctx, sys_ctx->attrs->data_attrs);
 	if (err)
 		return err;
 	err = damon_sysfs_set_sample_control(&ctx->sample_control,
-- 
2.47.3

