From: SJ Park <sj@kernel.org>
Date: Mon, 24 Aug 2026 21:40:28 -0700
Subject: [PATCH] mm/damon/core: implement damos_quota_goal->complement

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index c391172bf0c36..050f480c65e99 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3411,6 +3411,15 @@ static void damos_set_quota_goal_current_value(struct damon_ctx *c,
 	default:
 		break;
 	}
+	switch (goal->metric) {
+	case DAMOS_QUOTA_USER_INPUT:
+	case DAMOS_QUOTA_SOME_MEM_PSI_US:
+		return;
+	default:
+		break;
+	}
+	if (goal->complement)
+		goal->current_value = 10000 - goal->current_value;
 }
 
 /* Return the highest score since it makes schemes least aggressive */
-- 
2.47.3

