From: SeongJae Park <sj@kernel.org>
Date: Sun, 1 Dec 2024 12:32:48 -0800
Subject: [PATCH] mm/damon/syfs-schemes: remove damon_sysfs_schemes_region_idx

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs-schemes.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/mm/damon/sysfs-schemes.c b/mm/damon/sysfs-schemes.c
index d94db6cfe929..2d4db29276cd 100644
--- a/mm/damon/sysfs-schemes.c
+++ b/mm/damon/sysfs-schemes.c
@@ -2131,12 +2131,6 @@ void damon_sysfs_schemes_update_stats(
 	}
 }
 
-/*
- * damon_sysfs_schemes that need to update its schemes regions dir.  Protected
- * by damon_sysfs_lock
- */
-static int damon_sysfs_schemes_region_idx;
-
 /**
  * damos_sysfs_populate_region_dir() - Populate a schemes tried region dir.
  *
-- 
2.39.5

