From 27adb5268d37d60bdec3a17cb237be1fbde21c5c Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Sat, 11 Nov 2023 19:36:03 +0000
Subject: [PATCH] mm/damon/core: add debugging-purpose log of tuned esz

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 6ed0f1f9f477..88ccb8776f14 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1255,6 +1255,8 @@ static void damos_set_effective_quota(struct damos_quota *quota)
 		esz = quota->sz;
 
 	quota->esz = esz;
+
+	pr_info("esz %lu\n", esz);
 }
 
 static void damos_adjust_quota(struct damon_ctx *c, struct damos *s)
-- 
2.39.2

