From: SeongJae Park <sj@kernel.org>
Date: Mon, 22 Jun 2026 20:42:41 -0700
Subject: [PATCH] mm/damon/sysfs: support DAMON_FILTER_TYPE_PGIDLE

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index ac84c7570152f..0e4e940cc79c7 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -784,6 +784,10 @@ damon_sysfs_filter_type_names[] = {
 		.type = DAMON_FILTER_TYPE_MEMCG,
 		.name = "memcg",
 	},
+	{
+		.type = DAMON_FILTER_TYPE_PGIDLE,
+		.name = "pgidle",
+	},
 };
 
 static ssize_t type_show(struct kobject *kobj,
-- 
2.47.3

