From: SeongJae Park <sj@kernel.org>
Date: Sun, 11 May 2025 14:11:02 -0700
Subject: [PATCH] mm/damon/stat: fix const discard warning

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/stat.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/mm/damon/stat.c b/mm/damon/stat.c
index f6ecf651c763..10caf7f0a937 100644
--- a/mm/damon/stat.c
+++ b/mm/damon/stat.c
@@ -45,10 +45,10 @@ static unsigned int damon_stat_idletime(const struct damon_region *r)
 
 static int damon_stat_cmp_regions(const void *a, const void *b)
 {
-	const struct damon_region **ra = a;
-	const struct damon_region **rb = b;
+	const struct damon_region *ra = *(const struct damon_region **)a;
+	const struct damon_region *rb = *(const struct damon_region **)b;
 
-	return damon_stat_idletime(*ra) - damon_stat_idletime(*rb);
+	return damon_stat_idletime(ra) - damon_stat_idletime(rb);
 }
 
 static int damon_stat_sort_regions(struct damon_ctx *c,
-- 
2.39.5

