From: SeongJae Park <sj@kernel.org>
Date: Mon, 1 Dec 2025 22:31:37 -0800
Subject: [PATCH] mm/damon/core: fixup damon_access_report->pid caused build
 error

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index ee096a4edce3..5f29e4dee267 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2984,6 +2984,10 @@ static unsigned int kdamond_check_reported_accesses(struct damon_ctx *ctx)
 	struct damon_access_report *report;
 	struct damon_target *t;
 
+	/* currently damon_access_report supports only physical address */
+	if (damon_target_has_pid(ctx))
+		return 0;
+
 	mutex_lock(&damon_access_reports_lock);
 	for (i = 0; i < damon_access_reports_len; i++) {
 		report = &damon_access_reports[i];
@@ -2992,8 +2996,6 @@ static unsigned int kdamond_check_reported_accesses(struct damon_ctx *ctx)
 					usecs_to_jiffies(
 						ctx->attrs.sample_interval)))
 			continue;
-		if (report->pid && !damon_target_has_pid(ctx))
-			continue;
 		damon_for_each_target(t, ctx)
 			kdamond_apply_access_report(report, t, ctx);
 	}
-- 
2.47.3

