From: SeongJae Park <sj@kernel.org>
Date: Sat, 11 Oct 2025 11:57:11 -0700
Subject: [PATCH] mm/damon/sysfs: remove obsolete todo item

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index d3c18cf299c6..34bd1fdb4713 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1753,7 +1753,6 @@ static int damon_sysfs_commit_input(void *data)
 	test_ctx = damon_sysfs_new_test_ctx(kdamond->damon_ctx);
 	if (!test_ctx)
 		return -ENOMEM;
-	/* todo: commit kdamond->damon_ctx to test_ctx */
 	err = damon_commit_ctx(test_ctx, param_ctx);
 	if (err)
 		goto out;
-- 
2.47.3

