From: SJ Park <sj@kernel.org>
Date: Sat, 4 Jul 2026 14:12:58 -0700
Subject: [PATCH] mm/damon/core: rename max_nr_accesses to max_merge_score

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index df8aac6e72054..f1c54b611a0ed 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -4136,7 +4136,7 @@ static unsigned int kdamond_check_reported_accesses(struct damon_ctx *ctx)
 static int kdamond_fn(void *data)
 {
 	struct damon_ctx *ctx = data;
-	unsigned int max_nr_accesses = 0;
+	unsigned int max_merge_score = 0;
 	unsigned long sz_limit = 0;
 
 	pr_debug("kdamond (%d) starts\n", current->pid);
@@ -4186,10 +4186,10 @@ static int kdamond_fn(void *data)
 		if (!access_check_disabled) {
 			/* todo: make these non-exclusive */
 			if (ctx->sample_control.primitives_enabled.page_fault)
-				max_nr_accesses =
+				max_merge_score =
 					kdamond_check_reported_accesses(ctx);
 			else if (ctx->ops.check_accesses)
-				max_nr_accesses = ctx->ops.check_accesses(ctx);
+				max_merge_score = ctx->ops.check_accesses(ctx);
 		}
 
 		if (ctx->ops.apply_probes)
@@ -4199,7 +4199,7 @@ static int kdamond_fn(void *data)
 		if (time_after_eq(ctx->passed_sample_intervals,
 					next_aggregation_sis)) {
 			kdamond_merge_regions(ctx,
-					max_nr_accesses / 10,
+					max_merge_score / 10,
 					sz_limit);
 			/* online updates might be made */
 			sz_limit = damon_apply_min_nr_regions(ctx);
-- 
2.47.3

