From: SeongJae Park <sj@kernel.org>
Date: Sat, 14 Feb 2026 11:22:58 -0800
Subject: [PATCH] mm/damon/test/core-kunit: fixup

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/tests/core-kunit.h | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/mm/damon/tests/core-kunit.h b/mm/damon/tests/core-kunit.h
index fa97fc3d6c40d..547985dbf64cf 100644
--- a/mm/damon/tests/core-kunit.h
+++ b/mm/damon/tests/core-kunit.h
@@ -1241,11 +1241,12 @@ static void damon_test_set_filters_default_reject(struct kunit *test)
 	damos_free_filter(target_filter);
 }
 
-static void damon_test_ensure_sz_limit(struct kunit *test)
+static void damon_test_get_apply_max_region_sz(struct kunit *test)
 {
 	struct damon_ctx *ctx;
 	struct damon_target *t;
 	struct damon_region *r;
+	unsigned long max_region_size;
 
 	ctx = damon_new_ctx();
 	if (!ctx)
@@ -1262,7 +1263,10 @@ static void damon_test_ensure_sz_limit(struct kunit *test)
 		kunit_skip(test, "region alloc fail\n");
 	}
 	damon_add_region(r, t);
-	kdamond_ensure_sz_limit(ctx, 1);
+	ctx->min_region_sz = 1;
+	ctx->attrs.min_nr_regions = 10;
+	max_region_size = kdamond_get_apply_max_region_sz(ctx);
+	KUNIT_EXPECT_EQ(test, max_region_size, 1);
 	KUNIT_EXPECT_EQ(test, damon_nr_regions(t), 10);
 }
 
@@ -1292,7 +1296,7 @@ static struct kunit_case damon_test_cases[] = {
 	KUNIT_CASE(damos_test_filter_out),
 	KUNIT_CASE(damon_test_feed_loop_next_input),
 	KUNIT_CASE(damon_test_set_filters_default_reject),
-	KUNIT_CASE(damon_test_ensure_sz_limit),
+	KUNIT_CASE(damon_test_get_apply_max_region_sz),
 	{},
 };
 
-- 
2.47.3

