From: SeongJae Park <sj@kernel.org>
Date: Sat, 22 Nov 2025 14:03:56 -0800
Subject: [PATCH] mm/damon/core: commit max_stat

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 9262124883f3..a8e72fc89300 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1119,7 +1119,11 @@ static int damos_commit(struct damos *dst, struct damos *src)
 		return err;
 
 	err = damos_commit_filters(dst, src);
-	return err;
+	if (err)
+		return err;
+
+	dst->max_stat = src->max_stat;
+	return 0;
 }
 
 static int damon_commit_schemes(struct damon_ctx *dst, struct damon_ctx *src)
-- 
2.47.3

