From: SeongJae Park <sj@kernel.org>
Date: Sun, 19 Apr 2026 12:46:08 -0700
Subject: [PATCH] mm/damon/core: destroy data_attrs when destroying ctx

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 09b1731210e1a..2e27b628da643 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -837,12 +837,16 @@ void damon_destroy_ctx(struct damon_ctx *ctx)
 {
 	struct damos *s, *next_s;
 	struct damon_sample_filter *f, *next_f;
+	struct damon_data_attr *a, *next_a;
 
 	damon_destroy_targets(ctx);
 
 	damon_for_each_scheme_safe(s, next_s, ctx)
 		damon_destroy_scheme(s);
 
+	damon_for_each_data_attr_safe(a, next_a, ctx)
+		damon_destroy_data_attr(a);
+
 	damon_for_each_sample_filter_safe(f, next_f, &ctx->sample_control)
 		damon_destroy_sample_filter(f, &ctx->sample_control);
 
-- 
2.47.3

