From: SeongJae Park <sj@kernel.org>
Date: Tue, 21 Apr 2026 21:13:22 -0700
Subject: [PATCH] mm/damon: wordsmith test_counters description

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 51f9781bcb78d..9f3250164c063 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -688,7 +688,7 @@ enum damon_ops_id {
  * @update:			Update operations-related data structures.
  * @prepare_access_checks:	Prepare next access check of target regions.
  * @check_accesses:		Check the accesses to target regions.
- * @test_counters:		Run data attribute tests of target regions.
+ * @test_counters:		Run counter tests for each region.
  * @get_scheme_score:		Get the score of a region for a scheme.
  * @apply_scheme:		Apply a DAMON-based operation scheme.
  * @target_valid:		Determine if the target is valid.
-- 
2.47.3

