From: SeongJae Park <sj@kernel.org>
Date: Sat, 18 Apr 2026 12:15:35 -0700
Subject: [PATCH] mm/damon/core: execute damon_operation->check_data_attrs

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 3c355cd53633b..8f4500b120c24 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3472,6 +3472,8 @@ static int kdamond_fn(void *data)
 			max_nr_accesses = kdamond_check_reported_accesses(ctx);
 		else if (ctx->ops.check_accesses)
 			max_nr_accesses = ctx->ops.check_accesses(ctx);
+		if (ctx->ops.check_data_attrs && ctx->attrs.data_attrs.prev)
+			ctx->ops.check_data_attrs(ctx);
 
 		if (time_after_eq(ctx->passed_sample_intervals,
 					next_aggregation_sis)) {
-- 
2.47.3

