From: SeongJae Park <sj@kernel.org>
Date: Tue, 18 Feb 2025 21:06:47 -0800
Subject: [PATCH] mm/damon/core: change auto-tuning goal from samples ratio to
 heats ratio

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 5aefed309e3a..629603d4b2f6 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1427,19 +1427,20 @@ static unsigned long damon_get_intervals_adaptation_bp(struct damon_ctx *c)
 {
 	struct damon_target *t;
 	struct damon_region *r;
-	unsigned long nr_regions = 0, access_samples = 0;
+	unsigned long sz_regions = 0, heats = 0;
 	struct damon_intervals_goal *goal = &c->attrs.intervals_goal;
-	unsigned long max_samples, target_samples, score_bp;
+	unsigned long max_heats, target_heats, score_bp;
 	unsigned long adaptation_bp;
 
 	damon_for_each_target(t, c) {
-		nr_regions = damon_nr_regions(t);
-		damon_for_each_region(r, t)
-			access_samples += r->nr_accesses;
+		damon_for_each_region(r, t) {
+			sz_regions += r->ar.end - r->ar.start;
+			heats += (r->ar.end - r->ar.start) * r->nr_accesses;
+		}
 	}
-	max_samples = nr_regions * c->attrs.aggr_samples;
-	target_samples = max_samples * goal->samples_bp / 10000;
-	score_bp = access_samples * 10000 / target_samples;
+	max_heats = sz_regions * c->attrs.aggr_samples;
+	target_heats = max_heats * goal->samples_bp / 10000;
+	score_bp = heats * 10000 / target_heats;
 	adaptation_bp = damon_feed_loop_next_input(100000000, score_bp) /
 		10000;
 	/*
@@ -1449,9 +1450,8 @@ static unsigned long damon_get_intervals_adaptation_bp(struct damon_ctx *c)
 	if (adaptation_bp <= 10000)
 		adaptation_bp = 5000 + adaptation_bp / 2;
 
-	pr_info("access_samples %lu/%lu, score_bp %lu, adaptation bp %lu\n",
-			access_samples, target_samples, score_bp,
-			adaptation_bp);
+	pr_info("heats %lu/%lu, score_bp %lu, adaptation bp %lu\n",
+			heats, target_heats, score_bp, adaptation_bp);
 	return adaptation_bp;
 }
 
-- 
2.39.5

