From: SeongJae Park <sj@kernel.org>
Date: Sat, 6 Sep 2025 18:42:36 -0700
Subject: [PATCH] mm/damon/tests/core-kunit: test node_mem_used_bp

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/tests/core-kunit.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/mm/damon/tests/core-kunit.h b/mm/damon/tests/core-kunit.h
index 3513ea394a5e..1430664ac67f 100644
--- a/mm/damon/tests/core-kunit.h
+++ b/mm/damon/tests/core-kunit.h
@@ -485,6 +485,14 @@ static void damos_test_commit_quota_goal(struct kunit *test)
 	src->current_value = 789;
 	damos_commit_quota_goal(dst, src);
 	damos_test_commit_quota_goal_ensure_committed(test, dst, src, 456);
+
+	src->metric = DAMOS_QUOTA_NODE_MEM_FREE_BP;
+	src->target_value = 12;
+	src->current_value = 345;
+	src->nid = 3;
+	damos_commit_quota_goal(dst, src);
+	damos_test_commit_quota_goal_ensure_committed(test, dst, src, 0);
+
 	damos_destroy_quota_goal(dst);
 	damos_destroy_quota_goal(src);
 }
-- 
2.39.5

