From: SJ Park <sj@kernel.org>
Date: Sat, 15 Aug 2026 10:30:54 -0700
Subject: [PATCH] mm/damon/core: use abs_diff() in damon_feed_loop_next_input()

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 8ed24d093704d..5cf8202965ef6 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3098,10 +3098,7 @@ static unsigned long damon_feed_loop_next_input(unsigned long last_input,
 	if (score >= goal * 2)
 		return min_input;
 
-	if (over_achieving)
-		score_goal_diff = score - goal;
-	else
-		score_goal_diff = goal - score;
+	score_goal_diff = abs_diff(score, goal);
 
 	if (last_input < ULONG_MAX / score_goal_diff)
 		compensation = last_input * score_goal_diff / goal;
-- 
2.47.3

