From: SeongJae Park <sj@kernel.org>
Date: Wed, 8 Oct 2025 17:52:43 -0700
Subject: [PATCH] mm/damon: add damon_target->remove_on_commit

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 3 +++
 mm/damon/core.c       | 1 +
 2 files changed, 4 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 524dea87cac7..f14c305ed889 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -103,6 +103,9 @@ struct damon_target {
 	unsigned int nr_regions;
 	struct list_head regions_list;
 	struct list_head list;
+/* private: Interval use only */
+	/* damon_commit_targets() will remove matching dst target */
+	bool remove_on_commit;
 };
 
 /**
diff --git a/mm/damon/core.c b/mm/damon/core.c
index 1a501ed371bc..5a6ddf45c97b 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -492,6 +492,7 @@ struct damon_target *damon_new_target(void)
 	t->nr_regions = 0;
 	INIT_LIST_HEAD(&t->regions_list);
 	INIT_LIST_HEAD(&t->list);
+	t->remove_on_commit = false;
 
 	return t;
 }
-- 
2.47.3

