From: SeongJae Park <sj@kernel.org>
Date: Sun, 22 Mar 2026 16:00:14 -0700
Subject: [PATCH] mm/damon/core: call cleanup_fn()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c  | 3 +++
 mm/damon/sysfs.c | 1 +
 2 files changed, 4 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 526339e8d9983..f076d491b309a 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3206,6 +3206,9 @@ static void kdamond_call(struct damon_ctx *ctx, bool cancel)
 		else
 			control->return_code = control->fn(control->data);
 
+		if ((!control->repeat || cancel) && control->cleanup_fn)
+			control->cleanup_fn(control);
+
 		if (!control->repeat)
 			complete(&control->completion);
 		else if (control->canceled && control->dealloc_on_cancel)
diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index ed3fc4efe7e6d..ad6b9bf50fcaf 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -2325,6 +2325,7 @@ static int damon_sysfs_turn_damon_on(struct damon_sysfs_kdamond *kdamond)
 	repeat_call_control->data = kdamond;
 	repeat_call_control->repeat = true;
 	repeat_call_control->dealloc_on_cancel = true;
+	repeat_call_control->cleanup_fn = NULL;
 	if (damon_call(ctx, repeat_call_control))
 		kfree(repeat_call_control);
 	return err;
-- 
2.47.3

