From: SeongJae Park <sj@kernel.org>
Date: Sun, 8 Mar 2026 14:36:53 -0700
Subject: [PATCH] mm/damon/core: verify regions right after merge operation

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index cdf9c5afa1f9a..e793d408eeca2 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3043,6 +3043,32 @@ static void kdamond_merge_regions(struct damon_ctx *c, unsigned int threshold,
 			threshold / 2 < max_thres);
 }
 
+#ifdef CONFIG_DAMON_DEBUG_SANITY
+static void damon_verify_merged_regions(struct damon_ctx *c)
+{
+	struct damon_target *t;
+	struct damon_region *r;
+
+	damon_for_each_target(t, c) {
+		unsigned int nr_regions = 0;
+
+		damon_for_each_region(r, t) {
+			WARN_ONCE(!damon_sz_region(r),
+					"empty region [%lu,%lu)\n",
+					r->ar.start, r->ar.end);
+			nr_regions++;
+		}
+		WARN_ONCE(damon_nr_regions(t) != nr_regions,
+				"nr_regions mismatch: %u != %u\n",
+				damon_nr_regions(t), nr_regions);
+	}
+}
+#else
+static void damon_verify_merged_regions(struct damon_ctx *c)
+{
+}
+#endif
+
 #ifdef CONFIG_DAMON_DEBUG_SANITY
 static void damon_verify_split_region_at(struct damon_region *r,
 		unsigned long sz_r)
@@ -3492,6 +3518,7 @@ static int kdamond_fn(void *data)
 					sz_limit);
 			/* online updates might be made */
 			sz_limit = damon_apply_min_nr_regions(ctx);
+			damon_verify_merged_regions(ctx);
 		}
 
 		/*
-- 
2.47.3

