From: SeongJae Park <sj@kernel.org>
Date: Wed, 24 Jun 2026 20:29:19 -0700
Subject: [PATCH] mm/damon/core: commit damon_probe->weight

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index d0c80f9560df1..f283a404528b9 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1793,6 +1793,7 @@ static int damon_commit_probe(struct damon_probe *dst, struct damon_probe *src)
 {
 	int err;
 
+	dst->weight = src->weight;
 	err = damon_commit_preps(dst, src);
 	if (err)
 		return err;
-- 
2.47.3

