From: SeongJae Park <sj@kernel.org>
Date: Mon, 15 Jun 2026 21:37:46 -0700
Subject: [PATCH] mm/damon: add damon_ctx->preps

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 mm/damon/core.c       | 1 +
 2 files changed, 3 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index e92bdbeea0eb4..6297e3e504e29 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -1058,6 +1058,8 @@ struct damon_ctx {
 	struct damon_operations ops;
 	/* Head of monitoring targets (&damon_target) list. */
 	struct list_head adaptive_targets;
+	/* Head of damon_prep list. */
+	struct list_head preps;
 	/* Head of probes (&damon_probe) list. */
 	struct list_head probes;
 	struct damon_sample_control sample_control;
diff --git a/mm/damon/core.c b/mm/damon/core.c
index 5fdbf71263c70..2057c88019dc4 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -867,6 +867,7 @@ struct damon_ctx *damon_new_ctx(void)
 	ctx->attrs.max_nr_regions = 1000;
 
 	INIT_LIST_HEAD(&ctx->probes);
+	INIT_LIST_HEAD(&ctx->preps);
 
 	ctx->sample_control.primitives_enabled.page_table = true;
 	INIT_LIST_HEAD(&ctx->sample_control.sample_filters);
-- 
2.47.3

