From: SeongJae Park <sj@kernel.org>
Date: Sun, 11 May 2025 13:57:23 -0700
Subject: [PATCH] mm/damon/stat: handle boot time enabled case

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/stat.c | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/mm/damon/stat.c b/mm/damon/stat.c
index 8f61faf6fe42..0577b15c6e3a 100644
--- a/mm/damon/stat.c
+++ b/mm/damon/stat.c
@@ -185,6 +185,8 @@ static void damon_stat_stop(void)
 	damon_destroy_ctx(damon_stat_context);
 }
 
+static bool damon_stat_init_called = false;
+
 static int damon_stat_enable_store(
 		const char *val, const struct kernel_param *kp)
 {
@@ -198,6 +200,13 @@ static int damon_stat_enable_store(
 	if (enable == enabled)
 		return 0;
 
+	if (!damon_stat_init_called)
+		/*
+		 * probably called from command line parsing (parse_args()).
+		 * Cannot call damon_new_ctx().  Let damon_stat_init() handle.
+		 */
+		return 0;
+
 	if (enable)
 		return damon_stat_start();
 	damon_stat_stop();
@@ -206,7 +215,14 @@ static int damon_stat_enable_store(
 
 static int __init damon_stat_init(void)
 {
-	return 0;
+	int err = 0;
+
+	damon_stat_init_called = true;
+
+	/* probably set via command line */
+	if (enable)
+		err = damon_stat_start();
+	return err;
 }
 
 module_init(damon_stat_init);
-- 
2.39.5

