From: SeongJae Park <sj@kernel.org>
Date: Tue, 10 Dec 2024 17:20:25 -0800
Subject: [PATCH] mm/damon/sysfs: implement damon_call() wrapper

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 4daac92be30b..8c8c1e747092 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1525,6 +1525,18 @@ static int damon_sysfs_turn_damon_off(struct damon_sysfs_kdamond *kdamond)
 	 */
 }
 
+static int damon_sysfs_damon_call(int (*fn)(void *data),
+		struct damon_sysfs_kdamond *kdamond)
+{
+	struct damon_call_control call_control = {};
+
+	if (!kdamond->damon_ctx)
+		return -EINVAL;
+	call_control.fn = fn;
+	call_control.data = kdamond;
+	return damon_call(kdamond->damon_ctx, &call_control);
+}
+
 /*
  * damon_sysfs_handle_cmd() - Handle a command for a specific kdamond.
  * @cmd:	The command to handle.
-- 
2.39.5

