From: SJ Park <sj@kernel.org>
Date: Sat, 8 Aug 2026 09:53:56 -0700
Subject: [PATCH] mm/damon/Kconfig: add CONFIG_DAMON_AUTOTUNE_PARAMS

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/Kconfig | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/mm/damon/Kconfig b/mm/damon/Kconfig
index 34631a44cdecb..a927a97a3d62a 100644
--- a/mm/damon/Kconfig
+++ b/mm/damon/Kconfig
@@ -12,6 +12,17 @@ config DAMON
 	  See https://www.kernel.org/doc/html/latest/mm/damon/index.html for
 	  more information.
 
+config DAMON_AUTOTUNE_PARAMS
+	bool "Auto-tune DAMON Parameters"
+	depends on DAMON
+	help
+	  This enables DAMON parameters auto-tuning features by default.  This
+	  only changes the default parameter values to enable auto-tuning
+	  features.  Users can manually turn off auto-tuning if needed in
+	  runtime.
+
+	  If unsure, say Y.
+
 config DAMON_DEBUG_SANITY
 	bool "Check sanity of DAMON code"
 	depends on DAMON
-- 
2.47.3

