From: SeongJae Park <sj@kernel.org>
Date: Sun, 3 Sep 2023 05:02:44 +0000
Subject: [PATCH] mm/damon/core: add todo for DAMOS interval validation

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 0c161783916c..6183b4fc153e 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2726,6 +2726,7 @@ static void kdamond_init_ctx(struct damon_ctx *ctx)
 	ctx->next_intervals_tune_sis = ctx->next_aggregation_sis *
 		ctx->attrs.intervals_goal.aggrs;
 
+	/* todo: ensure apply_interval_us > sample_interval */
 	damon_for_each_scheme(scheme, ctx) {
 		apply_interval = scheme->apply_interval_us ?
 			scheme->apply_interval_us : ctx->attrs.aggr_interval;
-- 
2.47.3

