From: SeongJae Park <sj@kernel.org>
Date: Sat, 6 Sep 2025 19:26:29 -0700
Subject: [PATCH] mm/damon/core: use damos_commit_quota_goal() for new goal
 commit

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index b81c16b42a78..3329fb1f2c44 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -868,7 +868,7 @@ int damos_commit_quota_goals(struct damos_quota *dst, struct damos_quota *src)
 				src_goal->metric, src_goal->target_value);
 		if (!new_goal)
 			return -ENOMEM;
-		damos_commit_quota_goal_union(new_goal, src_goal);
+		damos_commit_quota_goal(new_goal, src_goal);
 		damos_add_quota_goal(dst, new_goal);
 	}
 	return 0;
-- 
2.47.3

