From: SeongJae Park <sj@kernel.org>
Date: Mon, 15 Jun 2026 21:37:46 -0700
Subject: [PATCH] mm/damon: add damon_ctx->preps

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 35b1b4e6f0dbc..f0ee7e5082fef 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -1061,6 +1061,8 @@ struct damon_ctx {
 	struct damon_operations ops;
 	/* Head of monitoring targets (&damon_target) list. */
 	struct list_head adaptive_targets;
+	/* Head of damon_prep list. */
+	struct list_head preps;
 	/* Head of probes (&damon_probe) list. */
 	struct list_head probes;
 	struct damon_sample_control sample_control;
-- 
2.47.3

