From: SeongJae Park <sj@kernel.org>
Date: Fri, 22 Nov 2024 14:41:15 -0800
Subject: [PATCH] mm/damon: more documentation

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index cff7e4e6e84c..745446740d17 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -366,6 +366,13 @@ struct damos_filter {
 struct damon_ctx;
 struct damos;
 
+/**
+ * struct damos_walk_control - Control walking DAMOS target regions.
+ *
+ * @arg:	Argument that will be passed to walk functions.
+ * @prep_fn:	Function to be called back once before walk_fn.
+ * @walk_fn:	Function to be called back for each region.
+ */
 struct damos_walk_control {
 	void *arg;
 	int (*prep_fn)(void *arg, struct damon_ctx *ctx);
@@ -373,7 +380,9 @@ struct damos_walk_control {
 			struct damon_target *t, struct damon_region *r,
 			struct damos *s);
 /* private: internal use only */
+	/* for waiting on walk completion from DAMON worker thrad. */
 	struct completion completion;
+	/* for saving if the walk is canceled. */
 	bool canceled;
 };
 
@@ -691,9 +700,11 @@ struct damon_ctx {
 	/* for scheme quotas prioritization */
 	unsigned long *regions_score_histogram;
 
+	/* for damon_call() */
 	struct damon_call_control *call_control;
 	struct mutex call_control_lock;
 
+	/* for damos_walk() */
 	struct damos_walk_control *walk_control;
 	struct mutex walk_control_lock;
 
-- 
2.39.5

