From: SeongJae Park <sj@kernel.org>
Date: Fri, 17 Apr 2026 07:33:56 -0700
Subject: [PATCH] mm/damon/reclaim: cleanup kdamond_pid detection

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/reclaim.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/mm/damon/reclaim.c b/mm/damon/reclaim.c
index 775b33dbc952f..68ac387711563 100644
--- a/mm/damon/reclaim.c
+++ b/mm/damon/reclaim.c
@@ -422,14 +422,12 @@ MODULE_PARM_DESC(enabled,
 static int damon_reclaim_kdamond_pid_load(char *buffer,
 		const struct kernel_param *kp)
 {
-	int kdamond_pid;
+	int kdamond_pid = -1;;
 
 	if (ctx) {
 		kdamond_pid = damon_kdamond_pid(ctx);
 		if (kdamond_pid < 0)
 			kdamond_pid = -1;
-	} else {
-		kdamond_pid = -1;
 	}
 	return sprintf(buffer, "%d\n", kdamond_pid);
 }
-- 
2.47.3

