From: SeongJae Park <sj@kernel.org>
Date: Sun, 19 Apr 2026 10:14:10 -0700
Subject: [PATCH] mm/damon/sysfs: fixup data_attr dir

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 36e65ef80d237..aa105ba0a8236 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1032,6 +1032,7 @@ static int damon_sysfs_data_attr_add_dirs(struct damon_sysfs_data_attr *attr)
 	filters = damon_sysfs_filters_alloc();
 	if (!filters)
 		return -ENOMEM;
+	attr->filters = filters;
 
 	err = kobject_init_and_add(&filters->kobj, &damon_sysfs_filters_ktype,
 			&attr->kobj, "filters");
@@ -1044,8 +1045,10 @@ static int damon_sysfs_data_attr_add_dirs(struct damon_sysfs_data_attr *attr)
 
 static void damon_sysfs_data_attr_rm_dirs(struct damon_sysfs_data_attr *attr)
 {
-	if (attr->filters)
+	if (attr->filters) {
+		damon_sysfs_filters_rm_dirs(attr->filters);
 		kobject_put(&attr->filters->kobj);
+	}
 }
 
 static void damon_sysfs_data_attr_release(struct kobject *kobj)
-- 
2.47.3

