From: SJ Park <sj@kernel.org>
Date: Mon, 13 Jul 2026 17:50:36 -0700
Subject: [PATCH] mm/damon/core: hide core-only fields for damon_ctx

Signed-off-by: SJ Park <sj@kernel.org>
---
 include/linux/damon.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index a0e7e83a3b55c..f223acee70728 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -846,8 +846,6 @@ struct damon_attrs {
  * thread other than the kdamond should be made using safe DAMON APIs,
  * including damon_call() and damos_walk().
  *
- * @ops:	Set of monitoring operations for given use cases.
- * @probes:	Head of probes (&damon_probe) list.
  * @addr_unit:	Scale factor for core to ops address conversion.
  * @min_region_sz:	Minimum region size.
  * @pause:	Pause kdamond main loop.
@@ -899,15 +897,17 @@ struct damon_ctx {
 	struct mutex kdamond_lock;
 
 /* public: */
-	struct damon_operations ops;
-	struct list_head probes;
 	unsigned long addr_unit;
 	unsigned long min_region_sz;
 	bool pause;
 
 /* private: */
+	/* Set of monitoring operations for given use cases. */
+	struct damon_operations ops;
 	/* Head of monitoring targets (&damon_target) list. */
 	struct list_head adaptive_targets;
+	/* Head of probes (&damon_probe) list. */
+	struct list_head probes;
 	/* Head of schemes (&damos) list. */
 	struct list_head schemes;
 
-- 
2.47.3

