From: SeongJae Park <sj@kernel.org>
Date: Sun, 22 Jun 2025 12:26:36 -0700
Subject: [PATCH] mm/damon: add INACTIVE_MEM_BP quota goal

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 13159897b779..90f3b98a2aa4 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -176,6 +176,7 @@ enum damos_action {
  * @DAMOS_QUOTA_NODE_MEM_USED_BP:	MemUsed ratio of a node.
  * @DAMOS_QUOTA_NODE_MEM_FREE_BP:	MemFree ratio of a node.
  * @DAMOS_QUOTA_ACTIVE_MEM_BP:		Active to [in]active memory ratio.
+ * @DAMOS_QUOTA_INACTIVE_MEM_BP:	Inactive to [in]active memory ratio.
  * @NR_DAMOS_QUOTA_GOAL_METRICS:	Number of DAMOS quota goal metrics.
  *
  * Metrics equal to larger than @NR_DAMOS_QUOTA_GOAL_METRICS are unsupported.
@@ -186,6 +187,7 @@ enum damos_quota_goal_metric {
 	DAMOS_QUOTA_NODE_MEM_USED_BP,
 	DAMOS_QUOTA_NODE_MEM_FREE_BP,
 	DAMOS_QUOTA_ACTIVE_MEM_BP,
+	DAMOS_QUOTA_INACTIVE_MEM_BP,
 	NR_DAMOS_QUOTA_GOAL_METRICS,
 };
 
-- 
2.39.5

