From: SJ Park <sj@kernel.org>
Date: Thu, 6 Aug 2026 07:56:36 -0700
Subject: [PATCH] mm/damon/vaddr: fixup compile warning on !HUGETLB

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/vaddr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/vaddr.c b/mm/damon/vaddr.c
index db58dd3701a20..f05fe45b3425f 100644
--- a/mm/damon/vaddr.c
+++ b/mm/damon/vaddr.c
@@ -596,7 +596,7 @@ static int damon_va_probe_hugetlb_entry(pte_t *pte, unsigned long hmask,
 	return 0;
 }
 #else
-#define damon_young_hugetlb_entry NULL
+#define damon_va_probe_hugetlb_entry NULL
 #endif /* CONFIG_HUGETLB_PAGE */
 
 static void __damon_va_apply_probes(struct damon_ctx *ctx,
-- 
2.47.3

