From: SeongJae Park <sj@kernel.org>
Date: Tue, 21 Jan 2025 17:52:00 -0800
Subject: [PATCH] mm/damon/sysfs: supporting reading tuned aggregation interval

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index a53413ea406b..8bc5e3e40893 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1133,6 +1133,11 @@ enum damon_sysfs_cmd {
 	 * effective size quota of the scheme in bytes.
 	 */
 	DAMON_SYSFS_CMD_UPDATE_SCHEMES_EFFECTIVE_QUOTAS,
+	/*
+	 * @DAMON_SYSFS_CMD_UPDATE_TUNED_AGGR_US: Update the tuned aggregation
+	 * interval in microseconds.
+	 */
+	DAMON_SYSFS_CMD_UPDATE_TUNED_AGGR_US,
 	/*
 	 * @NR_DAMON_SYSFS_CMDS: Total number of DAMON sysfs commands.
 	 */
@@ -1150,6 +1155,7 @@ static const char * const damon_sysfs_cmd_strs[] = {
 	"update_schemes_tried_regions",
 	"clear_schemes_tried_regions",
 	"update_schemes_effective_quotas",
+	"update_tuned_aggr_us",
 };
 
 /*
@@ -1427,6 +1433,16 @@ static int damon_sysfs_upd_schemes_effective_quotas(void *data)
 	return 0;
 }
 
+static int damon_sysfs_upd_tuned_aggr_us(void *data)
+{
+	struct damon_sysfs_kdamond *kdamond = data;
+	struct damon_ctx *ctx = kdamond->damon_ctx;
+
+	kdamond->contexts->contexts_arr[0]->attrs->intervals->max_aggr_us =
+		ctx->attrs.max_aggr_interval;
+	return 0;
+}
+
 
 /*
  * damon_sysfs_cmd_request_callback() - DAMON callback for handling requests.
@@ -1648,6 +1664,9 @@ static int damon_sysfs_handle_cmd(enum damon_sysfs_cmd cmd,
 		return damon_sysfs_damon_call(
 				damon_sysfs_upd_schemes_effective_quotas,
 				kdamond);
+	case DAMON_SYSFS_CMD_UPDATE_TUNED_AGGR_US:
+		return damon_sysfs_damon_call(
+				damon_sysfs_upd_tuned_aggr_us, kdamond);
 	default:
 		break;
 	}
-- 
2.39.5

