From: SeongJae Park <sj@kernel.org>
Date: Sat, 23 May 2026 13:46:17 -0700
Subject: [PATCH] mm/damon: filters and list of damon_probe as private

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 4c403058bce3b..c6b54bc8495c7 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -825,12 +825,12 @@ struct damon_filter {
 
 /**
  * struct damon_probe - Data region attribute probe.
- *
- * @filters:	Filters for assessing if a given region is for this probe.
- * @list:	Siblings list.
  */
 struct damon_probe {
+/* private: */
+	/* Filters for assessing if a given region is for this probe. */
 	struct list_head filters;
+	/* Siblings list. */
 	struct list_head list;
 };
 
-- 
2.47.3

