From: SeongJae Park <sj@kernel.org>
Date: Sun, 30 Nov 2025 13:33:07 -0800
Subject: [PATCH] mm/damon: remove operations_attr struct

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 22 ----------------------
 1 file changed, 22 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 1bf0d1ea43a6..287d84cf1c37 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -617,24 +617,6 @@ enum damon_ops_id {
 	NR_DAMON_OPS,
 };
 
-/**
- *
- * struct damon_operations_attrs - Monitoring operations control attributes.
- *
- * @use_reports:	Whether to use damon_report_access()-ed information.
- * @write_only:		Ignore none-write accesses.
- * @cpus:		Ignore accesses from CPUs not in this mask.
- * @tids:		Array of pids to monitor accesses for.
- * @nr_tids:		Size of @tids.
- */
-struct damon_operations_attrs {
-	bool use_reports;
-	bool write_only;
-	cpumask_t cpus;
-	pid_t *tids;
-	int nr_tids;
-};
-
 /**
  * struct damon_operations - Monitoring operations for given use cases.
  *
@@ -692,9 +674,6 @@ struct damon_operations {
 	void (*update)(struct damon_ctx *context);
 	void (*prepare_access_checks)(struct damon_ctx *context);
 	unsigned int (*check_accesses)(struct damon_ctx *context);
-	bool (*eligible_report)(struct damon_access_report *report,
-			struct damon_target *t,
-			struct damon_operations_attrs *ops_attrs);
 	int (*get_scheme_score)(struct damon_ctx *context,
 			struct damon_target *t, struct damon_region *r,
 			struct damos *scheme);
@@ -926,7 +905,6 @@ struct damon_ctx {
 	struct mutex kdamond_lock;
 
 	struct damon_operations ops;
-	struct damon_operations_attrs ops_attrs;
 	struct damon_sample_control sample_control;
 	unsigned long addr_unit;
 	unsigned long min_sz_region;
-- 
2.47.3

