From: SJ Park <sj@kernel.org>
Date: Tue, 23 Jun 2026 21:03:23 -0700
Subject: [PATCH] mm/damon: add damon_ctx->preps

Signed-off-by: SJ Park <sj@kernel.org>
---
 include/linux/damon.h | 1 +
 mm/damon/core.c       | 1 +
 2 files changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 212d136244725..3d4fdcb8c5a5d 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -1062,6 +1062,7 @@ struct damon_ctx {
 	struct damon_operations ops;
 	/* Head of monitoring targets (&damon_target) list. */
 	struct list_head adaptive_targets;
+	struct list_head preps;
 	/* Head of probes (&damon_probe) list. */
 	struct list_head probes;
 	struct damon_sample_control sample_control;
diff --git a/mm/damon/core.c b/mm/damon/core.c
index 0e7e25690952a..93de5d1a9598b 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -929,6 +929,7 @@ struct damon_ctx *damon_new_ctx(void)
 	ctx->attrs.max_nr_regions = 1000;
 
 	INIT_LIST_HEAD(&ctx->probes);
+	INIT_LIST_HEAD(&ctx->preps);
 
 	ctx->sample_control.primitives_enabled.page_table = true;
 	INIT_LIST_HEAD(&ctx->sample_control.sample_filters);
-- 
2.47.3

