From: SJ Park <sj@kernel.org>
Date: Sun, 12 Jul 2026 10:25:51 -0700
Subject: [PATCH] mm/damon/vaddr: remove same_target param (patch 3 fixup)

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/vaddr.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/mm/damon/vaddr.c b/mm/damon/vaddr.c
index 78e497be36495..011a64bb566b2 100644
--- a/mm/damon/vaddr.c
+++ b/mm/damon/vaddr.c
@@ -478,7 +478,7 @@ static bool damon_va_young(struct mm_struct *mm, unsigned long addr)
  * r	the region to be checked
  */
 static void __damon_va_check_access(struct mm_struct *mm,
-				struct damon_region *r, bool same_target)
+				struct damon_region *r)
 {
 	bool accessed;
 
@@ -497,15 +497,12 @@ static unsigned int damon_va_check_accesses(struct damon_ctx *ctx)
 	struct mm_struct *mm;
 	struct damon_region *r;
 	unsigned int max_nr_accesses = 0;
-	bool same_target;
 
 	damon_for_each_target(t, ctx) {
 		mm = damon_get_mm(t);
-		same_target = false;
 		damon_for_each_region(r, t) {
-			__damon_va_check_access(mm, r, same_target);
+			__damon_va_check_access(mm, r);
 			max_nr_accesses = max(r->nr_accesses, max_nr_accesses);
-			same_target = true;
 		}
 		if (mm)
 			mmput(mm);
-- 
2.47.3

