From: SeongJae Park <sj@kernel.org>
Date: Fri, 23 May 2025 11:06:29 -0700
Subject: [PATCH] mm/damon/core: add an hacking idea concept interface
 prototype

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index cd3e06b2e844..ed92a8b9da9a 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3118,6 +3118,25 @@ static int __init damon_init(void)
 	return 0;
 }
 
+#ifdef DAMON_HACK_IDEA
+/**
+ * damos_search() - Return damon_region of a given address and type.
+ * @addr:	the address of the region to search for.
+ * @pid:	PID of the address space.
+ *
+ * Search and return damon_region of a given address in a given address space.
+ * The caller could read returned damon_region to get its address pattern.
+ *
+ * If @pid is NULL, the physical address space will be searched.
+ *
+ * Return: damon_region covering the memory of the given address.
+ */
+struct damon_region *damon_search(unsigned long addr)
+{
+	return NULL;
+}
+#endif
+
 subsys_initcall(damon_init);
 
 #include "tests/core-kunit.h"
-- 
2.39.5

