From: SeongJae Park <sj@kernel.org>
Date: Sat, 18 Apr 2026 15:13:06 -0700
Subject: [PATCH] mm/damon/sysfs: connect data_attrs to monitoring_attrs

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 4ca54f8017146..36e65ef80d237 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1199,6 +1199,7 @@ struct damon_sysfs_attrs {
 	struct kobject kobj;
 	struct damon_sysfs_intervals *intervals;
 	struct damon_sysfs_ul_range *nr_regions_range;
+	struct damon_sysfs_data_attrs *data_attrs;
 	struct damon_sysfs_sample *sample;
 };
 
@@ -1216,6 +1217,7 @@ static int damon_sysfs_attrs_add_dirs(struct damon_sysfs_attrs *attrs)
 {
 	struct damon_sysfs_intervals *intervals;
 	struct damon_sysfs_ul_range *nr_regions_range;
+	struct damon_sysfs_data_attrs *data_attrs;
 	struct damon_sysfs_sample *sample;
 	int err;
 
@@ -1246,6 +1248,18 @@ static int damon_sysfs_attrs_add_dirs(struct damon_sysfs_attrs *attrs)
 		goto put_nr_regions_intervals_out;
 	attrs->nr_regions_range = nr_regions_range;
 
+	data_attrs = damon_sysfs_data_attrs_alloc();
+	if (!data_attrs) {
+		err = -ENOMEM;
+		goto put_nr_regions_intervals_out;
+	}
+	err = kobject_init_and_add(&data_attrs->kobj,
+			&damon_sysfs_data_attrs_ktype, &attrs->kobj,
+			"data_attrs");
+	if (err)
+		goto put_data_attrs_out;
+	attrs->data_attrs = data_attrs;
+
 	sample = damon_sysfs_sample_alloc();
 	if (!sample) {
 		err = -ENOMEM;
@@ -1265,6 +1279,9 @@ static int damon_sysfs_attrs_add_dirs(struct damon_sysfs_attrs *attrs)
 put_sample_out:
 	kobject_put(&sample->kobj);
 	attrs->sample = NULL;
+put_data_attrs_out:
+	kobject_put(&data_attrs->kobj);
+	attrs->data_attrs = NULL;
 put_nr_regions_intervals_out:
 	kobject_put(&nr_regions_range->kobj);
 	attrs->nr_regions_range = NULL;
@@ -1281,6 +1298,8 @@ static void damon_sysfs_attrs_rm_dirs(struct damon_sysfs_attrs *attrs)
 	kobject_put(&attrs->nr_regions_range->kobj);
 	damon_sysfs_intervals_rm_dirs(attrs->intervals);
 	kobject_put(&attrs->intervals->kobj);
+	damon_sysfs_data_attrs_rm_dirs(attrs->data_attrs);
+	kobject_put(&attrs->data_attrs->kobj);
 	damon_sysfs_sample_rm_dirs(attrs->sample);
 	kobject_put(&attrs->sample->kobj);
 }
-- 
2.47.3

