From: SeongJae Park <sj@kernel.org>
Date: Mon, 16 Feb 2026 14:44:20 -0800
Subject: [PATCH] mm/damon/core: make change minimum

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index af4e70abad392..fd1b2cbfe2c80 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1312,7 +1312,6 @@ static unsigned long damon_region_sz_limit(struct damon_ctx *ctx)
 		sz /= ctx->attrs.min_nr_regions;
 	if (sz < ctx->min_region_sz)
 		sz = ctx->min_region_sz;
-	sz = ALIGN(sz, ctx->min_region_sz);
 
 	return sz;
 }
@@ -1339,6 +1338,7 @@ static unsigned long damon_apply_min_nr_regions(struct damon_ctx *ctx)
 	struct damon_target *t;
 	struct damon_region *r, *next;
 
+	max_region_sz = ALIGN(max_region_sz, ctx->min_region_sz);
 	damon_for_each_target(t, ctx) {
 		damon_for_each_region_safe(r, next, t) {
 			while (damon_sz_region(r) > max_region_sz) {
@@ -2894,7 +2894,7 @@ static int kdamond_fn(void *data)
 				sample_interval;
 			if (ctx->ops.update)
 				ctx->ops.update(ctx);
-			sz_limit = damon_apply_min_nr_regions(ctx);
+			sz_limit = damon_region_sz_limit(ctx);
 		}
 	}
 done:
-- 
2.47.3

