From: SeongJae Park <sj@kernel.org>
Date: Sat, 3 Jan 2026 10:27:51 -0800
Subject: [PATCH] mm/damon/core: add damon_merge_regions_of() hardening

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 24 +++++++++++++++++-------
 1 file changed, 17 insertions(+), 7 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 5eef07c31fdc9..0d6bd82f46e88 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2743,6 +2743,22 @@ static void damon_merge_two_regions(struct damon_target *t,
 	damon_destroy_region(r, t);
 }
 
+#ifdef CONFIG_DAMON_HARDENED
+static void damon_verify_merge_regions_of(struct damon_region *r)
+{
+	if (r->nr_accesses == r->nr_accesses_bp / 10000)
+		return;
+	pr_err("nr_accesses (%u) != nr_accesses_bp (%u)\n",
+			r->nr_accesses, r->nr_accesses_bp);
+	BUG();
+}
+#else
+static void damon_verify_merge_regions_of(struct damon_region *r)
+{
+}
+#endif
+
+
 /*
  * Merge adjacent regions having similar access frequencies
  *
@@ -2762,13 +2778,7 @@ static void damon_merge_regions_of(struct damon_target *t, unsigned int thres,
 			r->age = 0;
 		else
 			r->age++;
-
-		if (r->nr_accesses != r->nr_accesses_bp / 10000) {
-			pr_err("nr_accesses (%u) != nr_accesses_bp (%u)\n",
-					r->nr_accesses, r->nr_accesses_bp);
-			BUG();
-		}
-
+		damon_verify_merge_regions_of(r);
 		if (prev && prev->ar.end == r->ar.start &&
 		    abs(prev->nr_accesses - r->nr_accesses) <= thres &&
 		    damon_sz_region(prev) + damon_sz_region(r) <= sz_limit)
-- 
2.47.3

