From: SJ Park <sj@kernel.org>
Date: Mon, 6 Jul 2026 17:57:40 -0700
Subject: [PATCH] mm/damon/core: disallow probe_hits overflow on attrs only
 monitoring

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index bf3745f92de77..17980fa2004af 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1485,9 +1485,15 @@ static void damos_set_filters_default_reject(struct damos *s)
 
 static bool damon_valid_probe_params(struct damon_ctx *ctx)
 {
+	unsigned long sample_interval;
+
 	if (!damon_has_probe_weights(ctx))
 		return true;
-	return false;
+
+	sample_interval = ctx->attrs.sample_interval ? : 1;
+	if (ctx->attrs.aggr_interval / sample_interval > U8_MAX)
+		return false;
+	return true;
 }
 
 /*
-- 
2.47.3

