From: SeongJae Park <sj@kernel.org>
Date: Sat, 18 Apr 2026 08:42:30 -0700
Subject: [PATCH] mm/damon/lru_sort: fixup

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/lru_sort.c | 26 +++++++++++++++-----------
 1 file changed, 15 insertions(+), 11 deletions(-)

diff --git a/mm/damon/lru_sort.c b/mm/damon/lru_sort.c
index 3ae8632d6b0a9..2eb559d913b62 100644
--- a/mm/damon/lru_sort.c
+++ b/mm/damon/lru_sort.c
@@ -458,27 +458,20 @@ static bool damon_lru_sort_enabled(void)
 static int damon_lru_sort_enabled_store(const char *val,
 		const struct kernel_param *kp)
 {
-	bool enable;
 	int err;
 
-	err = kstrtobool(val, &enable);
+	err = kstrtobool(val, &enabled);
 	if (err)
 		return err;
 
-	if (damon_lru_sort_enabled() == enable)
+	if (damon_lru_sort_enabled() == enabled)
 		return 0;
 
 	/* Called before init function.  The function will handle this. */
 	if (!damon_initialized())
-		goto set_param_out;
-
-	err = damon_lru_sort_turn(enable);
-	if (err)
-		return err;
+		return 0;
 
-set_param_out:
-	enabled = enable;
-	return err;
+	return damon_lru_sort_turn(enabled);
 }
 
 static int damon_lru_sort_enabled_load(char *buffer,
@@ -496,6 +489,16 @@ module_param_cb(enabled, &enabled_param_ops, &enabled, 0600);
 MODULE_PARM_DESC(enabled,
 	"Enable or disable DAMON_LRU_SORT (default: disabled)");
 
+static int damon_lru_sort_kdamond_pid_store(const char *val,
+		const struct kernel_param *kp)
+{
+	/*
+	 * kdamond_pid is read-only, but kernel command line could write it.
+	 * Do nothing here.
+	 */
+	return 0;
+}
+
 static int damon_lru_sort_kdamond_pid_load(char *buffer,
 		const struct kernel_param *kp)
 {
@@ -510,6 +513,7 @@ static int damon_lru_sort_kdamond_pid_load(char *buffer,
 }
 
 static const struct kernel_param_ops kdamond_pid_param_ops = {
+	.set = damon_lru_sort_kdamond_pid_store,
 	.get = damon_lru_sort_kdamond_pid_load,
 };
 
-- 
2.47.3

