From: SeongJae Park <sj@kernel.org>
Date: Tue, 2 Dec 2025 21:08:07 -0800
Subject: [PATCH] mm/damon: damon.h kernel-doc wordsmithing

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 39 +++++++++++++++++++++++++++++++--------
 1 file changed, 31 insertions(+), 8 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 6bd395038f6b..f9fd26f7eab8 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -113,11 +113,15 @@ struct damon_target {
 
 /**
  * struct damon_access_report - Represent single acces report information.
- * @addr:		The start address of the reporting region.
- * @size:		The size of the reporting region.
- * @cpu:		CPU that made the access.
- * @tid:		The task id of the accessing task.
+ * @addr:		The start address of the accessed address range.
+ * @size:		The size of the accessed address range.
+ * @cpu:		The id of the CPU that made the access.
+ * @tid:		The task id of the task that made the access.
  * @is_write:		Whether the access is write.
+ *
+ * Any DAMON API callers that notified access events can report the information
+ * to DAMON using damon_report_access().  This struct contains the reporting
+ * infomration.  Refer to damon_report_access() for more details.
  */
 struct damon_access_report {
 	unsigned long addr;
@@ -770,8 +774,11 @@ struct damon_attrs {
 
 /**
  * struct damon_primitives_enabled - Enablement of access sampling primitives.
- * @page_table:	Page table accessed bit scanning.
+ *
+ * @page_table:	Page table Accessed bits scanning.
  * @page_fault:	Page faults monitoring.
+ *
+ * Read &struct damon_sample_control for more details.
  */
 struct damon_primitives_enabled {
 	bool page_table;
@@ -780,9 +787,12 @@ struct damon_primitives_enabled {
 
 /**
  * enum damon_sample_filter_type - Type of &struct damon_sample_filter.
+ *
  * @DAMON_FILTER_TYPE_CPUMASK:	Filter by access-generated CPUs.
  * @DAMON_FILTER_TYPE_THREADS:	Filter by access-generated threads.
  * @DAMON_FILTER_TYPE_WRITE:	Filter by whether the access was for writing.
+ *
+ * Read &struct damon_sample_control for more details.
  */
 enum damon_sample_filter_type {
 	DAMON_FILTER_TYPE_CPUMASK,
@@ -792,14 +802,17 @@ enum damon_sample_filter_type {
 
 /**
  * struct damon_sample_filter - &struct damon_access_report filter.
+ *
  * @type:	The type of this filter.
  * @matching:	Whether it is for condition-matching reports.
  * @allow:	Whether to include or excludie the @matching reports.
- * @cpumask:	Access-originated CPUs if @type is DAMON_FILTER_TYPE_CPUMASK.
- * @tid_arr:	Access-originated thread ids array, if @type is
+ * @cpumask:	Access-generated CPUs if @type is DAMON_FILTER_TYPE_CPUMASK.
+ * @tid_arr:	Array of access-generated thread ids, if @type is
  *		DAMON_FILTER_TYPE_THREADS.
  * @nr_tids:	Size of @tid_arr, if @type is DAMON_FILTER_TYPE_THREADS.
  * @list:	List head for siblings.
+ *
+ * Read &struct damon_sample_control for more details.
  */
 struct damon_sample_filter {
 	enum damon_sample_filter_type type;
@@ -817,8 +830,18 @@ struct damon_sample_filter {
 
 /**
  * struct damon_sample_control - Low level access check sampling rules.
+ *
  * @primitives_enabled:	Enablement of access check primitives.
- * @sample_filters:		List of added access check report filters.
+ * @sample_filters:	List of access check sample filters.
+ *
+ * DAMON collect low level access information using sampling, and aggregate
+ * that to make higher access pattern picture.  It can use multiple sampling
+ * primitives including page table accessed bits and page fault events.  It can
+ * also filter in/out specific types of sampled access events to monitor
+ * accesses of specific types, such as access-generated CPUs, threads, and
+ * whether it was for read or writes.  This struct is for controlling what
+ * sampling primitives to use (enable), and what sampled access events should
+ * be filtered in/out.
  */
 struct damon_sample_control {
 	struct damon_primitives_enabled primitives_enabled;
-- 
2.47.3

