From: SeongJae Park <sj@kernel.org>
Date: Wed, 10 Jun 2026 07:51:42 -0700
Subject: [PATCH] mm/damon/core: return error for wrong region commit attempts

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 7f5df0c5362f2..ca9859cb15435 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1505,6 +1505,8 @@ static int damon_commit_target_regions(struct damon_target *dst,
 	int i = 0, err;
 
 	damon_for_each_region(src_region, src) {
+		if (src_region->ar.start >= src_region->ar.end)
+			return -EINVAL;
 		if (!prev || prev->ar.end != src_region->ar.start)
 			i++;
 		prev = src_region;
-- 
2.47.3

