From: SeongJae Park <sj@kernel.org>
Date: Sat, 3 Jan 2026 10:26:34 -0800
Subject: [PATCH] mm/damon/core: add damon_nr_regions() hardening

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index b848f44e08aa4..cfa2548f92fef 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -563,8 +563,33 @@ void damon_destroy_target(struct damon_target *t, struct damon_ctx *ctx)
 	damon_free_target(t);
 }
 
+#ifdef CONFIG_DAMON_HARDENED
+static void damon_verify_nr_regions(struct damon_target *t)
+{
+	struct damon_region *r;
+	unsigned int count = 0;
+	static unsigned called;
+
+	if (called++ % 100)
+		return;
+
+	damon_for_each_region(r, t)
+		count++;
+
+	if (count != t->nr_regions)
+		pr_err("%s expected %u but %u\n", __func__, count, t->nr_regions);
+	BUG_ON(count != t->nr_regions);
+}
+#else
+static void damon_verify_nr_regions(struct damon_target *t)
+{
+}
+#endif
+
 unsigned int damon_nr_regions(struct damon_target *t)
 {
+	damon_verify_nr_regions(t);
+
 	return t->nr_regions;
 }
 
-- 
2.47.3

