From: SeongJae Park <sj@kernel.org>
Date: Fri, 28 Nov 2025 12:31:55 -0800
Subject: [PATCH] mm/damon/core: initialize report filters on damon_new_ctx()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 818c54eeb664..02aaf82e396a 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -560,6 +560,8 @@ struct damon_ctx *damon_new_ctx(void)
 	ctx->attrs.min_nr_regions = 10;
 	ctx->attrs.max_nr_regions = 1000;
 
+	INIT_LIST_HEAD(&ctx->access_check_control.report_filters);
+
 	ctx->addr_unit = 1;
 	ctx->min_sz_region = DAMON_MIN_REGION;
 
-- 
2.47.3

