From: SeongJae Park <sj@kernel.org>
Date: Mon, 10 Feb 2025 16:53:30 -0800
Subject: [PATCH] mm/damon/core: add debugging log for intervals auto-tuning
 update

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 3d01ac4a7c17..fa43eae48d5d 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1463,6 +1463,9 @@ static void kdamond_tune_intervals(struct damon_ctx *c)
 			goal->min_sample_us);
 	new_attrs.aggr_interval = new_attrs.sample_interval *
 		c->attrs.aggr_samples;
+
+	pr_info("tune intervals to %lu %lu\n\n",
+			new_attrs.sample_interval, new_attrs.aggr_interval);
 	damon_set_attrs(c, &new_attrs);
 }
 
-- 
2.39.5

