From: SJ Park <sj@kernel.org>
Date: Thu, 23 Jul 2026 07:51:31 -0700
Subject: [PATCH] mm/damon/paddr: respect folio boundary for DAMOS_STAT

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/paddr.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index 95246b5023100..a4989bd81d963 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -489,15 +489,21 @@ static unsigned long damon_pa_stat(struct damon_region *r,
 
 	addr = damon_pa_phys_addr(r->ar.start, addr_unit);
 	while (addr < damon_pa_phys_addr(r->ar.end, addr_unit)) {
+		phys_addr_t folio_addr;
+
 		folio = damon_get_folio(PHYS_PFN(addr));
 		if (damon_pa_invalid_damos_folio(folio, s)) {
 			addr += PAGE_SIZE;
 			continue;
 		}
+		folio_addr = PFN_PHYS(folio_pfn(folio));
+		if (folio_addr < addr)
+			goto put_folio;
 
 		if (!damos_pa_filter_out(s, folio))
 			*sz_filter_passed += folio_size(folio) / addr_unit;
-		addr += folio_size(folio);
+put_folio:
+		addr = folio_addr + folio_size(folio);
 		folio_put(folio);
 	}
 	s->last_applied = folio;
-- 
2.47.3

