From: SeongJae Park <sj@kernel.org>
Date: Sat, 23 May 2026 13:01:03 -0700
Subject: [PATCH] mm/damon: mark damos_quota_goal->list as private

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index a3378ff444a01..8c497f5849c53 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -213,7 +213,6 @@ enum damos_quota_goal_metric {
  * @last_psi_total:	Last measured total PSI
  * @nid:		Node id.
  * @memcg_id:		Memcg id.
- * @list:		List head for siblings.
  *
  * Data structure for getting the current score of the quota tuning goal.  The
  * score is calculated by how close @current_value and @target_value are.  Then
@@ -242,6 +241,8 @@ struct damos_quota_goal {
 			u64 memcg_id;
 		};
 	};
+/* private: */
+	/* List head for siblings. */
 	struct list_head list;
 };
 
-- 
2.47.3

