From: SeongJae Park <sj@kernel.org>
Date: Mon, 11 May 2026 20:47:46 -0700
Subject: [PATCH] mm/damon.h: rename

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 4692f89264b0a..a62d883320d73 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -733,10 +733,10 @@ struct damon_intervals_goal {
 /**
  * enum damon_filter_type - Type of &struct damon_filter
  *
- * @DAMON_TEST_TYPE_ANON:	Anonymous pages.
+ * @DAMON_FILTER_TYPE_ANON:	Anonymous pages.
  */
 enum damon_filter_type {
-	DAMON_TEST_TYPE_ANON,
+	DAMON_FILTER_TYPE_ANON,
 };
 
 /**
-- 
2.47.3

