From: SeongJae Park <sj@kernel.org>
Date: Sun, 17 Aug 2025 12:13:39 -0700
Subject: [PATCH] mm/damon/stat: keep sign for negative idle time

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/stat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/stat.c b/mm/damon/stat.c
index 8268b3e04ca2..480f15367e61 100644
--- a/mm/damon/stat.c
+++ b/mm/damon/stat.c
@@ -122,7 +122,7 @@ static void damon_stat_set_idletime_percentiles(struct damon_ctx *c)
 		while (next_percentile <= accounted_bytes * 100 / total_sz)
 			memory_idle_ms_percentiles[next_percentile++] =
 				damon_stat_idletime(region) *
-				c->attrs.aggr_interval / USEC_PER_MSEC;
+				(long)c->attrs.aggr_interval / USEC_PER_MSEC;
 	}
 	kfree(sorted_regions);
 }
-- 
2.39.5

