From: SeongJae Park <sj@kernel.org>
Date: Sun, 22 Mar 2026 15:57:39 -0700
Subject: [PATCH] mm/damon: add damon_call_control->cleanup_fn

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index f1e8bbbfc9160..4ed4a37bea6ae 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -750,6 +750,7 @@ struct damon_operations {
  * @repeat:		Repeat invocations.
  * @return_code:	Return code from @fn invocation.
  * @dealloc_on_cancel:	If @repeat is true, de-allocate when canceled.
+ * @cleanup_fn:		Function to be called when this is removed.
  *
  * Control damon_call(), which requests specific kdamond to invoke a given
  * function.  Refer to damon_call() for more details.
@@ -760,6 +761,7 @@ struct damon_call_control {
 	bool repeat;
 	int return_code;
 	bool dealloc_on_cancel;
+	void (*cleanup_fn)(struct damon_call_control *control);
 /* private: internal use only */
 	/* informs if the kdamond finished handling of the request */
 	struct completion completion;
-- 
2.47.3

