From: SeongJae Park <sj@kernel.org>
Date: Fri, 17 Oct 2025 11:23:45 -0700
Subject: [PATCH] mm/damon/core: fixup damos_quota_node_memcg_free_bp
 implementation

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 8baa339424c1..5f0172598aaf 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -826,6 +826,7 @@ static void damos_commit_quota_goal_union(
 		dst->nid = src->nid;
 		break;
 	case DAMOS_QUOTA_NODE_MEMCG_USED_BP:
+	case DAMOS_QUOTA_NODE_MEMCG_FREE_BP:
 		dst->nid = src->nid;
 		dst->memcg_id = src->memcg_id;
 		break;
-- 
2.47.3

