From: SeongJae Park <sj@kernel.org>
Date: Sat, 23 May 2026 13:43:27 -0700
Subject: [PATCH] mm/damon: mark filters and last_applied of struct damos as
 private

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 58897c6fd9dd4..5cf7e35e6a261 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -561,9 +561,6 @@ struct damos_migrate_dests {
  * @wmarks:		Watermarks for automated (in)activation of this scheme.
  * @migrate_dests:	Destination nodes if @action is "migrate_{hot,cold}".
  * @target_nid:		Destination node if @action is "migrate_{hot,cold}".
- * @core_filters:	Additional set of &struct damos_filter for &action.
- * @ops_filters:	ops layer handling &struct damos_filter objects list.
- * @last_applied:	Last @action applied ops-managing entity.
  * @stat:		Statistics of this scheme.
  * @max_nr_snapshots:	Upper limit of nr_snapshots stat.
  * @list:		List head for siblings.
@@ -600,7 +597,7 @@ struct damos_migrate_dests {
  *
  * Before applying the &action to a memory region, &struct damon_operations
  * implementation could check pages of the region and skip &action to respect
- * &core_filters
+ * &struct damos_filter.
  *
  * The minimum entity that @action can be applied depends on the underlying
  * &struct damon_operations.  Since it may not be aligned with the core layer
@@ -650,9 +647,14 @@ struct damos {
 			struct damos_migrate_dests migrate_dests;
 		};
 	};
+/* private: */
+	/* Additional set of &struct damos_filter for &action. */
 	struct list_head core_filters;
+	/* ops layer handling &struct damos_filter objects list. */
 	struct list_head ops_filters;
+	/* Last @action applied ops-managing entity. */
 	void *last_applied;
+/* public: */
 	struct damos_stat stat;
 	unsigned long max_nr_snapshots;
 	struct list_head list;
-- 
2.47.3

