From: SJ Park <sj@kernel.org>
Date: Sun, 19 Jul 2026 10:59:13 -0700
Subject: [PATCH] mm/damon/sysfs: fix wrong prep dir creation

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index c070708ea9762..cf4ba873d8277 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -897,7 +897,8 @@ static int damon_sysfs_preps_add_dirs(struct damon_sysfs_preps *preps,
 		}
 
 		err = kobject_init_and_add(&prep->kobj,
-				&damon_sysfs_prep_ktype, &prep->kobj, "%d", i);
+				&damon_sysfs_prep_ktype, &preps->kobj, "%d",
+				i);
 		if (err) {
 			kobject_put(&prep->kobj);
 			damon_sysfs_preps_rm_dirs(preps);
-- 
2.47.3

