From: SeongJae Park <sj@kernel.org>
Date: Fri, 19 Jun 2026 09:26:57 -0700
Subject: [PATCH] mm/damon: mark damon_region->list as private

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 001b8a1924e0a..71e11909940d2 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -48,7 +48,6 @@ struct damon_size_range {
  * @sampling_addr:	Address of the sample for the next access check.
  * @nr_accesses:	Access frequency of this region.
  * @probe_hits:		Number of probe-positive region samples.
- * @list:		List head for siblings.
  * @age:		Age of this region.
  *
  * For any use case, @ar should be non-zero positive size.
@@ -69,9 +68,10 @@ struct damon_region {
 	unsigned long sampling_addr;
 	unsigned int nr_accesses;
 	unsigned char probe_hits[DAMON_MAX_PROBES];
-	struct list_head list;
 	unsigned int age;
 /* private: Internal value for age calculation. */
+	/* List head for siblings. */
+	struct list_head list;
 	unsigned int last_nr_accesses;
 	unsigned char last_probe_hits[DAMON_MAX_PROBES];
 	bool access_reported;
-- 
2.47.3

