From: SJ Park <sj@kernel.org>
Date: Mon, 6 Jul 2026 23:42:18 -0700
Subject: [PATCH] mm/damon/core: reuse damon_nr_samples_per_aggr()

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index c9a688451743f..d482ef2878dd6 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1498,8 +1498,7 @@ static bool damon_valid_probe_params(struct damon_ctx *ctx)
 		return false;
 
 	/* invlaid if probe hits weighted sum can overflow */
-	max_probe_hits = ctx->attrs.aggr_interval / sample_interval;
-	max_probe_hits = max_probe_hits ? : 1;
+	max_probe_hits = damon_nr_samples_per_aggr(&ctx->attrs);
 	wsum = 0;
 	damon_for_each_probe(probe, ctx) {
 		if (probe->weight > UINT_MAX / max_probe_hits)
-- 
2.47.3

