From: SeongJae Park <sj@kernel.org>
Date: Mon, 15 Jun 2026 21:37:46 -0700
Subject: [PATCH] mm/damon: add damon_ctx->preps

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 mm/damon/core.c       | 1 +
 2 files changed, 3 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 87eec3e7ec5fe..a6aa796458301 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -990,6 +990,7 @@ struct damon_sample_control {
  * including damon_call() and damos_walk().
  *
  * @ops:	Set of monitoring operations for given use cases.
+ * @preps:	Head of preps (&damon_prep) list.
  * @probes:	Head of probes (&damon_probe) list.
  * @addr_unit:	Scale factor for core to ops address conversion.
  * @min_region_sz:	Minimum region size.
@@ -1046,6 +1047,7 @@ struct damon_ctx {
 
 /* public: */
 	struct damon_operations ops;
+	struct list_head preps;
 	struct list_head probes;
 	struct damon_sample_control sample_control;
 	unsigned long addr_unit;
diff --git a/mm/damon/core.c b/mm/damon/core.c
index 3a1e57da8f396..bbab34200862d 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -882,6 +882,7 @@ struct damon_ctx *damon_new_ctx(void)
 	ctx->attrs.max_nr_regions = 1000;
 
 	INIT_LIST_HEAD(&ctx->probes);
+	INIT_LIST_HEAD(&ctx->preps);
 
 	ctx->sample_control.primitives_enabled.page_table = true;
 	INIT_LIST_HEAD(&ctx->sample_control.sample_filters);
-- 
2.47.3

