From: SeongJae Park <sj@kernel.org>
Date: Wed, 4 Feb 2026 21:56:26 -0800
Subject: [PATCH] mm/damon/stat: cover all system rams

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/stat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/stat.c b/mm/damon/stat.c
index 637d55cdf53ab..7e1a9e1adcd24 100644
--- a/mm/damon/stat.c
+++ b/mm/damon/stat.c
@@ -180,7 +180,7 @@ static struct damon_ctx *damon_stat_build_ctx(void)
 	if (!target)
 		goto free_out;
 	damon_add_target(ctx, target);
-	if (damon_set_region_biggest_system_ram_default(target, &start, &end,
+	if (damon_set_region_system_rams_default(target, &start, &end,
 				ctx->addr_unit, ctx->min_region_sz))
 		goto free_out;
 	return ctx;
-- 
2.47.3

