From: SeongJae Park <sj@kernel.org>
Date: Sat, 23 May 2026 14:08:09 -0700
Subject: [PATCH] mm/damon/reclaim: remove duplicated min_region_sz power of 2
 check

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/reclaim.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/mm/damon/reclaim.c b/mm/damon/reclaim.c
index a14f6ec202b06..2bb77bdf0de43 100644
--- a/mm/damon/reclaim.c
+++ b/mm/damon/reclaim.c
@@ -210,11 +210,6 @@ static int damon_reclaim_apply_parameters(void)
 	param_ctx->addr_unit = addr_unit;
 	param_ctx->min_region_sz = max(DAMON_MIN_REGION_SZ / addr_unit, 1);
 
-	if (!is_power_of_2(param_ctx->min_region_sz)) {
-		err = -EINVAL;
-		goto out;
-	}
-
 	if (!damon_reclaim_mon_attrs.aggr_interval) {
 		err = -EINVAL;
 		goto out;
-- 
2.47.3

