From: SeongJae Park <sj@kernel.org>
Date: Sat, 18 Apr 2026 11:57:49 -0700
Subject: [PATCH] mm/damon: add damon_attrs->data_attrs

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index dba172ce0f3d9..25a2677f22d57 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -792,6 +792,15 @@ struct damon_intervals_goal {
 	unsigned long max_sample_us;
 };
 
+/**
+ * struct damon_data_attr - Data attribute to monitor.
+ *
+ * @list:	List for siblings.
+ */
+struct damon_data_attr {
+	struct list_head list;
+};
+
 /**
  * struct damon_attrs - Monitoring attributes for accuracy/overhead control.
  *
@@ -819,6 +828,7 @@ struct damon_attrs {
 	unsigned long aggr_interval;
 	unsigned long ops_update_interval;
 	struct damon_intervals_goal intervals_goal;
+	struct list_head data_attrs;
 	unsigned long min_nr_regions;
 	unsigned long max_nr_regions;
 /* private: internal use only */
-- 
2.47.3

