From: SeongJae Park <sj@kernel.org>
Date: Mon, 16 Dec 2024 19:37:21 -0800
Subject: [PATCH] mm/damon: add DAMOS_STAT_FULL kernel-doc comment

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 115086ec2e4e..f55f79e85e6d 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -113,6 +113,7 @@ struct damon_target {
  * @DAMOS_FREE:		Return DAMOS_ALLOC-ed pages back to the system.
 #endif
  * @DAMOS_STAT:		Do nothing but count the stat.
+ * @DAMOS_STAT_FULL:	Do nothing but filtering and stat counting.
  * @NR_DAMOS_ACTIONS:	Total number of DAMOS actions
  *
  * The support of each action is up to running &struct damon_operations.
-- 
2.39.5

