From: SJ Park <sj@kernel.org>
Date: Mon, 6 Jul 2026 17:54:44 -0700
Subject: [PATCH] mm/damon/core: add probe params validation

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 0bc301e49708e..56827f5ada950 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1483,6 +1483,13 @@ static void damos_set_filters_default_reject(struct damos *s)
 		damos_filters_default_reject(&s->ops_filters);
 }
 
+static bool damon_valid_probe_params(struct damon_ctx *ctx)
+{
+	if (!damon_has_probe_weights(ctx))
+		return true;
+	return false;
+}
+
 /*
  * damos_commit_dests() - Copy migration destinations from @src to @dst.
  * @dst:	Destination structure to update.
@@ -2014,6 +2021,9 @@ static int __damon_commit_ctx(struct damon_ctx *dst, struct damon_ctx *src)
 		}
 	}
 
+	if (!damon_valid_probe_params(src))
+		return -EINVAL;
+
 	err = damon_commit_schemes(dst, src);
 	if (err)
 		return err;
-- 
2.47.3

