From: SeongJae Park <sj@kernel.org>
Date: Sat, 2 Aug 2025 10:38:12 -0700
Subject: [PATCH] mm/damon/core: use reports based on ops_attrs.use_reports

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index a073cb3e55ea..1166e02a50ce 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2706,10 +2706,10 @@ static int kdamond_fn(void *data)
 		kdamond_usleep(sample_interval);
 		ctx->passed_sample_intervals++;
 
-		if (ctx->ops.check_accesses)
-			max_nr_accesses = ctx->ops.check_accesses(ctx);
-		else
+		if (ctx->ops_attrs.use_reports)
 			max_nr_accesses = kdamond_check_reported_accesses(ctx);
+		else if (ctx->ops.check_accesses)
+			max_nr_accesses = ctx->ops.check_accesses(ctx);
 
 		if (ctx->passed_sample_intervals >= next_aggregation_sis)
 			kdamond_merge_regions(ctx,
-- 
2.39.5

