From: SeongJae Park <sj@kernel.org>
Date: Sat, 23 May 2026 14:01:44 -0700
Subject: [PATCH] mm/damon/core: validate src on damon_commit_ctx()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 30 +++++++++++++++++++++++++++++-
 1 file changed, 29 insertions(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index b2582e7437440..5dddcabfcaf1c 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1779,6 +1779,22 @@ static int __damon_commit_ctx(struct damon_ctx *dst, struct damon_ctx *src)
 	return 0;
 }
 
+static struct damon_ctx *damon_new_test_ctx(struct damon_ctx *dst)
+{
+	struct damon_ctx *test_ctx;
+	int err;
+
+	test_ctx = damon_new_ctx();
+	if (!test_ctx)
+		return NULL;
+	err = __damon_commit_ctx(test_ctx, dst);
+	if (err) {
+		damon_destroy_ctx(test_ctx);
+		return NULL;
+	}
+	return test_ctx;
+}
+
 /**
  * damon_commit_ctx() - Commit parameters of a DAMON context to another.
  * @dst:	The commit destination DAMON context.
@@ -1794,7 +1810,19 @@ static int __damon_commit_ctx(struct damon_ctx *dst, struct damon_ctx *src)
  */
 int damon_commit_ctx(struct damon_ctx *dst, struct damon_ctx *src)
 {
-	return __damon_commit_ctx(dst, src);
+	struct damon_ctx *test_ctx;
+	int err;
+
+	test_ctx = damon_new_test_ctx(dst);
+	if (!test_ctx)
+		return -ENOMEM;
+	err = __damon_commit_ctx(test_ctx, dst);
+	if (err)
+		goto out;
+	err = __damon_commit_ctx(dst, src);
+out:
+	damon_destroy_ctx(test_ctx);
+	return err;
 }
 
 /**
-- 
2.47.3

