From: SeongJae Park <sj@kernel.org>
Date: Sat, 3 Jan 2026 10:28:12 -0800
Subject: [PATCH] mm/damon/core: add damon_split_region_at() hardening

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index c01c7458f5e5b..2878677a34c99 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2872,6 +2872,24 @@ static void kdamond_merge_regions(struct damon_ctx *c, unsigned int threshold,
 			threshold / 2 < max_thres);
 }
 
+#ifdef CONFIG_DAMON_HARDENED
+static void damon_verify_split_region_at(unsigned long sz_r,
+		struct damon_region *r)
+{
+
+	if (sz_r == 0 || sz_r >= r->ar.end - r->ar.start) {
+		pr_err("%s called with region of %lu-%lu and sz_r %lu!\n",
+				__func__, r->ar.start, r->ar.end, sz_r);
+		BUG();
+	}
+}
+#else
+static void damon_verify_split_region_at(unsigned long sz_r,
+		struct damon_region *r)
+{
+}
+#endif
+
 /*
  * Split a region in two
  *
@@ -2883,6 +2901,8 @@ static void damon_split_region_at(struct damon_target *t,
 {
 	struct damon_region *new;
 
+	damon_verify_split_region_at(sz_r, r);
+
 	new = damon_new_region(r->ar.start + sz_r, r->ar.end);
 	if (!new)
 		return;
-- 
2.47.3

