From: SeongJae Park <sj@kernel.org>
Date: Tue, 17 Feb 2026 22:03:29 -0800
Subject: [PATCH] mm/damon/tests/.kunitconifg: turn on DAMON_HARDENED

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/tests/.kunitconfig | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mm/damon/tests/.kunitconfig b/mm/damon/tests/.kunitconfig
index 36a450f57b581..5153045e8e32f 100644
--- a/mm/damon/tests/.kunitconfig
+++ b/mm/damon/tests/.kunitconfig
@@ -13,3 +13,6 @@ CONFIG_DAMON_VADDR_KUNIT_TEST=y
 CONFIG_SYSFS=y
 CONFIG_DAMON_SYSFS=y
 CONFIG_DAMON_SYSFS_KUNIT_TEST=y
+
+# enable DAMON_HARDENED to catch any bug
+CONFIG_DAMON_HARDENED=y
-- 
2.47.3

