From: SJ Park <sj@kernel.org>
Date: Fri, 3 Jul 2026 15:46:47 -0700
Subject: [PATCH] mm/damon/paddr: set sampling address in apply_probes when
 weight is set

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/paddr.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index 6172464981730..da043f4324319 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -178,6 +178,9 @@ static void damon_pa_apply_probes(struct damon_ctx *ctx)
 			phys_addr_t pa;
 			struct folio *folio;
 
+			if (ctx->has_probe_weights)
+				r->sampling_addr = damon_rand(ctx, r->ar.start,
+						r->ar.end);
 			pa = damon_pa_phys_addr(r->sampling_addr,
 					ctx->addr_unit);
 			folio = damon_get_folio(PHYS_PFN(pa));
@@ -188,6 +191,7 @@ static void damon_pa_apply_probes(struct damon_ctx *ctx)
 			}
 			if (folio)
 				folio_put(folio);
+
 		}
 	}
 }
-- 
2.47.3

