From: SeongJae Park <sj@kernel.org>
Date: Sun, 30 Nov 2025 08:26:47 -0800
Subject: [PATCH] mm/damon/sysfs: s/access_check/sample/g

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 102 +++++++++++++++++++++++------------------------
 1 file changed, 51 insertions(+), 51 deletions(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index a5db8029f804..1d3e3a5485d8 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1191,28 +1191,28 @@ static const struct kobj_type damon_sysfs_primitives_ktype = {
 };
 
 /*
- * access_check directory
+ * sample directory
  */
 
-struct damon_sysfs_access_check {
+struct damon_sysfs_sample {
 	struct kobject kobj;
 	struct damon_sysfs_primitives *primitives;
 	struct damon_sysfs_report_filters *filters;
 };
 
-static struct damon_sysfs_access_check *damon_sysfs_access_check_alloc(void)
+static struct damon_sysfs_sample *damon_sysfs_sample_alloc(void)
 {
-	struct damon_sysfs_access_check *access_check = kmalloc(
-			sizeof(*access_check), GFP_KERNEL);
+	struct damon_sysfs_sample *sample = kmalloc(
+			sizeof(*sample), GFP_KERNEL);
 
-	if (!access_check)
+	if (!sample)
 		return NULL;
-	access_check->kobj = (struct kobject){};
-	return access_check;
+	sample->kobj = (struct kobject){};
+	return sample;
 }
 
-static int damon_sysfs_access_check_add_dirs(
-		struct damon_sysfs_access_check *access_check)
+static int damon_sysfs_sample_add_dirs(
+		struct damon_sysfs_sample *sample)
 {
 	struct damon_sysfs_primitives *primitives;
 	struct damon_sysfs_report_filters *filters;
@@ -1222,56 +1222,56 @@ static int damon_sysfs_access_check_add_dirs(
 	if (!primitives)
 		return -ENOMEM;
 	err = kobject_init_and_add(&primitives->kobj,
-			&damon_sysfs_primitives_ktype, &access_check->kobj,
+			&damon_sysfs_primitives_ktype, &sample->kobj,
 			"primitives");
 	if (err)
 		goto put_primitives_out;
-	access_check->primitives = primitives;
+	sample->primitives = primitives;
 
 	filters = damon_sysfs_report_filters_alloc();
 	if (!filters)
 		return -ENOMEM;
 	err = kobject_init_and_add(&filters->kobj,
-			&damon_sysfs_report_filters_ktype, &access_check->kobj,
+			&damon_sysfs_report_filters_ktype, &sample->kobj,
 			"report_filters");
 	if (err)
 		goto put_filters_out;
-	access_check->filters = filters;
+	sample->filters = filters;
 	return 0;
 put_filters_out:
 	kobject_put(&filters->kobj);
-	access_check->filters = NULL;
+	sample->filters = NULL;
 put_primitives_out:
 	kobject_put(&primitives->kobj);
-	access_check->primitives = NULL;
+	sample->primitives = NULL;
 	return err;
 }
 
-static void damon_sysfs_access_check_rm_dirs(
-		struct damon_sysfs_access_check *access_check)
+static void damon_sysfs_sample_rm_dirs(
+		struct damon_sysfs_sample *sample)
 {
-	if (access_check->primitives)
-		kobject_put(&access_check->primitives->kobj);
-	if (access_check->filters) {
-		damon_sysfs_report_filters_rm_dirs(access_check->filters);
-		kobject_put(&access_check->filters->kobj);
+	if (sample->primitives)
+		kobject_put(&sample->primitives->kobj);
+	if (sample->filters) {
+		damon_sysfs_report_filters_rm_dirs(sample->filters);
+		kobject_put(&sample->filters->kobj);
 	}
 }
 
-static void damon_sysfs_access_check_release(struct kobject *kobj)
+static void damon_sysfs_sample_release(struct kobject *kobj)
 {
-	kfree(container_of(kobj, struct damon_sysfs_access_check, kobj));
+	kfree(container_of(kobj, struct damon_sysfs_sample, kobj));
 }
 
-static struct attribute *damon_sysfs_access_check_attrs[] = {
+static struct attribute *damon_sysfs_sample_attrs[] = {
 	NULL,
 };
-ATTRIBUTE_GROUPS(damon_sysfs_access_check);
+ATTRIBUTE_GROUPS(damon_sysfs_sample);
 
-static const struct kobj_type damon_sysfs_access_check_ktype = {
-	.release = damon_sysfs_access_check_release,
+static const struct kobj_type damon_sysfs_sample_ktype = {
+	.release = damon_sysfs_sample_release,
 	.sysfs_ops = &kobj_sysfs_ops,
-	.default_groups = damon_sysfs_access_check_groups,
+	.default_groups = damon_sysfs_sample_groups,
 };
 
 /*
@@ -1282,7 +1282,7 @@ struct damon_sysfs_attrs {
 	struct kobject kobj;
 	struct damon_sysfs_intervals *intervals;
 	struct damon_sysfs_ul_range *nr_regions_range;
-	struct damon_sysfs_access_check *access_check;
+	struct damon_sysfs_sample *sample;
 };
 
 static struct damon_sysfs_attrs *damon_sysfs_attrs_alloc(void)
@@ -1299,7 +1299,7 @@ static int damon_sysfs_attrs_add_dirs(struct damon_sysfs_attrs *attrs)
 {
 	struct damon_sysfs_intervals *intervals;
 	struct damon_sysfs_ul_range *nr_regions_range;
-	struct damon_sysfs_access_check *access_check;
+	struct damon_sysfs_sample *sample;
 	int err;
 
 	intervals = damon_sysfs_intervals_alloc(5000, 100000, 60000000);
@@ -1329,25 +1329,25 @@ static int damon_sysfs_attrs_add_dirs(struct damon_sysfs_attrs *attrs)
 		goto put_nr_regions_intervals_out;
 	attrs->nr_regions_range = nr_regions_range;
 
-	access_check = damon_sysfs_access_check_alloc();
-	if (!access_check) {
+	sample = damon_sysfs_sample_alloc();
+	if (!sample) {
 		err = -ENOMEM;
 		goto put_nr_regions_intervals_out;
 	}
-	err = kobject_init_and_add(&access_check->kobj,
-			&damon_sysfs_access_check_ktype, &attrs->kobj,
-			"access_check");
+	err = kobject_init_and_add(&sample->kobj,
+			&damon_sysfs_sample_ktype, &attrs->kobj,
+			"sample");
 	if (err)
-		goto put_access_check_out;
-	err = damon_sysfs_access_check_add_dirs(access_check);
+		goto put_sample_out;
+	err = damon_sysfs_sample_add_dirs(sample);
 	if (err)
-		goto put_access_check_out;
-	attrs->access_check = access_check;
+		goto put_sample_out;
+	attrs->sample = sample;
 	return 0;
 
-put_access_check_out:
-	kobject_put(&access_check->kobj);
-	attrs->access_check = NULL;
+put_sample_out:
+	kobject_put(&sample->kobj);
+	attrs->sample = NULL;
 put_nr_regions_intervals_out:
 	kobject_put(&nr_regions_range->kobj);
 	attrs->nr_regions_range = NULL;
@@ -1364,8 +1364,8 @@ static void damon_sysfs_attrs_rm_dirs(struct damon_sysfs_attrs *attrs)
 	kobject_put(&attrs->nr_regions_range->kobj);
 	damon_sysfs_intervals_rm_dirs(attrs->intervals);
 	kobject_put(&attrs->intervals->kobj);
-	damon_sysfs_access_check_rm_dirs(attrs->access_check);
-	kobject_put(&attrs->access_check->kobj);
+	damon_sysfs_sample_rm_dirs(attrs->sample);
+	kobject_put(&attrs->sample->kobj);
 }
 
 static void damon_sysfs_attrs_release(struct kobject *kobj)
@@ -2257,15 +2257,15 @@ static int damon_sysfs_set_report_filters(
 
 static int damon_sysfs_set_sample_control(
 		struct damon_sample_control *control,
-		struct damon_sysfs_access_check *sysfs_access_check)
+		struct damon_sysfs_sample *sysfs_sample)
 {
 	control->primitives_enabled.page_table_scan =
-		sysfs_access_check->primitives->page_table;
+		sysfs_sample->primitives->page_table;
 	control->primitives_enabled.page_faults_monitoring =
-		sysfs_access_check->primitives->page_faults;
+		sysfs_sample->primitives->page_faults;
 
 	return damon_sysfs_set_report_filters(control,
-			sysfs_access_check->filters);
+			sysfs_sample->filters);
 }
 
 static int damon_sysfs_set_ops_attrs_tids(
@@ -2317,7 +2317,7 @@ static int damon_sysfs_apply_inputs(struct damon_ctx *ctx,
 	if (err)
 		return err;
 	err = damon_sysfs_set_sample_control(&ctx->sample_control,
-			sys_ctx->attrs->access_check);
+			sys_ctx->attrs->sample);
 	if (err)
 		return err;
 	err = damon_sysfs_add_targets(ctx, sys_ctx->targets);
-- 
2.47.3

