From e945d7b52fbd430556120e996ad02c202360cd60 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Sun, 3 Sep 2023 05:02:44 +0000
Subject: [PATCH] mm/damon/core: add todo for DAMOS interval validation

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 601215427bf8..7a1eaeea37a4 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1635,6 +1635,7 @@ static void kdamond_init_intervals_sis(struct damon_ctx *ctx)
 	ctx->next_ops_update_sis = ctx->attrs.ops_update_interval /
 		sample_interval;
 
+	/* todo: ensure apply_interval_us > sample_interval */
 	damon_for_each_scheme(scheme, ctx) {
 		apply_interval = scheme->apply_interval_us ?
 			scheme->apply_interval_us : ctx->attrs.aggr_interval;
-- 
2.39.2

