From: SJ Park <sj@kernel.org>
Date: Mon, 22 Jun 2026 20:33:20 -0700
Subject: [PATCH] mm/damon: introduce DAMON_FILTER_TYPE_PGIDLE

Signed-off-by: SJ Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index e8e378d25a322..c2336974deae0 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -796,10 +796,12 @@ struct damon_intervals_goal {
  *
  * @DAMON_FILTER_TYPE_ANON:	Anonymous pages.
  * @DAMON_FILTER_TYPE_MEMCG:	Specific memcg's pages.
+ * @DAMON_FILTER_TYPE_PGIDLE:	Idle pages.
  */
 enum damon_filter_type {
 	DAMON_FILTER_TYPE_ANON,
 	DAMON_FILTER_TYPE_MEMCG,
+	DAMON_FILTER_TYPE_PGIDLE,
 };
 
 /**
-- 
2.47.3

