From: SJ Park <sj@kernel.org>
Date: Mon, 27 Jul 2026 08:03:32 -0700
Subject: [PATCH] mm/damon/core: evaluate prep_probe first for do_prep (patch 7
 fixup)

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 805a92c67e87b..03d57f8cc6999 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -4258,7 +4258,7 @@ static int kdamond_fn(void *data)
 		if (kdamond_wait_activation(ctx))
 			break;
 
-		do_prep = damon_has_prep(ctx) && ctx->ops.prep_probes;
+		do_prep = ctx->ops.prep_probes && damon_has_prep(ctx);
 
 		if (!access_check_disabled && ctx->ops.prepare_access_checks)
 			ctx->ops.prepare_access_checks(ctx);
-- 
2.47.3

