From: SeongJae Park <sj@kernel.org>
Date: Sat, 3 Jan 2026 10:04:42 -0800
Subject: [PATCH] mm/damon: add CONFIG_DAMON_HARDENED

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/Kconfig | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/mm/damon/Kconfig b/mm/damon/Kconfig
index d5f8db812dd3..cac18b5b617c 100644
--- a/mm/damon/Kconfig
+++ b/mm/damon/Kconfig
@@ -120,4 +120,14 @@ config DAMON_ACMA
 	  min/max memory for the system and maximum memory pressure stall time
 	  ratio.
 
+config DAMON_HARDENED
+	bool "Check integrity of DAMON code"
+	help
+	  This enables additional DAMON debugging-purpose integrity checks in
+	  DAMON code.  This can impose additional overhead but can be useful at
+	  finding bugs.  Hence it is enouraged to enable on DAMON development
+	  setups.
+
+	  If unsure, say N.
+
 endmenu
-- 
2.47.3

