From: SeongJae Park <sj@kernel.org>
Date: Sat, 14 Mar 2026 14:16:10 -0700
Subject: [PATCH] mm/damon/core: allow stop during pause

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 7141489763f77..0641cdedc8255 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3012,6 +3012,8 @@ static int kdamond_fn(void *data)
 		 */
 		kdamond_call(ctx, false);
 		while (ctx->pause) {
+			if (kdamond_need_stop(ctx))
+				goto done;
 			kdamond_usleep(ctx->attrs.sample_interval);
 			/* allow caller unset pause via damon_call() */
 			kdamond_call(ctx, false);
-- 
2.47.3

