From: SJ Park <sj@kernel.org>
Date: Tue, 23 Jun 2026 21:03:23 -0700
Subject: [PATCH] mm/damon: add damon_ctx->preps

Signed-off-by: SJ Park <sj@kernel.org>
---
 include/linux/damon.h | 1 +
 mm/damon/core.c       | 1 +
 2 files changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index ddac549135afe..c0bda589476c2 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -934,6 +934,7 @@ struct damon_ctx {
 	struct damon_operations ops;
 	/* Head of monitoring targets (&damon_target) list. */
 	struct list_head adaptive_targets;
+	struct list_head preps;
 	/* Head of probes (&damon_probe) list. */
 	struct list_head probes;
 	/* Head of schemes (&damos) list. */
diff --git a/mm/damon/core.c b/mm/damon/core.c
index 5eb0400d5ce16..7c74ac802919c 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -852,6 +852,7 @@ struct damon_ctx *damon_new_ctx(void)
 	ctx->attrs.max_nr_regions = 1000;
 
 	INIT_LIST_HEAD(&ctx->probes);
+	INIT_LIST_HEAD(&ctx->preps);
 
 	ctx->addr_unit = 1;
 	ctx->min_region_sz = DAMON_MIN_REGION_SZ;
-- 
2.47.3

