From: SeongJae Park <sj@kernel.org>
Date: Wed, 24 Jun 2026 20:43:05 -0700
Subject: [PATCH] mm/damon/core: implement damon_merge_score()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 6b6ac1e89576b..23c99a773b6c5 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3556,6 +3556,16 @@ static unsigned long damon_probe_hits_wsum(struct damon_region *r, bool last,
 	return sum;
 }
 
+static unsigned long damon_merge_score(struct damon_region *r, bool last,
+		struct damon_ctx *ctx)
+{
+	if (ctx->has_probe_weights)
+		return damon_probe_hits_wsum(r, last, ctx);
+	if (last)
+		return r->last_nr_accesses;
+	return r->nr_accesses;
+}
+
 /*
  * Merge adjacent regions having similar access frequencies
  *
-- 
2.47.3

