From: SeongJae Park <sj@kernel.org>
Date: Fri, 5 Dec 2025 17:13:12 -0800
Subject: [PATCH] mm/damon: extend damon_access_report for virtual address

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index ddf1c7271f10..754fa109b407 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -114,6 +114,7 @@ struct damon_target {
 /**
  * struct damon_access_report - Represent single acces report information.
  * @paddr:		Start physical address of the accessed address range.
+ * @vaddr:		Start virtual address of the accessed address range.
  * @size:		The size of the accessed address range.
  * @cpu:		The id of the CPU that made the access.
  * @tid:		The task id of the task that made the access.
@@ -125,6 +126,7 @@ struct damon_target {
  */
 struct damon_access_report {
 	unsigned long paddr;
+	unsigned long vaddr;
 	unsigned long size;
 	unsigned int cpu;
 	pid_t tid;
-- 
2.47.3

