From: SeongJae Park <sj@kernel.org>
Date: Fri, 14 Feb 2025 17:12:49 -0800
Subject: [PATCH] mm/damon/paddr: respect damos->reject_by_default

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index b1514dbc2a22..2af5c07fe2e4 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -254,7 +254,7 @@ static bool damos_pa_filter_out(struct damos *scheme, struct folio *folio)
 		if (damos_pa_filter_match(filter, folio))
 			return !filter->allow;
 	}
-	return false;
+	return scheme->reject_by_default;
 }
 
 static bool damon_pa_invalid_damos_folio(struct folio *folio, struct damos *s)
-- 
2.39.5

