From: SJ Park <sj@kernel.org>
Date: Sat, 29 Aug 2026 09:45:38 -0700
Subject: [PATCH] mm/damon/sysfs: implement avail_types file under filter dir

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index c4691070a17ef..cad2d70788537 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1023,6 +1023,21 @@ damon_sysfs_filter_type_names[] = {
 	},
 };
 
+static ssize_t avail_types_show(struct kobject *kobj,
+		struct kobj_attribute *attr, char *buf)
+{
+	int len = 0;
+	int i;
+
+	for (i = 0; i < ARRAY_SIZE(damon_sysfs_filter_type_names); i++) {
+		const struct damon_sysfs_filter_type_name *type_name;
+
+		type_name = &damon_sysfs_filter_type_names[i];
+		len += sysfs_emit_at(buf, len, "%s\n", type_name->name);
+	}
+	return len;
+}
+
 static ssize_t type_show(struct kobject *kobj,
 		struct kobj_attribute *attr, char *buf)
 {
@@ -1152,6 +1167,9 @@ static void damon_sysfs_filter_release(struct kobject *kobj)
 	kfree(filter);
 }
 
+static struct kobj_attribute damon_sysfs_filter_avail_types_attr =
+		__ATTR_RO_MODE(avail_types, 0400);
+
 static struct kobj_attribute damon_sysfs_filter_type_attr =
 		__ATTR_RW_MODE(type, 0600);
 
@@ -1165,6 +1183,7 @@ static struct kobj_attribute damon_sysfs_filter_path_attr =
 		__ATTR_RW_MODE(path, 0600);
 
 static struct attribute *damon_sysfs_filter_attrs[] = {
+	&damon_sysfs_filter_avail_types_attr.attr,
 	&damon_sysfs_filter_type_attr.attr,
 	&damon_sysfs_filter_matching_attr.attr,
 	&damon_sysfs_filter_allow_attr.attr,
-- 
2.47.3

