From: SJ Park <sj@kernel.org>
Date: Sat, 4 Jul 2026 14:19:22 -0700
Subject: [PATCH] mm/damon/core: use unsigned int for merge score

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index cd29ffc186b48..6a86295596bf9 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3218,11 +3218,11 @@ static void damon_merge_two_regions(struct damon_target *t,
 	damon_destroy_region(r, t);
 }
 
-static unsigned long damon_probe_hits_wsum(struct damon_region *r, bool last,
+static unsigned int damon_probe_hits_wsum(struct damon_region *r, bool last,
 		struct damon_ctx *ctx)
 {
 	struct damon_probe *probe;
-	unsigned long sum = 0;
+	unsigned int sum = 0;
 	int i = 0;
 
 	damon_for_each_probe(probe, ctx) {
@@ -3234,7 +3234,7 @@ static unsigned long damon_probe_hits_wsum(struct damon_region *r, bool last,
 	return sum;
 }
 
-static unsigned long damon_merge_score(struct damon_region *r, bool last,
+static unsigned int damon_merge_score(struct damon_region *r, bool last,
 		struct damon_ctx *ctx, bool use_probe_hits)
 {
 	if (use_probe_hits)
@@ -3258,7 +3258,7 @@ static void damon_merge_regions_of(struct damon_target *t, unsigned int thres,
 	bool use_probe_hits = damon_has_probe_weights(ctx);
 
 	damon_for_each_region_safe(r, next, t) {
-		unsigned long score, last_score;
+		unsigned int score, last_score;
 
 		score = damon_merge_score(r, false, ctx, use_probe_hits);
 		last_score = damon_merge_score(r, true, ctx, use_probe_hits);
-- 
2.47.3

