From: SeongJae Park <sj@kernel.org>
Date: Sat, 3 Jan 2026 10:26:34 -0800
Subject: [PATCH] mm/damon/core: add damon_nr_regions() hardening

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 6896c7d0cba3..c314b9a810f8 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -550,7 +550,8 @@ void damon_destroy_target(struct damon_target *t, struct damon_ctx *ctx)
 	damon_free_target(t);
 }
 
-static void damon_nr_regions_verify(struct damon_target *t)
+#ifdef CONFIG_DAMON_HARDENED
+static void damon_verify_nr_regions(struct damon_target *t)
 {
 	struct damon_region *r;
 	unsigned int count = 0;
@@ -566,10 +567,15 @@ static void damon_nr_regions_verify(struct damon_target *t)
 		pr_err("%s expected %u but %u\n", __func__, count, t->nr_regions);
 	BUG_ON(count != t->nr_regions);
 }
+#else
+static void damon_verify_nr_regions(struct damon_target *t)
+{
+}
+#endif
 
 unsigned int damon_nr_regions(struct damon_target *t)
 {
-	damon_nr_regions_verify(t);
+	damon_verify_nr_regions(t);
 
 	return t->nr_regions;
 }
-- 
2.47.3

