From: SeongJae Park <sj@kernel.org>
Date: Mon, 11 May 2026 20:48:01 -0700
Subject: [PATCH] mm/damon/paddr: rename

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index 5d9636f23a66f..c3fd939f7d8e1 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -128,7 +128,7 @@ static bool damon_pa_filter_pass(phys_addr_t pa, struct damon_probe *p)
 	damon_for_each_filter(f, p) {
 		bool matched = false;
 
-		if (f->type == DAMON_TEST_TYPE_ANON) {
+		if (f->type == DAMON_FILTER_TYPE_ANON) {
 			struct folio *folio = damon_get_folio(PHYS_PFN(pa));
 
 			if (folio)
-- 
2.47.3

