From: SeongJae Park <sj@kernel.org>
Date: Sat, 2 Aug 2025 13:08:28 -0700
Subject: [PATCH] mm/damon: add damon_access_report->is_write

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 99418e4dfd5e..e08763ba9e24 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -111,6 +111,7 @@ struct damon_target {
  *			NULL if it is of the physical address.
  * @addr:		The start address of the reporting region.
  * @size:		The size of the reporting region.
+ * @is_write:		Whether the access is write.
  *
  * @pid could be stale, and hence shouldn't be de-referenced.
  */
@@ -118,6 +119,7 @@ struct damon_access_report {
 	struct pid *pid;
 	unsigned long addr;
 	unsigned long size;
+	bool is_write;
 /* private: */
 	unsigned long report_jiffies;	/* when this report is made */
 };
-- 
2.39.5

