From 19a2b03e8cfe244163070079336a49083e562c63 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Mon, 6 May 2024 17:49:44 -0700
Subject: [PATCH] mm/damon/acma: add the three major parameters

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/acma.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/mm/damon/acma.c b/mm/damon/acma.c
index 3ed2a350e7f1..8646cb612b45 100644
--- a/mm/damon/acma.c
+++ b/mm/damon/acma.c
@@ -47,6 +47,15 @@ static bool enabled __read_mostly;
 static bool commit_inputs __read_mostly;
 module_param(commit_inputs, bool, 0600);
 
+static unsigned long min_mem_kb __read_mostly;
+module_param(min_mem, ulong, 0600);
+
+static unsigned long max_mem_kb __read_mostly;
+module_param(max_mem, ulong, 0600);
+
+static unsigned int mem_pressure_ratio_bp __read_mostly;
+module_param(mem_pressure_ratio, ulong, 0600);
+
 /*
  * Access frequency threshold for hot memory regions identification in permil.
  *
-- 
2.39.2

