From: SeongJae Park <sj@kernel.org>
Date: Mon, 15 Jun 2026 21:50:14 -0700
Subject: [PATCH] mm/damon: mark ops, probes and sample_control of damon_ctx
 private

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index d50d74e380d25..b05681eef7c9c 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -973,8 +973,6 @@ struct damon_sample_control {
  * thread other than the kdamond should be made using safe DAMON APIs,
  * including damon_call() and damos_walk().
  *
- * @ops:	Set of monitoring operations for given use cases.
- * @probes:	Head of probes (&damon_probe) list.
  * @addr_unit:	Scale factor for core to ops address conversion.
  * @min_region_sz:	Minimum region size.
  * @pause:	Pause kdamond main loop.
@@ -1026,16 +1024,18 @@ struct damon_ctx {
 	struct mutex kdamond_lock;
 
 /* public: */
-	struct damon_operations ops;
-	struct list_head probes;
-	struct damon_sample_control sample_control;
 	unsigned long addr_unit;
 	unsigned long min_region_sz;
 	bool pause;
 
 /* private: */
+	/* Set of monitoring operations for given use cases. */
+	struct damon_operations ops;
 	/* Head of monitoring targets (&damon_target) list. */
 	struct list_head adaptive_targets;
+	/* Head of probes (&damon_probe) list. */
+	struct list_head probes;
+	struct damon_sample_control sample_control;
 	/* Head of schemes (&damos) list. */
 	struct list_head schemes;
 
-- 
2.47.3

