From: SeongJae Park <sj@kernel.org>
Date: Wed, 14 May 2025 07:54:02 -0700
Subject: [PATCH] mm/damon/stat: wordsmith parameter descriptions

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/stat.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mm/damon/stat.c b/mm/damon/stat.c
index 06237f9ab00a..796125d63b75 100644
--- a/mm/damon/stat.c
+++ b/mm/damon/stat.c
@@ -31,12 +31,12 @@ MODULE_PARM_DESC(enabled, "Enable of disable DAMON_STAT");
 static unsigned long estimated_memory_bandwidth __read_mostly = 0;
 module_param(estimated_memory_bandwidth, ulong, 0400);
 MODULE_PARM_DESC(estimated_memory_bandwidth,
-		"Estimated memory bandwidth usage");
+		"Estimated memory bandwidth usage in bytes per second");
 
 static unsigned long memory_idle_ms_percentiles[101] __read_mostly = {0,};
 module_param_array(memory_idle_ms_percentiles, ulong, NULL, 0400);
 MODULE_PARM_DESC(memory_idle_ms_percentiles,
-		"Per-page idle time percentiles");
+		"Memory idle time percentiles in milliseconds");
 
 static struct damon_ctx *damon_stat_context;
 
-- 
2.39.5

