From: SeongJae Park <sj@kernel.org>
Date: Tue, 3 Mar 2026 21:13:13 -0800
Subject: [PATCH] mm/damon/core: fixup build error

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index ff456d780b052..0cac46ed0bff1 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3104,7 +3104,7 @@ int damon_set_region_biggest_system_ram_default(struct damon_target *t,
 		return -EINVAL;
 
 	if (!*start && !*end &&
-		!damon_find_biggest_system_ram(start, end), 1)
+		!damon_find_biggest_system_ram(start, end, 1))
 		return -EINVAL;
 
 	addr_range.start = *start;
-- 
2.47.3

