From c19d2e62318316d8fec73100044d023be4931b67 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Sun, 5 May 2024 11:55:40 -0700
Subject: [PATCH] mm/damon/paddr: link preempt/yield DAMOS actions to
 implementations

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index 23bddce92fb9..b8ab82c55411 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -424,6 +424,12 @@ static unsigned long damon_pa_apply_scheme(struct damon_ctx *ctx,
 		return damon_pa_mark_accessed(r, scheme);
 	case DAMOS_LRU_DEPRIO:
 		return damon_pa_deactivate_pages(r, scheme);
+#ifdef CONFIG_ACMA
+	case DAMOS_PREEMPT:
+		return damon_pa_preempt(r, scheme);
+	case DAMOS_YIELD:
+		return damon_pa_yield(r, scheme);
+#endif
 	case DAMOS_STAT:
 		break;
 	default:
-- 
2.39.2

