From d3047f3d9d642c6d70a9701ae768cb8f6748baad Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Mon, 6 May 2024 18:26:38 -0700
Subject: [PATCH] mm/damon/acma: implement three schemes generator functions

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/acma.c | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/mm/damon/acma.c b/mm/damon/acma.c
index 8646cb612b45..b84e23c5a770 100644
--- a/mm/damon/acma.c
+++ b/mm/damon/acma.c
@@ -200,6 +200,30 @@ static struct damos *damon_acma_new_cold_scheme(unsigned int cold_thres)
 	return damon_acma_new_scheme(&pattern, DAMOS_LRU_DEPRIO);
 }
 
+/*
+ * Reclaim cold pages on entire physical address space
+ */
+static struct damos *damon_acma_new_relclaim_scheme()
+{
+	return NULL;
+}
+
+/*
+ * Preempt/report cold pages
+ */
+static struct damos *damon_acma_new_scale_down_scheme()
+{
+	return NULL;
+}
+
+/*
+ * Yield pages
+ */
+static struct damos *damon_acma_new_scale_up_scheme()
+{
+	return NULL;
+}
+
 static void damon_acma_copy_quota_status(struct damos_quota *dst,
 		struct damos_quota *src)
 {
-- 
2.39.2

