From: SeongJae Park <sj@kernel.org>
Date: Fri, 15 May 2026 17:12:14 -0700
Subject: [PATCH] mm/damon: (fixup 3) fix typo in damon_filter kernel-doc

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 91c6e034f257a..aeb4e8eec54ae 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -809,7 +809,7 @@ enum damon_filter_type {
  * struct damon_filter - DAMON region filter for &struct damon_probe.
  *
  * @type:	Type of the region.
- * @matcing:	Whether this filter is for the type-matching ones.
+ * @matching:	Whether this filter is for the type-matching ones.
  * @allow:	Whether the @type-@matching ones should pass this filter.
  * @memcg_id:	Memcg id of the question if @type is DAMON_FILTER_MEMCG.
  * @list:	Siblings list.
-- 
2.47.3

