From: SJ Park <sj@kernel.org>
Date: Sat, 4 Jul 2026 12:52:11 -0700
Subject: [PATCH] mm/damon/core: fixup damon_call()

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index e10a5f59c47ed..fddef00e24940 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2309,7 +2309,8 @@ int damon_call(struct damon_ctx *ctx, struct damon_call_control *control)
 	return 0;
 
 canceled:
-	__damon_stop(ctx);
+	while (damon_is_running(ctx))
+		schedule_timeout_idle(msecs_to_jiffies(100));
 	return -ECANCELED;
 
 }
-- 
2.47.3

