From: SeongJae Park <sj@kernel.org>
Date: Mon, 3 Nov 2025 21:51:58 -0800
Subject: [PATCH] mm/damon/tests/core-kunit: add damon_commit_target_regions()
 test

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/tests/core-kunit.h | 63 +++++++++++++++++++++++++++++++++++++
 1 file changed, 63 insertions(+)

diff --git a/mm/damon/tests/core-kunit.h b/mm/damon/tests/core-kunit.h
index 17da4ed3e2a4..db9c54628ba4 100644
--- a/mm/damon/tests/core-kunit.h
+++ b/mm/damon/tests/core-kunit.h
@@ -964,6 +964,68 @@ static void damos_test_commit(struct kunit *test)
 			});
 }
 
+static void damon_test_commit_target_regions_for(struct kunit *test,
+		unsigned long dst_start_end[][2], int nr_dst_regions,
+		unsigned long src_start_end[][2], int nr_src_regions,
+		unsigned long expect_start_end[][2], int nr_expect_regions)
+{
+	struct damon_target *dst_target, *src_target;
+	struct damon_region *r;
+	int i;
+
+	dst_target = damon_new_target();
+	if (!dst_target)
+		kunit_skip(test, "dst target alloc fail");
+	src_target = damon_new_target();
+	if (!src_target) {
+		damon_free_target(dst_target);
+		kunit_skip(test, "src target alloc fail");
+	}
+	for (i = 0; i < nr_dst_regions; i++) {
+		r = damon_new_region(dst_start_end[i][0], dst_start_end[i][1]);
+		if (!r) {
+			damon_free_target(dst_target);
+			damon_free_target(src_target);
+			kunit_skip(test, "region alloc fail");
+		}
+		damon_add_region(r, dst_target);
+	}
+	for (i = 0; i < nr_src_regions; i++) {
+		r = damon_new_region(src_start_end[i][0], src_start_end[i][1]);
+		if (!r) {
+			damon_free_target(dst_target);
+			damon_free_target(src_target);
+			kunit_skip(test, "region alloc fail");
+		}
+		damon_add_region(r, src_target);
+	}
+
+	damon_commit_target_regions(dst_target, src_target, 1);
+
+	i = 0;
+	damon_for_each_region(r, dst_target) {
+		KUNIT_EXPECT_EQ(test, r->ar.start, expect_start_end[i][0]);
+		KUNIT_EXPECT_EQ(test, r->ar.end, expect_start_end[i][1]);
+		i++;
+	}
+	KUNIT_EXPECT_EQ(test, i, nr_expect_regions);
+
+	damon_free_target(dst_target);
+	damon_free_target(src_target);
+}
+
+static void damon_test_commit_target_regions(struct kunit *test)
+{
+	damon_test_commit_target_regions_for(test,
+			(unsigned long[][2]){{3, 8}, {8, 10}}, 2,
+			(unsigned long[][2]){{4, 6}}, 1,
+			(unsigned long[][2]){{4, 6}}, 1);
+	damon_test_commit_target_regions_for(test,
+			(unsigned long[][2]){{3, 8}, {8, 10}}, 2,
+			(unsigned long[][2]){}, 0,
+			(unsigned long[][2]){{3, 8}, {8, 10}}, 2);
+}
+
 static void damos_test_filter_out(struct kunit *test)
 {
 	struct damon_target *t;
@@ -1167,6 +1229,7 @@ static struct kunit_case damon_test_cases[] = {
 	KUNIT_CASE(damos_test_commit_dests),
 	KUNIT_CASE(damos_test_commit_filter),
 	KUNIT_CASE(damos_test_commit),
+	KUNIT_CASE(damon_test_commit_target_regions),
 	KUNIT_CASE(damos_test_filter_out),
 	KUNIT_CASE(damon_test_feed_loop_next_input),
 	KUNIT_CASE(damon_test_set_filters_default_reject),
-- 
2.47.3

