From: SeongJae Park <sj@kernel.org>
Date: Tue, 9 Jun 2026 22:29:44 -0700
Subject: [PATCH] mm/damon/core: ensure all contexts are stopped by
 damon_stop()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 2a709751494f1..d49740abb62b1 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2049,20 +2049,15 @@ static void __damon_stop(struct damon_ctx *ctx)
  * damon_stop() - Stops the monitorings for a given group of contexts.
  * @ctxs:	an array of the pointers for contexts to stop monitoring
  * @nr_ctxs:	size of @ctxs
- *
- * Return: 0 on success, negative error code otherwise.
  */
 int damon_stop(struct damon_ctx **ctxs, int nr_ctxs)
 {
-	int i, err = 0;
+	int i;
 
-	for (i = 0; i < nr_ctxs; i++) {
+	for (i = 0; i < nr_ctxs; i++)
 		/* nr_running_ctxs is decremented in kdamond_fn */
-		err = __damon_stop(ctxs[i]);
-		if (err)
-			break;
-	}
-	return err;
+		__damon_stop(ctxs[i]);
+	return 0;
 }
 
 /**
-- 
2.47.3

