From: SJ Park <sj@kernel.org>
Date: Fri, 12 Jun 2026 18:07:47 -0700
Subject: [PATCH] mm/damon: introduce damon_operations->prep_probes

Signed-off-by: SJ Park <sj@kernel.org>
---
 include/linux/damon.h | 4 ++++
 mm/damon/core.c       | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 466e8c002595c..15ab990afd3ab 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -683,6 +683,7 @@ enum damon_ops_id {
  * @update:			Update operations-related data structures.
  * @prepare_access_checks:	Prepare next access check of target regions.
  * @check_accesses:		Check the accesses to target regions.
+ * @prep_probes:		Prepare applying probes for each region.
  * @apply_probes:		Apply probes for each region.
  * @get_scheme_score:		Get the score of a region for a scheme.
  * @apply_scheme:		Apply a DAMON-based operation scheme.
@@ -710,6 +711,8 @@ enum damon_ops_id {
  * last preparation and update the number of observed accesses of each region.
  * It should also return max number of observed accesses that made as a result
  * of its update.  The value will be used for regions adjustment threshold.
+ * @prep_probes should execute required &struct damon_prep for next &struct
+ * damon_probe applications to each region.
  * @apply_probes should apply the data attribute probes to each region and
  * accordingly update the probe hits counter of the region.
  * @get_scheme_score should return the priority score of a region for a scheme
@@ -729,6 +732,7 @@ struct damon_operations {
 	void (*update)(struct damon_ctx *context);
 	void (*prepare_access_checks)(struct damon_ctx *context);
 	unsigned int (*check_accesses)(struct damon_ctx *context);
+	void (*prep_probes)(struct damon_ctx *context);
 	void (*apply_probes)(struct damon_ctx *context);
 	int (*get_scheme_score)(struct damon_ctx *context,
 			struct damon_region *r, struct damos *scheme);
diff --git a/mm/damon/core.c b/mm/damon/core.c
index ac93ac841d53b..59244270f7e71 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -4063,6 +4063,8 @@ static int kdamond_fn(void *data)
 
 		if (!ctx->has_probe_weights && ctx->ops.prepare_access_checks)
 			ctx->ops.prepare_access_checks(ctx);
+		if (ctx->ops.prep_probes)
+			ctx->ops.prep_probes(ctx);
 
 		kdamond_usleep(sample_interval);
 		ctx->passed_sample_intervals++;
-- 
2.47.3

