From: SJ Park <sj@kernel.org>
Date: Sat, 23 May 2026 13:01:03 -0700
Subject: [PATCH] mm/damon: mark damos_quota_goal->{last_psi_total,list} as
 private

Signed-off-by: SJ Park <sj@kernel.org>
---
 include/linux/damon.h | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index a04d2bfca349e..e0f4715a5ed5b 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -174,10 +174,8 @@ enum damos_quota_goal_metric {
  * @metric:		Metric to be used for representing the goal.
  * @target_value:	Target value of @metric to achieve with the tuning.
  * @current_value:	Current value of @metric.
- * @last_psi_total:	Last measured total PSI
  * @nid:		Node id.
  * @memcg_id:		Memcg id.
- * @list:		List head for siblings.
  *
  * Data structure for getting the current score of the quota tuning goal.  The
  * score is calculated by how close @current_value and @target_value are.  Then
@@ -200,12 +198,16 @@ struct damos_quota_goal {
 	unsigned long current_value;
 	/* metric-dependent fields */
 	union {
-		u64 last_psi_total;
 		struct {
 			int nid;
 			u64 memcg_id;
 		};
+/* private: */
+		/* Last measured total PSI */
+		u64 last_psi_total;
+
 	};
+	/* List head for siblings. */
 	struct list_head list;
 };
 
-- 
2.47.3

