From: SeongJae Park <sj@kernel.org>
Date: Sat, 2 Aug 2025 13:27:00 -0700
Subject: [PATCH] mm/damon/sysfs: pass write_only to core

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 8ac85d72c50e..7a18678a7a74 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1555,6 +1555,7 @@ static int damon_sysfs_apply_inputs(struct damon_ctx *ctx,
 	if (err)
 		return err;
 	ctx->ops_attrs.use_reports = sys_ctx->ops_attrs->use_reports;
+	ctx->ops_attrs.write_only = sys_ctx->ops_attrs->write_only;
 	ctx->addr_unit = sys_ctx->addr_unit;
 	/* addr_unit is respected by only DAMON_OPS_PADDR */
 	if (sys_ctx->ops_id == DAMON_OPS_PADDR)
-- 
2.39.5

