From: SeongJae Park <sj@kernel.org>
Date: Thu, 26 Mar 2026 06:56:56 -0700
Subject: [PATCH] mm/damon/core: fixup damon_call() race fix

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 724e059e810cd..1e93692f0ef31 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1794,6 +1794,9 @@ int damon_kdamond_pid(struct damon_ctx *ctx)
  * synchronization.  The return value of the function will be saved in
  * &damon_call_control->return_code.
  *
+ * Note that @ctx should be started (damon_start() with it has succeed.
+ * Otherwise, this function could indefinitely wait.
+ *
  * Return: 0 on success, negative error code otherwise.
  */
 int damon_call(struct damon_ctx *ctx, struct damon_call_control *control)
@@ -3219,9 +3222,6 @@ static void kdamond_init_ctx(struct damon_ctx *ctx)
 		damos_set_next_apply_sis(scheme, ctx);
 		damos_set_filters_default_reject(scheme);
 	}
-	mutex_lock(&ctx->call_controls_lock);
-	ctx->call_controls_obsolete = false;
-	mutex_unlock(&ctx->call_controls_lock);
 }
 
 static bool damon_sample_filter_matching(struct damon_access_report *report,
@@ -3348,6 +3348,9 @@ static int kdamond_fn(void *data)
 
 	pr_debug("kdamond (%d) starts\n", current->pid);
 
+	mutex_lock(&ctx->call_controls_lock);
+	ctx->call_controls_obsolete = false;
+	mutex_unlock(&ctx->call_controls_lock);
 	complete(&ctx->kdamond_started);
 	kdamond_init_ctx(ctx);
 
-- 
2.47.3

