From: SeongJae Park <sj@kernel.org>
Date: Fri, 15 May 2026 17:40:58 -0700
Subject: [PATCH] mm/damon: (fixup 6) add detailed kernel-doc comment for
 apply_probes

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index e432a504b7960..df7910a39b407 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -713,6 +713,8 @@ enum damon_ops_id {
  * last preparation and update the number of observed accesses of each region.
  * It should also return max number of observed accesses that made as a result
  * of its update.  The value will be used for regions adjustment threshold.
+ * @apply_probes should apply the data attribute probes to each region and
+ * accordingly update the probe hits counter of the region.
  * @get_scheme_score should return the priority score of a region for a scheme
  * as an integer in [0, &DAMOS_MAX_SCORE].
  * @apply_scheme is called from @kdamond when a region for user provided
-- 
2.47.3

