From: SeongJae Park <sj@kernel.org>
Date: Mon, 29 Dec 2025 21:07:33 -0800
Subject: [PATCH] mm/damon/core: fixup damon_call() uaf fix

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 23d8602f5a49..bf582140e428 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3278,13 +3278,13 @@ static int kdamond_fn(void *data)
 	if (ctx->ops.cleanup)
 		ctx->ops.cleanup(ctx);
 	kfree(ctx->regions_score_histogram);
+	kdamond_call(ctx, true);
 
 	pr_debug("kdamond (%d) finishes\n", current->pid);
 	mutex_lock(&ctx->kdamond_lock);
 	ctx->kdamond = NULL;
 	mutex_unlock(&ctx->kdamond_lock);
 
-	kdamond_call(ctx, true);
 	damos_walk_cancel(ctx);
 
 	mutex_lock(&damon_lock);
-- 
2.47.3

