From: SJ Park <sj@kernel.org>
Date: Sat, 29 Aug 2026 11:33:19 -0700
Subject: [PATCH] mm/damon/core: mark probe hits wsum damos filter as core
 filter

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 05d0253d68f71..0fded011b7423 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -663,6 +663,7 @@ bool damos_filter_for_ops(enum damos_filter_type type)
 	switch (type) {
 	case DAMOS_FILTER_TYPE_ADDR:
 	case DAMOS_FILTER_TYPE_TARGET:
+	case DAMOS_FILTER_TYPE_PROBE_HITS_WSUM:
 		return false;
 	default:
 		break;
-- 
2.47.3

