From: SeongJae Park <sj@kernel.org>
Date: Mon, 16 Mar 2026 23:48:36 -0700
Subject: [PATCH] mm/damon/core: fixup find_system_rams_range()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 625d4b63765d..acbd7dedcc8f 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3166,7 +3166,7 @@ static unsigned long damon_res_to_core_addr(resource_size_t ra,
 	return ra / addr_unit;
 }
 
-static bool damon_find_system_ram_range(unsigned long *start,
+static bool damon_find_system_rams_range(unsigned long *start,
 		unsigned long *end, unsigned long addr_unit)
 {
 	struct damon_system_ram_range_walk_arg arg = {};
@@ -3176,6 +3176,8 @@ static bool damon_find_system_ram_range(unsigned long *start,
 		return false;
 	*start = damon_res_to_core_addr(arg.res.start, addr_unit);
 	*end = damon_res_to_core_addr(arg.res.end + 1, addr_unit);
+	if (*end <= *start)
+		return false;
 	return true;
 }
 
@@ -3206,7 +3208,7 @@ int damon_set_region_system_rams_default(struct damon_target *t,
 		return -EINVAL;
 
 	if (!*start && !*end &&
-		!damon_find_system_ram_range(start, end, addr_unit))
+		!damon_find_system_rams_range(start, end, addr_unit))
 		return -EINVAL;
 
 	addr_range.start = *start;
-- 
2.47.3

