From: SeongJae Park <sj@kernel.org>
Date: Wed, 2 Jul 2025 20:10:07 -0700
Subject: [PATCH] mm/damon/core: fix wrong repeat mode damon_call()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index f23857562cd9..1db4049a6bf7 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2572,14 +2572,14 @@ static void kdamond_call(struct damon_ctx *ctx, bool cancel)
 		list_del(&control->list);
 		mutex_unlock(&ctx->call_controls_lock);
 		if (control->repeat)
-			list_add(&repeat_controls, &control->list);
+			list_add(&control->list, &repeat_controls);
 	}
 	control = list_first_entry_or_null(&repeat_controls,
 			struct damon_call_control, list);
 	if (!control || cancel)
 		return;
 	mutex_lock(&ctx->call_controls_lock);
-	list_add_tail(&ctx->call_controls, &control->list);
+	list_add_tail(&control->list, &ctx->call_controls);
 	mutex_unlock(&ctx->call_controls_lock);
 }
 
-- 
2.39.5

