From: SeongJae Park <sj@kernel.org>
Date: Fri, 15 May 2026 18:00:19 -0700
Subject: [PATCH] mm/damon/sysfs: (fixup 10) add static keyword for probe_ktype

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 7cfc5c780fb25..fc7256e522a69 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1105,7 +1105,7 @@ static struct attribute *damon_sysfs_probe_attrs[] = {
 };
 ATTRIBUTE_GROUPS(damon_sysfs_probe);
 
-const struct kobj_type damon_sysfs_probe_ktype = {
+static const struct kobj_type damon_sysfs_probe_ktype = {
 	.release = damon_sysfs_probe_release,
 	.sysfs_ops = &kobj_sysfs_ops,
 	.default_groups = damon_sysfs_probe_groups,
-- 
2.47.3

