From: SeongJae Park <sj@kernel.org>
Date: Sat, 3 Jan 2026 10:25:24 -0800
Subject: [PATCH] mm/damon/core: add damon_new_region() hardening

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 36ac0191671bb..183ac2816380e 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -116,6 +116,22 @@ int damon_select_ops(struct damon_ctx *ctx, enum damon_ops_id id)
 	return err;
 }
 
+#ifdef CONFIG_DAMON_HARDENED
+static void damon_verify_new_region(unsigned long start, unsigned long end)
+{
+
+	if (start < end)
+		return;
+	pr_err("damon_new_region() s called with start %lu and end %lu!\n",
+			start, end);
+	BUG();
+}
+#else
+static void damon_verify_new_region(unsigned long start, unsigned long end)
+{
+}
+#endif
+
 /*
  * Construct a damon_region struct
  *
@@ -129,6 +145,8 @@ struct damon_region *damon_new_region(unsigned long start, unsigned long end)
 	if (!region)
 		return NULL;
 
+	damon_verify_new_region(start, end);
+
 	region->ar.start = start;
 	region->ar.end = end;
 	region->nr_accesses = 0;
-- 
2.47.3

