From: SJ Park <sj@kernel.org>
Date: Tue, 4 Aug 2026 07:15:31 -0700
Subject: [PATCH] mm/damon/core: handle NULL ctx parameter in damon_call()

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 1d122a21e39e3..fa233dedcd993 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2369,6 +2369,8 @@ int damon_kdamond_pid(struct damon_ctx *ctx)
  */
 int damon_call(struct damon_ctx *ctx, struct damon_call_control *control)
 {
+	if (!ctx)
+		return -EINVAL;
 	if (!control->repeat)
 		init_completion(&control->completion);
 	control->canceled = false;
-- 
2.47.3

