From: SJ Park <sj@kernel.org>
Date: Sat, 25 Jul 2026 14:46:55 -0700
Subject: [PATCH] mm/damon/core: fix sample_addr for all cases (patch 7 fix)

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 760559d495b3d..0e9afc64df621 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -4251,7 +4251,7 @@ static int kdamond_fn(void *data)
 		unsigned long next_ops_update_sis = ctx->next_ops_update_sis;
 		unsigned long sample_interval = ctx->attrs.sample_interval;
 		bool access_check_disabled = damon_has_probe_weights(ctx);
-		bool has_prep = damon_has_prep(ctx);
+		bool do_prep = damon_has_prep(ctx) && ctx->ops.prep_probes;
 		unsigned int max_merge_score = 0, max_wsum;
 		bool get_max_wsum;
 
@@ -4260,8 +4260,8 @@ static int kdamond_fn(void *data)
 
 		if (!access_check_disabled && ctx->ops.prepare_access_checks)
 			ctx->ops.prepare_access_checks(ctx);
-		if (ctx->ops.prep_probes)
-			ctx->ops.prep_probes(ctx, has_prep);
+		if (do_prep)
+			ctx->ops.prep_probes(ctx, access_check_disabled);
 
 		kdamond_usleep(sample_interval);
 		ctx->passed_sample_intervals++;
@@ -4283,7 +4283,7 @@ static int kdamond_fn(void *data)
 			else
 				get_max_wsum = false;
 			max_wsum = ctx->ops.apply_probes(ctx,
-					access_check_disabled && !has_prep,
+					access_check_disabled && !do_prep,
 					get_max_wsum);
 			if (get_max_wsum)
 				max_merge_score = max_wsum;
-- 
2.47.3

