From: SJ Park <sj@kernel.org>
Date: Mon, 3 Aug 2026 07:40:28 -0700
Subject: [PATCH] mm/damon/core: error damon_call() for NULL ctx

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 0093f67cf6cee..a01cc70844cec 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2364,7 +2364,7 @@ int damon_kdamond_pid(struct damon_ctx *ctx)
  */
 int damon_call(struct damon_ctx *ctx, struct damon_call_control *control)
 {
-	if (!completion_done(ctx->kdamond_started))
+	if (!ctx || !completion_done(&ctx->kdamond_started))
 		return -EINVAL;
 	if (!control->repeat)
 		init_completion(&control->completion);
-- 
2.47.3

