From: SJ Park <sj@kernel.org>
Date: Sat, 11 Jul 2026 09:55:40 -0700
Subject: [PATCH] mm/damon/core: use damon_nr_samples_per_aggr() for max merge
 threshold

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index dd2abaca9511d..9530a43f15a42 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3707,8 +3707,7 @@ static void kdamond_merge_regions(struct damon_ctx *c, unsigned int threshold,
 	unsigned int max_thres;
 	bool count_age = true;
 
-	max_thres = c->attrs.aggr_interval /
-		(c->attrs.sample_interval ?  c->attrs.sample_interval : 1);
+	max_thres = damon_nr_samples_per_aggr(&c->attrs);
 	while (true) {
 		nr_regions = 0;
 		damon_for_each_target(t, c) {
-- 
2.47.3

