From: SeongJae Park <sj@kernel.org>
Date: Fri, 6 Mar 2026 16:20:54 -0800
Subject: [PATCH] mm/damon/core: fixup: drop unnecessary reset of goal_tuner

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index c5503fdb10bb7..39af6f6a8176b 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -432,7 +432,6 @@ struct damos *damon_new_scheme(struct damos_access_pattern *pattern,
 	scheme->quota = *(damos_quota_init(quota));
 	/* quota.goals should be separately set by caller */
 	INIT_LIST_HEAD(&scheme->quota.goals);
-	scheme->quota.goal_tuner = quota->goal_tuner;
 
 	scheme->wmarks = *wmarks;
 	scheme->wmarks.activated = true;
-- 
2.47.3

