From: SeongJae Park <sj@kernel.org>
Date: Sat, 18 Apr 2026 11:54:16 -0700
Subject: [PATCH] mm/damon: add damon_region->attr_counts

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index e9a5a40401bcb..dba172ce0f3d9 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -16,6 +16,8 @@
 
 /* Minimal region size.  Every damon_region is aligned by this. */
 #define DAMON_MIN_REGION_SZ	PAGE_SIZE
+/* Maximum number of data attributes that can be monitored. */
+#define DAMON_MAX_DATA_ATTRS	(4)
 /* Max priority score for DAMON-based operation schemes */
 #define DAMOS_MAX_SCORE		(99)
 
@@ -52,6 +54,7 @@ struct damon_size_range {
  * @nr_accesses:	Access frequency of this region.
  * @nr_accesses_bp:	@nr_accesses in basis point (0.01%) that updated for
  *			each sampling interval.
+ * @attr_counts:	Number of samples of each monitoring data attribute.
  * @list:		List head for siblings.
  * @age:		Age of this region.
  *
@@ -80,6 +83,7 @@ struct damon_region {
 	unsigned long sampling_addr;
 	unsigned int nr_accesses;
 	unsigned int nr_accesses_bp;
+	unsigned char attr_counts[DAMON_MAX_DATA_ATTRS];
 	struct list_head list;
 
 	unsigned int age;
-- 
2.47.3

