From: SeongJae Park <sj@kernel.org>
Date: Tue, 21 Jan 2025 16:58:44 -0800
Subject: [PATCH] mm/damon/core: fixup max aggregation interval logic

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 06dfc41fad79..a939eb6a74a2 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1305,7 +1305,7 @@ static void kdamond_tune_intervals(struct damon_ctx *c)
 	adaptation_bp = damon_feed_loop_next_input(10000, score_bp);
 
 	new_attrs = c->attrs;
-	new_attrs.aggr_interval = max(
+	new_attrs.aggr_interval = min(
 			c->attrs.aggr_interval * adaptation_bp / 10000,
 			c->attrs.max_aggr_interval);
 	sample_to_aggr_bp = c->attrs.sample_interval * 10000 /
-- 
2.39.5

