From: SeongJae Park <sj@kernel.org>
Date: Sun, 15 Dec 2024 12:43:53 -0800
Subject: [PATCH] mm/damon/paddr: support DAMOS filtered_out stat

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index e6be2659f18c..0ca1bfd3d300 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -240,8 +240,10 @@ static bool damos_pa_filter_out(struct damos *scheme, struct folio *folio)
 	struct damos_filter *filter;
 
 	damos_for_each_filter(filter, scheme) {
-		if (__damos_pa_filter_out(filter, folio))
+		if (__damos_pa_filter_out(filter, folio)) {
+			scheme->stat.sz_filtered_out += folio_size(folio);
 			return true;
+		}
 	}
 	return false;
 }
-- 
2.39.5

