From: SeongJae Park <sj@kernel.org>
Date: Fri, 7 Feb 2025 14:23:49 -0800
Subject: [PATCH] mm/damon/core: avoid too rapid inteervals reduction from
 auto-tuning

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index c19b010a0b41..b0c34189c177 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1323,6 +1323,12 @@ static unsigned long damon_get_intervals_adaptation_bp(struct damon_ctx *c)
 	score_bp = access_samples * 10000 / target_samples;
 	adaptation_bp = damon_feed_loop_next_input(100000000, score_bp) /
 		10000;
+	/*
+	 * adaptaion_bp ranges from 1 to 20,000.  Avoid too rapid reduction of
+	 * the intervals by rescaling [1,9999] to [5000, 9999].
+	 */
+	if (adaptation_bp < 10000)
+		adaptation_bp = 5000 + adaptation_bp / 2;
 
 	pr_info("access_samples %lu/%lu, score_bp %lu, adaptation bp %lu\n",
 			access_samples, target_samples, score_bp,
-- 
2.39.5

