From: SeongJae Park <sj@kernel.org>
Date: Sat, 26 Jul 2025 10:32:04 -0700
Subject: [PATCH] mm/damon/paddr: use MM_CP_PROT_DAMON

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index 5a731369cae4..5695415f811b 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -109,8 +109,7 @@ static bool damon_pa_fault_change_protection_one(struct folio *folio,
 
 	tlb_gather_mmu(&tlb, vma->vm_mm);
 
-	/* TODO: define and use MM_CP_PROT_DAMON */
-	change_protection(&tlb, vma, addr, addr + PAGE_SIZE, MM_CP_PROT_NUMA);
+	change_protection(&tlb, vma, addr, addr + PAGE_SIZE, MM_CP_PROT_DAMON);
 
 	tlb_finish_mmu(&tlb);
 	return true;
-- 
2.39.5

