From: SeongJae Park <sj@kernel.org>
Date: Wed, 24 Jun 2026 20:28:40 -0700
Subject: [PATCH] mm/damon/core: implement damon_probe->weight

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 1 +
 mm/damon/core.c       | 1 +
 2 files changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index add632de60370..7eea6ade3bd2d 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -856,6 +856,7 @@ struct damon_filter {
  * struct damon_probe - Data region attribute probe.
  */
 struct damon_probe {
+	unsigned int weight;
 /* private: */
 	/* Preparation actions to apply to each probing memory. */
 	struct list_head preps;
diff --git a/mm/damon/core.c b/mm/damon/core.c
index bfdfe47b837b3..a124c4174832a 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -220,6 +220,7 @@ struct damon_probe *damon_new_probe(void)
 	p = kmalloc_obj(*p);
 	if (!p)
 		return NULL;
+	p->weight = 0;
 	INIT_LIST_HEAD(&p->preps);
 	INIT_LIST_HEAD(&p->filters);
 	INIT_LIST_HEAD(&p->list);
-- 
2.47.3

