From: SeongJae Park <sj@kernel.org>
Date: Mon, 16 Dec 2024 19:40:08 -0800
Subject: [PATCH] mm/damon: revert DAMOS_STAT_FULL intro

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index f55f79e85e6d..8023d62516bf 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -113,7 +113,6 @@ struct damon_target {
  * @DAMOS_FREE:		Return DAMOS_ALLOC-ed pages back to the system.
 #endif
  * @DAMOS_STAT:		Do nothing but count the stat.
- * @DAMOS_STAT_FULL:	Do nothing but filtering and stat counting.
  * @NR_DAMOS_ACTIONS:	Total number of DAMOS actions
  *
  * The support of each action is up to running &struct damon_operations.
@@ -137,7 +136,6 @@ enum damos_action {
 	DAMOS_FREE,
 #endif
 	DAMOS_STAT,		/* Do nothing but only record the stat */
-	DAMOS_STAT_FULL,	/* Handle and account filters */
 	NR_DAMOS_ACTIONS,
 };
 
-- 
2.39.5

