From: SeongJae Park <sj@kernel.org>
Date: Sun, 23 Nov 2025 08:43:59 -0800
Subject: [PATCH] mm/damon: cleanup comment of max_nr_snapshots

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index c338dc21e2c0..ef1d22c98a0e 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -589,8 +589,8 @@ struct damos_migrate_dests {
  *
  * After applying the &action to each region, &stat is updated.
  *
- * If &max_nr_snapshots is set and &stat.nr_snapsghots exceeds it, the scheme
- * is deactivated.
+ * If &max_nr_snapshots is set as non-zero and &stat.nr_snapshots be same to or
+ * greater than it, the scheme is deactivated.
  */
 struct damos {
 	struct damos_access_pattern pattern;
-- 
2.47.3

