From: SeongJae Park <sj@kernel.org>
Date: Mon, 24 Nov 2025 07:39:35 -0800
Subject: [PATCH] mm/damon/core: use Steve-suggested code

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 2f212a18e4a0..bd4a6c2fd8e1 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2261,6 +2261,9 @@ static void damos_trace_stat(struct damon_ctx *c, struct damos *s)
 	unsigned int cidx = 0, sidx = 0;
 	struct damos *siter;
 
+	if (!trace_damos_stat_after_apply_interval_enabled())
+		return;
+
 	damon_for_each_scheme(siter, c) {
 		if (siter == s)
 			break;
@@ -2307,8 +2310,7 @@ static void kdamond_apply_schemes(struct damon_ctx *c)
 			(s->apply_interval_us ? s->apply_interval_us :
 			 c->attrs.aggr_interval) / sample_interval;
 		s->last_applied = NULL;
-		if (trace_damos_stat_after_apply_interval_enabled())
-			damos_trace_stat(c, s);
+		damos_trace_stat(c, s);
 	}
 	mutex_unlock(&c->walk_control_lock);
 }
-- 
2.47.3

