From: SeongJae Park <sj@kernel.org>
Date: Wed, 3 Jun 2026 22:44:13 -0700
Subject: [PATCH] mm/damon/core: (fixup) further cleanup

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 14bd9d4307c07..7e4b9affc5b06 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1388,15 +1388,15 @@ static int damon_commit_target(
 }
 
 /*
- * damon_commit_target_fail() - handle damon_commit_target() failure.
+ * damon_revert_target_commits() - revert unsuccessful target commits.
  * @dst:	Commit destination context
  * @failed:	Commit failed destination target
  * @src:	Commit source context
  *
- * This function is called by damon_commit_targets() for dammon_commit_target()
- * failure for immediate states cleanups.
+ * Revert target states that changed by damon_commit_target(), and cannot be
+ * cleaned up by the destination context's ops.cleanup_target().
  */
-static void damon_commit_target_fail(struct damon_ctx *dst,
+static void damon_revert_target_commits(struct damon_ctx *dst,
 		struct damon_target *failed, struct damon_ctx *src)
 {
 	struct damon_target *target;
@@ -1473,7 +1473,7 @@ static int damon_commit_targets(
 	return 0;
 
 out:
-	damon_commit_target_fail(dst, failed, src);
+	damon_revert_target_commits(dst, failed, src);
 	return err;
 }
 
@@ -1609,7 +1609,7 @@ int damon_commit_ctx(struct damon_ctx *dst, struct damon_ctx *src)
 	if (!damon_attrs_equals(&dst->attrs, &src->attrs)) {
 		err = damon_set_attrs(dst, &src->attrs);
 		if (err) {
-			damon_commit_target_fail(dst, NULL, src);
+			damon_revert_target_commits(dst, NULL, src);
 			return err;
 		}
 	}
-- 
2.47.3

