From: SeongJae Park <sj@kernel.org>
Date: Sat, 2 Aug 2025 13:06:29 -0700
Subject: [PATCH] mm/damon: add operations_attrs->write_only

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 7d745164adae..1cb04aaf23a6 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -659,9 +659,11 @@ struct damon_operations {
  * struct damon_operations_attrs - Monitoring operations control attributes.
  *
  * @use_reports:	Whether to use damon_report_access()-ed information.
+ * @write_only:		Ignore none-write accesses.
  */
 struct damon_operations_attrs {
 	bool use_reports;
+	bool write_only;
 };
 
 /*
-- 
2.39.5

