From: SeongJae Park <sj@kernel.org>
Date: Sat, 6 Sep 2025 19:15:31 -0700
Subject: [PATCH] mm/damon/tests/core-kunit: test all quota goal metrics

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/tests/core-kunit.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/mm/damon/tests/core-kunit.h b/mm/damon/tests/core-kunit.h
index 5ace327d364d..73662ab8064c 100644
--- a/mm/damon/tests/core-kunit.h
+++ b/mm/damon/tests/core-kunit.h
@@ -514,6 +514,19 @@ static void damos_test_commit_quota_goal(struct kunit *test)
 		kunit_skip(test, "NODE_MEM_FREE_BP test error\n");
 		return;
 	}
+	if (damos_test_commit_quota_goal_for(test, dst,
+				DAMOS_QUOTA_NODE_MEM_USED_BP, 12, 345, 6)) {
+		kfree(dst);
+		kunit_skip(test, "NODE_MEM_USED_BP test error\n");
+		return;
+	}
+	if (damos_test_commit_quota_goal_for(test, dst,
+				DAMOS_QUOTA_USER_INPUT, 789, 12, 0)) {
+		kfree(dst);
+		kunit_skip(test, "USER_INPUT test error\n");
+		return;
+	}
+
 	damos_destroy_quota_goal(dst);
 }
 
-- 
2.47.3

