From: SeongJae Park <sj@kernel.org>
Date: Fri, 15 May 2026 17:30:51 -0700
Subject: [PATCH] mm/damon/core: (fixup 5) copy probe_hits when split region

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 078a53e26e852..8c0d2524e956b 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3114,6 +3114,8 @@ static void damon_split_region_at(struct damon_target *t,
 	new->last_nr_accesses = r->last_nr_accesses;
 	new->nr_accesses_bp = r->nr_accesses_bp;
 	new->nr_accesses = r->nr_accesses;
+	/* todo: do this for only installed probes */
+	memcpy(new->probe_hits, r->probe_hits, sizeof(r->probe_hits));
 
 	damon_insert_region(new, r, damon_next_region(r), t);
 }
-- 
2.47.3

