From: SeongJae Park <sj@kernel.org>
Date: Mon, 16 Feb 2026 14:33:38 -0800
Subject: [PATCH] mm/damon/core: keep old condition

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 12402f5df5d7c..f2b360142b0c8 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1310,7 +1310,7 @@ static unsigned long damon_region_sz_limit(struct damon_ctx *ctx)
 
 	if (ctx->attrs.min_nr_regions)
 		sz /= ctx->attrs.min_nr_regions;
-	if (!sz)
+	if (sz < ctx->min_region_sz)
 		sz = ctx->min_region_sz;
 	sz = ALIGN(sz, ctx->min_region_sz);
 
-- 
2.47.3

