From: SeongJae Park <sj@kernel.org>
Date: Sat, 25 Apr 2026 09:07:30 -0700
Subject: [PATCH] mm/damon/sysfs: further rename

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 82 ++++++++++++++++++++++++------------------------
 1 file changed, 41 insertions(+), 41 deletions(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 6575cb458d9b4..809c6095bbe6a 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1068,12 +1068,12 @@ const struct kobj_type damon_sysfs_probe_ktype = {
 };
 
 /*
- * coutners directory
+ * probes directory
  */
 
 struct damon_sysfs_probes {
 	struct kobject kobj;
-	struct damon_sysfs_probe **attrs_arr;
+	struct damon_sysfs_probe **probes_arr;
 	int nr;
 };
 
@@ -1083,78 +1083,78 @@ static struct damon_sysfs_probes *damon_sysfs_probes_alloc(void)
 }
 
 static void damon_sysfs_probes_rm_dirs(
-		struct damon_sysfs_probes *attrs)
+		struct damon_sysfs_probes *probes)
 {
-	struct damon_sysfs_probe **attrs_arr = attrs->attrs_arr;
+	struct damon_sysfs_probe **probes_arr = probes->probes_arr;
 	int i;
 
-	for (i = 0; i < attrs->nr; i++) {
-		damon_sysfs_probe_rm_dirs(attrs_arr[i]);
-		kobject_put(&attrs_arr[i]->kobj);
+	for (i = 0; i < probes->nr; i++) {
+		damon_sysfs_probe_rm_dirs(probes_arr[i]);
+		kobject_put(&probes_arr[i]->kobj);
 	}
-	attrs->nr = 0;
-	kfree(attrs_arr);
-	attrs->attrs_arr = NULL;
+	probes->nr = 0;
+	kfree(probes_arr);
+	probes->probes_arr = NULL;
 }
 
 static int damon_sysfs_probes_add_dirs(
-		struct damon_sysfs_probes *attrs, int nr_attrs)
+		struct damon_sysfs_probes *probes, int nr_probes)
 {
-	struct damon_sysfs_probe **attrs_arr, *attr;
+	struct damon_sysfs_probe **probes_arr, *probe;
 	int err, i;
 
-	damon_sysfs_probes_rm_dirs(attrs);
-	if (!nr_attrs)
+	damon_sysfs_probes_rm_dirs(probes);
+	if (!nr_probes)
 		return 0;
 
-	attrs_arr = kmalloc_objs(*attrs_arr, nr_attrs,
+	probes_arr = kmalloc_objs(*probes_arr, nr_probes,
 				   GFP_KERNEL | __GFP_NOWARN);
-	if (!attrs_arr)
+	if (!probes_arr)
 		return -ENOMEM;
-	attrs->attrs_arr = attrs_arr;
+	probes->probes_arr = probes_arr;
 
-	for (i = 0; i < nr_attrs; i++) {
-		attr = damon_sysfs_probe_alloc();
-		if (!attr) {
-			damon_sysfs_probes_rm_dirs(attrs);
+	for (i = 0; i < nr_probes; i++) {
+		probe = damon_sysfs_probe_alloc();
+		if (!probe) {
+			damon_sysfs_probes_rm_dirs(probes);
 			return -ENOMEM;
 		}
 
-		err = kobject_init_and_add(&attr->kobj,
-				&damon_sysfs_probe_ktype, &attrs->kobj,
+		err = kobject_init_and_add(&probe->kobj,
+				&damon_sysfs_probe_ktype, &probes->kobj,
 				"%d", i);
 		if (err) {
-			kobject_put(&attr->kobj);
-			damon_sysfs_probes_rm_dirs(attrs);
+			kobject_put(&probe->kobj);
+			damon_sysfs_probes_rm_dirs(probes);
 			return err;
 		}
 
-		err = damon_sysfs_probe_add_dirs(attr);
+		err = damon_sysfs_probe_add_dirs(probe);
 		if (err) {
-			kobject_put(&attr->kobj);
-			damon_sysfs_probes_rm_dirs(attrs);
+			kobject_put(&probe->kobj);
+			damon_sysfs_probes_rm_dirs(probes);
 			return err;
 		}
 
-		attrs_arr[i] = attr;
-		attrs->nr++;
+		probes_arr[i] = probe;
+		probes->nr++;
 	}
 	return 0;
 }
 
-static ssize_t nr_attrs_show(struct kobject *kobj,
+static ssize_t nr_probes_show(struct kobject *kobj,
 		struct kobj_attribute *attr, char *buf)
 {
-	struct damon_sysfs_probes *attrs = container_of(kobj,
+	struct damon_sysfs_probes *probes = container_of(kobj,
 			struct damon_sysfs_probes, kobj);
 
-	return sysfs_emit(buf, "%d\n", attrs->nr);
+	return sysfs_emit(buf, "%d\n", probes->nr);
 }
 
-static ssize_t nr_attrs_store(struct kobject *kobj,
+static ssize_t nr_probes_store(struct kobject *kobj,
 		struct kobj_attribute *attr, const char *buf, size_t count)
 {
-	struct damon_sysfs_probes *attrs;
+	struct damon_sysfs_probes *probes;
 	int nr, err = kstrtoint(buf, 0, &nr);
 
 	if (err)
@@ -1162,11 +1162,11 @@ static ssize_t nr_attrs_store(struct kobject *kobj,
 	if (nr < 0)
 		return -EINVAL;
 
-	attrs = container_of(kobj, struct damon_sysfs_probes, kobj);
+	probes = container_of(kobj, struct damon_sysfs_probes, kobj);
 
 	if (!mutex_trylock(&damon_sysfs_lock))
 		return -EBUSY;
-	err = damon_sysfs_probes_add_dirs(attrs, nr);
+	err = damon_sysfs_probes_add_dirs(probes, nr);
 	mutex_unlock(&damon_sysfs_lock);
 	if (err)
 		return err;
@@ -1179,11 +1179,11 @@ static void damon_sysfs_probes_release(struct kobject *kobj)
 	kfree(container_of(kobj, struct damon_sysfs_probes, kobj));
 }
 
-static struct kobj_attribute damon_sysfs_probes_nr_attr =
-		__ATTR_RW_MODE(nr_attrs, 0600);
+static struct kobj_attribute damon_sysfs_probes_nr_probes =
+		__ATTR_RW_MODE(nr_probes, 0600);
 
 static struct attribute *damon_sysfs_probes_attrs[] = {
-	&damon_sysfs_probes_nr_attr.attr,
+	&damon_sysfs_probes_nr_probes.attr,
 	NULL,
 };
 ATTRIBUTE_GROUPS(damon_sysfs_probes);
@@ -1884,7 +1884,7 @@ static int damon_sysfs_set_probes(struct damon_ctx *ctx,
 
 	for (i = 0; i < sys_probes->nr; i++) {
 		struct damon_sysfs_filters *sys_filters =
-			sys_probes->attrs_arr[i]->filters;
+			sys_probes->probes_arr[i]->filters;
 		struct damon_probe *c;
 		int j;
 
-- 
2.47.3

