From: SeongJae Park <sj@kernel.org>
Date: Sat, 29 Nov 2025 11:10:54 -0800
Subject: [PATCH] mm/damon/core: deallocate report filters when destroying
 context

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 0518f94684be..1c02aad97ac8 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -664,12 +664,17 @@ static void damon_destroy_targets(struct damon_ctx *ctx)
 void damon_destroy_ctx(struct damon_ctx *ctx)
 {
 	struct damos *s, *next_s;
+	struct damon_report_filter *f, *next_f;
 
 	damon_destroy_targets(ctx);
 
 	damon_for_each_scheme_safe(s, next_s, ctx)
 		damon_destroy_scheme(s);
 
+	damon_for_each_report_filter_safe(f, next_f,
+			(&ctx->access_check_control))
+		damon_destroy_report_filter(f, &ctx->access_check_control);
+
 	if (ctx->ops_attrs.nr_tids)
 		kfree(ctx->ops_attrs.tids);
 
-- 
2.47.3

