From: SeongJae Park <sj@kernel.org>
Date: Sat, 18 Apr 2026 08:38:59 -0700
Subject: [PATCH] mm/damon/reclaim: fixup kdamond_pid set

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/reclaim.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/mm/damon/reclaim.c b/mm/damon/reclaim.c
index 8821c28a55832..7126d47fb8b2f 100644
--- a/mm/damon/reclaim.c
+++ b/mm/damon/reclaim.c
@@ -412,6 +412,16 @@ module_param_cb(enabled, &enabled_param_ops, &enabled, 0600);
 MODULE_PARM_DESC(enabled,
 	"Enable or disable DAMON_RECLAIM (default: disabled)");
 
+static int damon_reclaim_kdamond_pid_store(const char *val,
+		const struct kernel_param *kp)
+{
+	/*
+	 * kdamond_pid is read-only, but kernel command line could write it.
+	 * Do nothing here.
+	 */
+	return 0;
+}
+
 static int damon_reclaim_kdamond_pid_load(char *buffer,
 		const struct kernel_param *kp)
 {
@@ -426,6 +436,7 @@ static int damon_reclaim_kdamond_pid_load(char *buffer,
 }
 
 static const struct kernel_param_ops kdamond_pid_param_ops = {
+	.set = damon_reclaim_kdamond_pid_store,
 	.get = damon_reclaim_kdamond_pid_load,
 };
 
-- 
2.47.3

