From 19e83abda7d617c720419644d12d2aeaf8edeecc Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Sat, 4 May 2024 15:23:05 -0700
Subject: [PATCH] mm/damon/paddr: implement DAMOS_STEAL action

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index 18797c1b419b..ccdfc627596a 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -325,6 +325,31 @@ static unsigned long damon_pa_deactivate_pages(struct damon_region *r,
 	return damon_pa_mark_accessed_or_deactivate(r, s, false);
 }
 
+#ifdef CONFIG_ACMA
+
+static unsigned long damon_pa_steal(struct damon_region *r, struct damos *s)
+{
+	unsigned long pfn;
+	unsigned long applied = 0;
+
+	for (pfn = PHYS_PFN(r->start); pfn < PHYS_PFN(r->end); pfn += 512) {
+		/* allocate the contig pages */
+		if (alloc_contig_range(pfn, pfn + 512, MIGRATE_MOVABLE,
+				GFP_KERNEL))
+			continue;
+		/* report as free to use */
+		if (page_report(pfn, 512)) {
+			free_contig_range(pfn, 512);
+			continue;
+		}
+		stolen_pfns[nr_stolen_pfns++] = pfn;
+		applied += 1;
+	}
+	return applied * PAGE_SIZE * 512;
+}
+
+#endif
+
 static unsigned long damon_pa_apply_scheme(struct damon_ctx *ctx,
 		struct damon_target *t, struct damon_region *r,
 		struct damos *scheme)
-- 
2.39.2

