From: SJ Park <sj@kernel.org>
Date: Mon, 22 Jun 2026 20:42:41 -0700
Subject: [PATCH] mm/damon/sysfs: support DAMON_FILTER_TYPE_PGIDLE

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index d08ddfc2c0b7f..952a6a64e1dd7 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -782,6 +782,10 @@ damon_sysfs_filter_type_names[] = {
 		.type = DAMON_FILTER_TYPE_MEMCG,
 		.name = "memcg",
 	},
+	{
+		.type = DAMON_FILTER_TYPE_PGIDLE,
+		.name = "pgidle",
+	},
 };
 
 static ssize_t type_show(struct kobject *kobj,
-- 
2.47.3

