From: SeongJae Park <sj@kernel.org>
Date: Mon, 16 Dec 2024 19:57:03 -0800
Subject: [PATCH] mm/damon: fixup kernel-doc comment for apply_scheme operation

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 31efec092017..be78a0cc8a47 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -567,8 +567,8 @@ enum damon_ops_id {
  * @apply_scheme is called from @kdamond when a region for user provided
  * DAMON-based operation scheme is found.  It should apply the scheme's action
  * to the region and return bytes of the region that the action is successfully
- * applied.  It should also report how many bytes of the region are excluded
- * due to &struct damos_filter.
+ * applied.  It should also report how many bytes of the region has passed
+ * filters (&struct damos_filter) that handled by itself.
  * @target_valid should check whether the target is still valid for the
  * monitoring.
  * @cleanup is called from @kdamond just before its termination.
-- 
2.39.5

