From: SJ Park <sj@kernel.org>
Date: Sat, 4 Jul 2026 14:28:36 -0700
Subject: [PATCH] mm/damon/core: use max_wsum as merge threshold

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index f9c81dc44a4ce..5a3462bc22d74 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -4170,7 +4170,7 @@ static int kdamond_fn(void *data)
 		unsigned long sample_interval = ctx->attrs.sample_interval;
 		bool access_check_disabled = damon_has_probe_weights(ctx);
 		bool has_prep = damon_has_prep(ctx);
-		unsigned int max_merge_score = 0;
+		unsigned int max_merge_score = 0, max_wsum;
 
 		if (kdamond_wait_activation(ctx))
 			break;
@@ -4192,9 +4192,12 @@ static int kdamond_fn(void *data)
 				max_merge_score = ctx->ops.check_accesses(ctx);
 		}
 
-		if (ctx->ops.apply_probes)
-			ctx->ops.apply_probes(ctx, access_check_disabled &&
-					!has_prep);
+		if (ctx->ops.apply_probes) {
+			max_wsum = ctx->ops.apply_probes(ctx,
+					access_check_disabled && !has_prep);
+			if (access_check_disabled)
+				max_merge_score = max_wsum;
+		}
 
 		if (time_after_eq(ctx->passed_sample_intervals,
 					next_aggregation_sis)) {
-- 
2.47.3

