From: SeongJae Park <sj@kernel.org>
Date: Sun, 27 Jul 2025 13:10:15 -0700
Subject: [PATCH] mm/mprotect: fix typo: ARCH_SUPPORTS -> CONFIG_ARCH_SUPPORTS

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/mprotect.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/mprotect.c b/mm/mprotect.c
index 56c61b7d0d2f..e8a76114e4f9 100644
--- a/mm/mprotect.c
+++ b/mm/mprotect.c
@@ -714,7 +714,7 @@ long change_protection(struct mmu_gather *tlb,
 	WARN_ON_ONCE(cp_flags & MM_CP_PROT_NUMA);
 #endif
 
-#ifdef ARCH_SUPPORTS_NUMA_BALANCING
+#ifdef CONFIG_ARCH_SUPPORTS_NUMA_BALANCING
 	if (cp_flags & MM_CP_DAMON)
 		newprot = PAGE_NONE;
 #endif
-- 
2.39.5

