From: SeongJae Park <sj@kernel.org>
Date: Mon, 17 Nov 2025 22:08:38 -0800
Subject: [PATCH] mm/memory: implement functions and data structures for page
 faults monitoring

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/internal.h |  9 +++++++++
 mm/memory.c   | 24 ++++++++++++++++++++++++
 2 files changed, 33 insertions(+)

diff --git a/mm/internal.h b/mm/internal.h
index d67e8bb757349..0e7c828286df5 100644
--- a/mm/internal.h
+++ b/mm/internal.h
@@ -1770,4 +1770,13 @@ static inline int io_remap_pfn_range_complete(struct vm_area_struct *vma,
 	return remap_pfn_range_complete(vma, addr, pfn, size, prot);
 }
 
+struct faults_monitor_control {
+	void (*pte_fault)(struct vm_fault *vmf);
+	void (*huge_pmd_fault)(struct vm_fault *vmf);
+	struct list_head list;
+};
+
+void faults_monitor_register(struct faults_monitor_control *c);
+void faults_monitor_unregister(struct faults_monitor_control *c);
+
 #endif	/* __MM_INTERNAL_H */
diff --git a/mm/memory.c b/mm/memory.c
index e989a452b9062..a5ef610939b33 100644
--- a/mm/memory.c
+++ b/mm/memory.c
@@ -7548,3 +7548,27 @@ void vma_pgtable_walk_end(struct vm_area_struct *vma)
 	if (is_vm_hugetlb_page(vma))
 		hugetlb_vma_unlock_read(vma);
 }
+
+static LIST_HEAD(faults_monitor_controls);
+DEFINE_SPINLOCK(faults_monitor_controls_lock);
+
+void faults_monitor_register(struct faults_monitor_control *control)
+{
+	spin_lock(&faults_monitor_controls_lock);
+	list_add_tail(&control->list, &faults_monitor_controls);
+	spin_unlock(&faults_monitor_controls_lock);
+}
+
+void faults_monitor_unregister(struct faults_monitor_control *control)
+{
+	struct faults_monitor_control *c, *next;
+
+	spin_lock(&faults_monitor_controls_lock);
+	list_for_each_entry_safe(c, next, &faults_monitor_controls, list) {
+		if (c != control)
+			continue;
+		list_del(&control->list);
+		break;
+	}
+	spin_unlock(&faults_monitor_controls_lock);
+}
-- 
2.47.3

