From: SeongJae Park <sj@kernel.org>
Date: Sat, 26 Jul 2025 10:30:53 -0700
Subject: [PATCH] mm/mprotect: use PAGE_NONE for change_protection(...,
 MM_CP_PROT_DAMON)

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/mprotect.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/mm/mprotect.c b/mm/mprotect.c
index 15af72b8e302..1b5f5e08acde 100644
--- a/mm/mprotect.c
+++ b/mm/mprotect.c
@@ -721,6 +721,11 @@ long change_protection(struct mmu_gather *tlb,
 	WARN_ON_ONCE(cp_flags & MM_CP_PROT_NUMA);
 #endif
 
+#ifdef CONFIG_DAMON
+	if (cp_flags & MM_CP_PROT_DAMON)
+		newprot = PAGE_NONE;
+#endif
+
 	if (is_vm_hugetlb_page(vma))
 		pages = hugetlb_change_protection(vma, start, end, newprot,
 						  cp_flags);
-- 
2.39.5

