From: SeongJae Park <sj@kernel.org>
Date: Sat, 2 Aug 2025 13:11:43 -0700
Subject: [PATCH] mm/memory: set damon_access_report->is_write from
 do_damon_page()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/memory.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/memory.c b/mm/memory.c
index ae93d012d819..50acd705b3d6 100644
--- a/mm/memory.c
+++ b/mm/memory.c
@@ -6170,6 +6170,7 @@ static vm_fault_t do_damon_page(struct vm_fault *vmf, bool huge_pmd)
 	struct damon_access_report access_report = {
 		.addr = vmf->address,
 		.size = 1,
+		.is_write = vmf->flags & FAULT_FLAG_WRITE,
 	};
 	struct vm_area_struct *vma = vmf->vma;
 	struct folio *folio;
-- 
2.39.5

