From: SeongJae Park <sj@kernel.org>
Date: Sat, 26 Jul 2025 13:50:29 -0700
Subject: [PATCH] mm/mprotect: remove test-only code

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/mprotect.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/mm/mprotect.c b/mm/mprotect.c
index 2eb2f5402e39..8b830755fec1 100644
--- a/mm/mprotect.c
+++ b/mm/mprotect.c
@@ -38,9 +38,6 @@
 #include <asm/tlbflush.h>
 #include <asm/tlb.h>
 
-/* For PADDR_FAULT testing only */
-#include <linux/damon.h>
-
 #include "internal.h"
 
 static bool maybe_change_pte_writable(struct vm_area_struct *vma, pte_t pte)
-- 
2.39.5

