From: SeongJae Park <sj@kernel.org>
Date: Sun, 3 Aug 2025 11:15:18 -0700
Subject: [PATCH] mm/memory: set damon_access_report->cpu

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/memory.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/memory.c b/mm/memory.c
index 2f3c8f7aeeb6..e46b96ac9334 100644
--- a/mm/memory.c
+++ b/mm/memory.c
@@ -6179,6 +6179,7 @@ static vm_fault_t do_damon_page(struct vm_fault *vmf, bool huge_pmd)
 		.addr = vmf->address,
 		.size = 1,
 		.is_write = vmf->flags & FAULT_FLAG_WRITE,
+		.cpu = smp_processor_id(),
 	};
 	struct vm_area_struct *vma = vmf->vma;
 	struct folio *folio;
-- 
2.47.3

