From: SeongJae Park <sj@kernel.org>
Date: Wed, 28 Aug 2024 01:59:30 -0700
Subject: [PATCH] mm/vma: fixup build on CONFIG_MMU unset case

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/vma.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/vma.h b/mm/vma.h
index d85a99a6742b..44f9c5584ffb 100644
--- a/mm/vma.h
+++ b/mm/vma.h
@@ -136,6 +136,7 @@ int vma_expand(struct vma_merge_struct *vmg);
 int vma_shrink(struct vma_iterator *vmi, struct vm_area_struct *vma,
 	       unsigned long start, unsigned long end, pgoff_t pgoff);
 
+#ifdef CONFIG_MMU
 /*
  * init_vma_munmap() - Initializer wrapper for vma_munmap_struct
  * @vms: The vma munmap struct
@@ -169,6 +170,7 @@ static inline void init_vma_munmap(struct vma_munmap_struct *vms,
 	vms->clear_ptes = false;
 	vms->closed_vm_ops = false;
 }
+#endif
 
 int vms_gather_munmap_vmas(struct vma_munmap_struct *vms,
 		struct ma_state *mas_detach);
-- 
2.39.2

