From: Andrew Morton <akpm@linux-foundation.org>
Date: Mon, 28 Apr 2025 02:19:47 -0700
Subject: [PATCH] mm-zblock-avoid-failing-the-build-fix

s/#warn/#warning/

Cc: Igor Belousov <igor.b@beldev.am>
Cc: David Hildenbrand <david@redhat.com>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Cc: Nhat Pham <nphamcs@gmail.com>
Cc: Shakeel Butt <shakeel.butt@linux.dev>
Cc: Vitaly Wool <vitaly.wool@konsulko.se>
Cc: Yosry Ahmed <yosry.ahmed@linux.dev>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 mm/zblock.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/zblock.h b/mm/zblock.h
index 8610196b651e..9af11f392f97 100644
--- a/mm/zblock.h
+++ b/mm/zblock.h
@@ -21,7 +21,7 @@
 /* max 256 slots per block, max table size 64 */
 #define SLOT_BITS 8
 #else
-#warn This PAGE_SIZE is not quite supported yet
+#warning This PAGE_SIZE is not quite supported yet
 #define SLOT_BITS 8
 #endif
 
-- 
2.39.5

