From: SeongJae Park <sj@kernel.org>
Date: Sat, 25 Apr 2026 12:00:16 -0700
Subject: [PATCH] revert mm/damon/sysfs: implement filters directory

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 65 +-----------------------------------------------
 1 file changed, 1 insertion(+), 64 deletions(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 250208d3de21b..fac4e37aa2851 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -747,42 +747,12 @@ static const struct kobj_type damon_sysfs_intervals_ktype = {
 	.default_groups = damon_sysfs_intervals_groups,
 };
 
-/*
- * filters directory
- */
-
-struct damon_sysfs_filters {
-	struct kobject kobj;
-};
-
-static struct damon_sysfs_filters *damon_sysfs_filters_alloc(void)
-{
-	return kzalloc_obj(struct damon_sysfs_filters);
-}
-
-static void damon_sysfs_filters_release(struct kobject *kobj)
-{
-	kfree(container_of(kobj, struct damon_sysfs_filters, kobj));
-}
-
-static struct attribute *damon_sysfs_filters_attrs[] = {
-	NULL,
-};
-ATTRIBUTE_GROUPS(damon_sysfs_filters);
-
-static const struct kobj_type damon_sysfs_filters_ktype = {
-	.release = damon_sysfs_filters_release,
-	.sysfs_ops = &kobj_sysfs_ops,
-	.default_groups = damon_sysfs_filters_groups,
-};
-
 /*
  * probe directory
  */
 
 struct damon_sysfs_probe {
 	struct kobject kobj;
-	struct damon_sysfs_filters *filters;
 };
 
 static struct damon_sysfs_probe *damon_sysfs_probe_alloc(void)
@@ -790,30 +760,6 @@ static struct damon_sysfs_probe *damon_sysfs_probe_alloc(void)
 	return kzalloc_obj(struct damon_sysfs_probe);
 }
 
-static int damon_sysfs_probe_add_dirs(struct damon_sysfs_probe *attr)
-{
-	struct damon_sysfs_filters *filters;
-	int err;
-
-	filters = damon_sysfs_filters_alloc();
-	if (!filters)
-		return -ENOMEM;
-	attr->filters = filters;
-
-	err = kobject_init_and_add(&filters->kobj, &damon_sysfs_filters_ktype,
-			&attr->kobj, "filters");
-	if (err) {
-		kobject_put(&filters->kobj);
-		attr->filters = NULL;
-	}
-	return err;
-}
-
-static void damon_sysfs_probe_rm_dirs(struct damon_sysfs_probe *attr)
-{
-	kobject_put(&attr->filters->kobj);
-}
-
 static void damon_sysfs_probe_release(struct kobject *kobj)
 {
 	kfree(container_of(kobj, struct damon_sysfs_probe, kobj));
@@ -851,10 +797,8 @@ static void damon_sysfs_probes_rm_dirs(
 	struct damon_sysfs_probe **probes_arr = probes->probes_arr;
 	int i;
 
-	for (i = 0; i < probes->nr; i++) {
-		damon_sysfs_probe_rm_dirs(probes_arr[i]);
+	for (i = 0; i < probes->nr; i++)
 		kobject_put(&probes_arr[i]->kobj);
-	}
 	probes->nr = 0;
 	kfree(probes_arr);
 	probes->probes_arr = NULL;
@@ -892,13 +836,6 @@ static int damon_sysfs_probes_add_dirs(
 			return err;
 		}
 
-		err = damon_sysfs_probe_add_dirs(probe);
-		if (err) {
-			kobject_put(&probe->kobj);
-			damon_sysfs_probes_rm_dirs(probes);
-			return err;
-		}
-
 		probes_arr[i] = probe;
 		probes->nr++;
 	}
-- 
2.47.3

