From: SeongJae Park <sj@kernel.org>
Date: Tue, 2 Dec 2025 21:53:35 -0800
Subject: [PATCH] revert "mm/damon/core: report access-generated thread id"

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index b287ed961336..d2f842d8afd2 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1586,7 +1586,6 @@ void damon_report_page_fault(struct vm_fault *vmf, bool huge_pmd)
 	struct damon_access_report access_report = {
 		.size = 1,	/* todo: set appripriately */
 		.cpu = smp_processor_id(),
-		.tid = task_pid_vnr(current),
 	};
 
 	if (huge_pmd)
-- 
2.47.3

