From: SeongJae Park <sj@kernel.org>
Date: Sat, 25 Apr 2026 12:33:01 -0700
Subject: [PATCH] revert probe helper fixup

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 4638f24d9bb50..7314fd9ff0d9b 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -883,6 +883,7 @@ static inline unsigned long damon_sz_region(struct damon_region *r)
 	return r->ar.end - r->ar.start;
 }
 
+
 #define damon_for_each_region(r, t) \
 	list_for_each_entry(r, &t->regions_list, list)
 
-- 
2.47.3

