From: SJ Park <sj@kernel.org>
Date: Sat, 25 Jul 2026 11:45:59 -0700
Subject: [PATCH] revert mm/damon/sysfs: support pgidle_unset probe filter type

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 376c6ecbd4bf3..053ba2e53b969 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -782,10 +782,6 @@ damon_sysfs_filter_type_names[] = {
 		.type = DAMON_FILTER_TYPE_MEMCG,
 		.name = "memcg",
 	},
-	{
-		.type = DAMON_FILTER_TYPE_PGIDLE_UNSET,
-		.name = "pgidle_unset",
-	},
 };
 
 static ssize_t type_show(struct kobject *kobj,
-- 
2.47.3

