From: SeongJae Park <sj@kernel.org>
Date: Thu, 16 Oct 2025 13:46:42 -0700
Subject: [PATCH] revert "mm/damon/tests/core-kunit: test target->obsolete"

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/tests/core-kunit.h | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/mm/damon/tests/core-kunit.h b/mm/damon/tests/core-kunit.h
index 9027b60cabb4..22cec9ec20c5 100644
--- a/mm/damon/tests/core-kunit.h
+++ b/mm/damon/tests/core-kunit.h
@@ -1021,16 +1021,6 @@ static void damon_test_commit_targets(struct kunit *test)
 		i++;
 	KUNIT_EXPECT_EQ(test, i, 5);
 
-	damon_nth_target(1, src)->obsolete = true;
-	damon_nth_target(2, src)->obsolete = true;
-	damon_nth_target(4, src)->obsolete = true;
-	err = damon_commit_targets(dst, src);
-	KUNIT_EXPECT_EQ(test, err, 0);
-	i = 0;
-	damon_for_each_target(t, dst)
-		i++;
-	KUNIT_EXPECT_EQ(test, i, 2);
-
 	damon_destroy_ctx(src);
 	damon_destroy_ctx(dst);
 }
-- 
2.47.3

