From: SeongJae Park <sj@kernel.org>
Date: Tue, 18 Mar 2025 22:18:16 -0700
Subject: [PATCH] samples/damon/mtier: add young page filter

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 samples/damon/mtier.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/samples/damon/mtier.c b/samples/damon/mtier.c
index 820f8c13ba90..9917908f07ea 100644
--- a/samples/damon/mtier.c
+++ b/samples/damon/mtier.c
@@ -45,6 +45,7 @@ static struct damon_ctx *damon_sample_mtier_build_ctx(bool promote)
 	struct damon_region *region;
 	struct damos *scheme;
 	struct damos_quota_goal *quota_goal;
+	struct damos_filter *filter;
 
 	pr_info("start\n");
 
@@ -117,6 +118,12 @@ static struct damon_ctx *damon_sample_mtier_build_ctx(bool promote)
 	}
 	quota_goal->nid = 0;
 	damos_add_quota_goal(&scheme->quota, quota_goal);
+	filter = damos_new_filter(DAMOS_FILTER_TYPE_YOUNG, true, promote);
+	if (!filter) {
+		damon_destroy_ctx(ctx);
+		return NULL;
+	}
+	damos_add_filter(scheme, filter);
 	return ctx;
 }
 
-- 
2.39.5

