From: SeongJae Park <sj@kernel.org>
Date: Sat, 28 Jun 2025 14:00:24 -0700
Subject: [PATCH] samples/damon/mtier: use damon_initialized()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 samples/damon/mtier.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/samples/damon/mtier.c b/samples/damon/mtier.c
index 741c8146f8af..5ab5c590efe5 100644
--- a/samples/damon/mtier.c
+++ b/samples/damon/mtier.c
@@ -162,8 +162,6 @@ static void damon_sample_mtier_stop(void)
 	damon_destroy_ctx(ctxs[1]);
 }
 
-static bool damon_sample_mtier_init_called;
-
 static int damon_sample_mtier_enable_store(
 		const char *val, const struct kernel_param *kp)
 {
@@ -182,7 +180,7 @@ static int damon_sample_mtier_enable_store(
 	 * damon_new_ctx() since kzalloc() is not ready.  Let
 	 * damon_sample_mtier_init() handle.
 	 */
-	if (!damon_sample_mtier_init_called)
+	if (!damon_initialized())
 		return 0;
 
 	if (enable) {
@@ -199,7 +197,8 @@ static int __init damon_sample_mtier_init(void)
 {
 	int err = 0;
 
-	damon_sample_mtier_init_called = true;
+	if (!damon_initialized())
+		return -ENOMEM;
 
 	if (enable)
 		err = damon_sample_mtier_start();
-- 
2.39.5

