From: SeongJae Park <sj@kernel.org>
Date: Tue, 18 Mar 2025 22:18:56 -0700
Subject: [PATCH] samples/damon/mtier: use goto for cleanup

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 samples/damon/mtier.c | 39 +++++++++++++++------------------------
 1 file changed, 15 insertions(+), 24 deletions(-)

diff --git a/samples/damon/mtier.c b/samples/damon/mtier.c
index 9917908f07ea..87bdc1c29782 100644
--- a/samples/damon/mtier.c
+++ b/samples/damon/mtier.c
@@ -60,24 +60,18 @@ static struct damon_ctx *damon_sample_mtier_build_ctx(bool promote)
 		.access_bp = 400, .aggrs = 3,
 		.min_sample_us = 5000, .max_sample_us = 10000000,
 	};
-	if (damon_select_ops(ctx, DAMON_OPS_PADDR)) {
-		damon_destroy_ctx(ctx);
-		return NULL;
-	}
+	if (damon_select_ops(ctx, DAMON_OPS_PADDR))
+		goto free_out;
 
 	target = damon_new_target();
-	if (!target) {
-		damon_destroy_ctx(ctx);
-		return NULL;
-	}
+	if (!target)
+		goto free_out;
 	damon_add_target(ctx, target);
 	region = damon_new_region(
 			promote ? node1_start_addr : node0_start_addr,
 			promote ? node1_end_addr : node0_end_addr);
-	if (!region) {
-		damon_destroy_ctx(ctx);
-		return NULL;
-	}
+	if (!region)
+		goto free_out;
 	damon_add_region(region, target);
 
 	scheme = damon_new_scheme(
@@ -103,28 +97,25 @@ static struct damon_ctx *damon_sample_mtier_build_ctx(bool promote)
 			},
 			&(struct damos_watermarks){},
 			promote ? 0 : 1);	/* migrate target node id */
-	if (!scheme) {
-		damon_destroy_ctx(ctx);
-		return NULL;
-	}
+	if (!scheme)
+		goto free_out;
 	damon_set_schemes(ctx, &scheme, 1);
 	quota_goal = damos_new_quota_goal(
 			promote ? DAMOS_QUOTA_NODE_MEM_USED_BP :
 			DAMOS_QUOTA_NODE_MEM_FREE_BP,
 			promote ? 9970 : 50);
-	if (!quota_goal) {
-		damon_destroy_ctx(ctx);
-		return NULL;
-	}
+	if (!quota_goal)
+		goto free_out;
 	quota_goal->nid = 0;
 	damos_add_quota_goal(&scheme->quota, quota_goal);
 	filter = damos_new_filter(DAMOS_FILTER_TYPE_YOUNG, true, promote);
-	if (!filter) {
-		damon_destroy_ctx(ctx);
-		return NULL;
-	}
+	if (!filter)
+		goto free_out;
 	damos_add_filter(scheme, filter);
 	return ctx;
+free_out:
+	damon_destroy_ctx(ctx);
+	return NULL;
 }
 
 static int damon_sample_mtier_start(void)
-- 
2.39.5

