From: SeongJae Park <sj@kernel.org>
Date: Mon, 12 May 2025 17:57:28 -0700
Subject: [PATCH] samples/damon/mtier: support boot time enable setup

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 samples/damon/mtier.c | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/samples/damon/mtier.c b/samples/damon/mtier.c
index 7cee7ef0747e..ed0665362927 100644
--- a/samples/damon/mtier.c
+++ b/samples/damon/mtier.c
@@ -156,6 +156,8 @@ static void damon_sample_mtier_stop(void)
 	damon_destroy_ctx(ctxs[1]);
 }
 
+static bool damon_sample_mtier_init_called;
+
 static int damon_sample_mtier_enable_store(
 		const char *val, const struct kernel_param *kp)
 {
@@ -169,6 +171,14 @@ static int damon_sample_mtier_enable_store(
 	if (enable == enabled)
 		return 0;
 
+	/*
+	 * If called before init, probably from parse_args(), cannot call
+	 * damon_new_ctx() since kzalloc() is not ready.  Let
+	 * damon_sample_mtier_init() handle.
+	 */
+	if (!damon_sample_mtier_init_called)
+		return 0;
+
 	if (enable)
 		return damon_sample_mtier_start();
 	damon_sample_mtier_stop();
@@ -177,7 +187,13 @@ static int damon_sample_mtier_enable_store(
 
 static int __init damon_sample_mtier_init(void)
 {
-	return 0;
+	int err = 0;
+
+	damon_sample_mtier_init_called = true;
+
+	if (enable)
+		err = damon_sample_mtier_start();
+	return err;
 }
 
 module_init(damon_sample_mtier_init);
-- 
2.39.5

