From: SeongJae Park <sj@kernel.org>
Date: Mon, 12 May 2025 17:50:25 -0700
Subject: [PATCH] samples/damon/mtier: rename to have damon_sample_ prefix

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 samples/damon/mtier.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/samples/damon/mtier.c b/samples/damon/mtier.c
index a8950d85cdb1..5299b451e40e 100644
--- a/samples/damon/mtier.c
+++ b/samples/damon/mtier.c
@@ -12,6 +12,11 @@
 #include <linux/kernel.h>
 #include <linux/module.h>
 
+#ifdef MODULE_PARAM_PREFIX
+#undef MODULE_PARAM_PREFIX
+#endif
+#define MODULE_PARAM_PREFIX "damon_sample_mtier."
+
 static unsigned long node0_start_addr __read_mostly;
 module_param(node0_start_addr, ulong, 0600);
 
-- 
2.39.5

