From: SeongJae Park <sj@kernel.org>
Date: Tue, 12 Nov 2024 16:42:13 -0800
Subject: [PATCH] samples/damon/prcl: add min age

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 samples/damon/prcl.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/samples/damon/prcl.c b/samples/damon/prcl.c
index cf8c6f29d8a3..5f91046d2156 100644
--- a/samples/damon/prcl.c
+++ b/samples/damon/prcl.c
@@ -81,7 +81,9 @@ static int damon_sample_prcl_start(void)
 			.min_sz_region = PAGE_SIZE,
 			.max_sz_region = ULONG_MAX,
 			.min_nr_accesses = 0,
-			.max_nr_accesses = 0},
+			.max_nr_accesses = 0,
+			.min_age_region = 50,
+			.max_age_region = ULONG_MAX},
 			DAMOS_PAGEOUT,
 			0,
 			&(struct damos_quota){},
-- 
2.39.5

