From: SeongJae Park <sj@kernel.org>
Date: Sat, 28 Jun 2025 14:00:24 -0700
Subject: [PATCH] samples/damon/mtier: use damon_initialized()

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 samples/damon/mtier.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/samples/damon/mtier.c b/samples/damon/mtier.c
index 7ebd352138e4..74ea881d68b7 100644
--- a/samples/damon/mtier.c
+++ b/samples/damon/mtier.c
@@ -193,8 +193,6 @@ static void damon_sample_mtier_stop(void)
 	damon_destroy_ctx(ctxs[1]);
 }
 
-static bool init_called;
-
 static int damon_sample_mtier_enable_store(
 		const char *val, const struct kernel_param *kp)
 {
@@ -208,6 +206,9 @@ static int damon_sample_mtier_enable_store(
 	if (enabled == is_enabled)
 		return 0;
 
+	if (!damon_initialized())
+		return 0;
+
 	if (enabled) {
 		err = damon_sample_mtier_start();
 		if (err)
@@ -222,13 +223,15 @@ static int __init damon_sample_mtier_init(void)
 {
 	int err = 0;
 
-	init_called = true;
+	if (!damon_initialized())
+		return -ENOMEM;
+
 	if (enabled) {
 		err = damon_sample_mtier_start();
 		if (err)
 			enabled = false;
 	}
-	return 0;
+	return err;
 }
 
 module_init(damon_sample_mtier_init);
-- 
2.39.5

