From: SeongJae Park <sj@kernel.org>
Date: Tue, 18 Mar 2025 22:20:14 -0700
Subject: [PATCH] samples/damon/mtier: remove meaningless start log

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 samples/damon/mtier.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/samples/damon/mtier.c b/samples/damon/mtier.c
index 87bdc1c29782..3390b7d30c26 100644
--- a/samples/damon/mtier.c
+++ b/samples/damon/mtier.c
@@ -47,8 +47,6 @@ static struct damon_ctx *damon_sample_mtier_build_ctx(bool promote)
 	struct damos_quota_goal *quota_goal;
 	struct damos_filter *filter;
 
-	pr_info("start\n");
-
 	ctx = damon_new_ctx();
 	if (!ctx)
 		return NULL;
-- 
2.39.5

