From: SeongJae Park <sj@kernel.org>
Date: Fri, 18 Jul 2025 18:26:29 -0700
Subject: [PATCH] selftest/damon/sysfs.py: test dests

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 tools/testing/selftests/damon/sysfs.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tools/testing/selftests/damon/sysfs.py b/tools/testing/selftests/damon/sysfs.py
index 70ec599e5bc5..f1f57a89499f 100755
--- a/tools/testing/selftests/damon/sysfs.py
+++ b/tools/testing/selftests/damon/sysfs.py
@@ -200,6 +200,12 @@ def main():
                 watermarks=_damon_sysfs.DamosWatermarks(
                     metric = 'free_mem_rate', interval = 5000000, # 5 secodns
                     high = 500, mid = 400, low = 50),
+                target_nid=1,
+                apply_interval_us=1000000,
+                dests=_damon_sysfs.DamosDests(
+                    dests=[_damon_sysfs.DamosDest(id=1, weight=30),
+                           _damon_sysfs.DamosDest(id=0, weight=70)]),
+
                 )])
     context.idx = 0
     context.kdamond = kdamonds.kdamonds[0]
-- 
2.39.5

