From: SeongJae Park <sj@kernel.org>
Date: Sun, 13 Jul 2025 13:10:01 -0700
Subject: [PATCH] selftests/damon/sysfs: test ops id

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 tools/testing/selftests/damon/sysfs.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tools/testing/selftests/damon/sysfs.py b/tools/testing/selftests/damon/sysfs.py
index b67e90410bbb..34b1d3494046 100755
--- a/tools/testing/selftests/damon/sysfs.py
+++ b/tools/testing/selftests/damon/sysfs.py
@@ -46,6 +46,8 @@ def main():
         fail('number of contexts', status)
 
     ctx = status['contexts'][0]
+    if ctx['ops']['id'] != '(enum damon_ops_id)DAMON_OPS_PADDR':
+        fail('ops id', status)
     attrs = ctx['attrs']
     if attrs['sample_interval'] != 5000:
         fail('sample interval', status)
-- 
2.39.5

