From: SeongJae Park <sj@kernel.org>
Date: Sat, 21 Jun 2025 14:08:39 -0700
Subject: [PATCH] selftests/damon/sysfs.py: test intervals

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 tools/testing/selftests/damon/sysfs.py | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/tools/testing/selftests/damon/sysfs.py b/tools/testing/selftests/damon/sysfs.py
index add04638be65..317621d066a6 100755
--- a/tools/testing/selftests/damon/sysfs.py
+++ b/tools/testing/selftests/damon/sysfs.py
@@ -35,6 +35,19 @@ def main():
     if len(status['contexts']) != 1:
         print('number of contexts: %d' % len(status['contexts']))
         exit(1)
+
+    ctx = status['contexts'][0]
+    attrs = ctx['attrs']
+    if attrs['sample_interval'] != 5000:
+        print('unexpected sample interval')
+        exit(1)
+    if attrs['aggr_interval'] != 100000:
+        print('unexpected aggr interval')
+        exit(1)
+    if attrs['ops_update_interval'] != 1000000:
+        print('unexpected ops update interval')
+        exit(1)
+
     kdamonds.stop()
 
 if __name__ == '__main__':
-- 
2.39.5

