From: SeongJae Park <sj@kernel.org>
Date: Fri, 18 Jul 2025 18:16:27 -0700
Subject: [PATCH] selftests/damon/sysfs: test watermarks

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 tools/testing/selftests/damon/sysfs.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tools/testing/selftests/damon/sysfs.py b/tools/testing/selftests/damon/sysfs.py
index 7e4d0136817c..70ec599e5bc5 100755
--- a/tools/testing/selftests/damon/sysfs.py
+++ b/tools/testing/selftests/damon/sysfs.py
@@ -197,6 +197,9 @@ def main():
                     weight_sz_permil=20,
                     weight_nr_accesses_permil=200,
                     weight_age_permil=1000),
+                watermarks=_damon_sysfs.DamosWatermarks(
+                    metric = 'free_mem_rate', interval = 5000000, # 5 secodns
+                    high = 500, mid = 400, low = 50),
                 )])
     context.idx = 0
     context.kdamond = kdamonds.kdamonds[0]
-- 
2.39.5

