From: SJ Park <sj@kernel.org>
Date: Sat, 25 Jul 2026 10:55:16 -0700
Subject: [PATCH] stat fixup

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/paddr.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index d2a59f666cfdc..61f45a5c2fb67 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -472,21 +472,16 @@ static unsigned long damon_pa_stat(struct damon_region *r,
 
 	addr = damon_pa_phys_addr(r->ar.start, addr_unit);
 	while (addr < damon_pa_phys_addr(r->ar.end, addr_unit)) {
-		phys_addr_t folio_addr;
-
 		folio = damon_get_folio(PHYS_PFN(addr));
 		if (damon_pa_invalid_damos_folio(folio, s)) {
 			addr += PAGE_SIZE;
 			continue;
 		}
-		folio_addr = PFN_PHYS(folio_pfn(folio));
-		if (folio_addr < addr)
-			goto put_folio;
 
 		if (!damos_pa_filter_out(s, folio))
 			*sz_filter_passed += folio_size(folio) / addr_unit;
 put_folio:
-		addr = folio_addr + folio_size(folio);
+		addr = PFN_PHYS(folio_pfn(folio)) + folio_size(folio);
 		folio_put(folio);
 	}
 	s->last_applied = folio;
-- 
2.47.3

