From: SeongJae Park <sj@kernel.org>
Date: Sat, 21 Jun 2025 14:54:52 -0700
Subject: [PATCH] sysfs.py: test action, apply_itnerval, target_nid

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 tools/testing/selftests/damon/sysfs.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tools/testing/selftests/damon/sysfs.py b/tools/testing/selftests/damon/sysfs.py
index 9b042dabd48f..ef8573db0590 100755
--- a/tools/testing/selftests/damon/sysfs.py
+++ b/tools/testing/selftests/damon/sysfs.py
@@ -81,6 +81,12 @@ def main():
             'max_age_region': 2**32 - 1,
             }:
         fail('damos pattern', status)
+    if scheme['action'] != 9:   # stat
+        fail('damos action', status)
+    if scheme['apply_interval_us'] != 0:
+        fail('damos apply interval', status)
+    if scheme['target_nid'] != -1:
+        fail('damos target nid', status)
 
     kdamonds.stop()
 
-- 
2.39.5

