From: SeongJae Park <sj@kernel.org>
Date: Sat, 21 Jun 2025 14:57:19 -0700
Subject: [PATCH] sysfs.py: test quota

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 tools/testing/selftests/damon/sysfs.py | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/tools/testing/selftests/damon/sysfs.py b/tools/testing/selftests/damon/sysfs.py
index ef8573db0590..b50cf243e567 100755
--- a/tools/testing/selftests/damon/sysfs.py
+++ b/tools/testing/selftests/damon/sysfs.py
@@ -88,6 +88,18 @@ def main():
     if scheme['target_nid'] != -1:
         fail('damos target nid', status)
 
+    if scheme['quota'] != {
+            'reset_interval': 0,
+            'ms': 0,
+            'sz': 0,
+            'goals': [],
+            'esz': 0,
+            'weight_sz': 0,
+            'weight_nr_accesses': 0,
+            'weight_age': 0,
+            }:
+        fail('damos quota', status)
+
     kdamonds.stop()
 
 if __name__ == '__main__':
-- 
2.39.5

