From: SeongJae Park <sj@kernel.org>
Date: Sat, 21 Jun 2025 14:58:24 -0700
Subject: [PATCH] sysfs.py: test wmarks

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 tools/testing/selftests/damon/sysfs.py | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/tools/testing/selftests/damon/sysfs.py b/tools/testing/selftests/damon/sysfs.py
index b50cf243e567..ec8efbf7f056 100755
--- a/tools/testing/selftests/damon/sysfs.py
+++ b/tools/testing/selftests/damon/sysfs.py
@@ -100,6 +100,15 @@ def main():
             }:
         fail('damos quota', status)
 
+    if scheme['wmarks'] != {
+            'metric': 0,
+            'interval': 0,
+            'high': 0,
+            'mid': 0,
+            'low': 0,
+            }:
+        fail('damos wmarks', status)
+
     kdamonds.stop()
 
 if __name__ == '__main__':
-- 
2.39.5

