From: SeongJae Park <sj@kernel.org>
Date: Sat, 21 Jun 2025 14:16:25 -0700
Subject: [PATCH] sysfs: test adasptive_targets and schemes

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 tools/testing/selftests/damon/sysfs.py | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/tools/testing/selftests/damon/sysfs.py b/tools/testing/selftests/damon/sysfs.py
index f637f83ffc39..cf92ba7668f6 100755
--- a/tools/testing/selftests/damon/sysfs.py
+++ b/tools/testing/selftests/damon/sysfs.py
@@ -62,6 +62,14 @@ def main():
         print('unexpected max_nr_regions')
         exit(1)
 
+    if ctx['adaptive_targets'] != []:
+        print('unexpected adaptive_targets')
+        exit(1)
+
+    if ctx['schemes'] != []:
+        print('unexpected schemes')
+        exit(1)
+
     kdamonds.stop()
 
 if __name__ == '__main__':
-- 
2.39.5

