From: SeongJae Park <sj@kernel.org>
Date: Sat, 21 Jun 2025 14:12:00 -0700
Subject: [PATCH] sysfs.py: test intervals goal

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 tools/testing/selftests/damon/sysfs.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tools/testing/selftests/damon/sysfs.py b/tools/testing/selftests/damon/sysfs.py
index 317621d066a6..8bb7259e2827 100755
--- a/tools/testing/selftests/damon/sysfs.py
+++ b/tools/testing/selftests/damon/sysfs.py
@@ -48,6 +48,12 @@ def main():
         print('unexpected ops update interval')
         exit(1)
 
+    if attrs['intervals_goal'] != {
+            'access_bp': 0, 'aggrs': 0,
+            'min_sample_us': 0, 'max_sample_us': 0}:
+        print('unexpected intervals goal')
+        exit(1)
+
     kdamonds.stop()
 
 if __name__ == '__main__':
-- 
2.39.5

