From: SeongJae Park <sj@kernel.org>
Date: Sat, 21 Jun 2025 14:16:02 -0700
Subject: [PATCH] sysfs.py: test {min,max}_nr_regions

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 tools/testing/selftests/damon/sysfs.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/tools/testing/selftests/damon/sysfs.py b/tools/testing/selftests/damon/sysfs.py
index 20961e248737..f637f83ffc39 100755
--- a/tools/testing/selftests/damon/sysfs.py
+++ b/tools/testing/selftests/damon/sysfs.py
@@ -55,6 +55,13 @@ def main():
         print('unexpected intervals goal')
         exit(1)
 
+    if attrs['min_nr_regions'] != 10:
+        print('unexpected min_nr_regions')
+        exit(1)
+    if attrs['max_nr_regions'] != 1000:
+        print('unexpected max_nr_regions')
+        exit(1)
+
     kdamonds.stop()
 
 if __name__ == '__main__':
-- 
2.39.5

